--- libreoffice-5.0.2.1/sdext/StaticLibrary_pdfimport_s.mk~	2015-11-18 08:03:28.000000000 +0100
+++ libreoffice-5.0.2.1/sdext/StaticLibrary_pdfimport_s.mk	2015-11-20 08:35:33.833069957 +0100
@@ -20,6 +20,11 @@
     zlib \
 ))
 
+$(if $(SYSTEM_BOOST), \
+    $(call gb_LinkTarget_add_ldflags,$(1),$(BOOST_LDFLAGS)) \
+    $(call gb_LinkTarget_add_libs,$(1),$(BOOST_SYSTEM_LIB)) \
+)
+
 $(eval $(call gb_StaticLibrary_use_custom_headers,pdfimport_s,sdext/pdfimport))
 
 $(eval $(call gb_StaticLibrary_set_include,pdfimport_s,\
