diff -ur libreoffice-5.0.2.1/filter/Library_svgfilter.mk libreoffice-5.0.2.1-boost/filter/Library_svgfilter.mk
--- libreoffice-5.0.2.1/filter/Library_svgfilter.mk	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1-boost/filter/Library_svgfilter.mk	2015-11-20 23:46:59.702815320 +0100
@@ -76,4 +76,6 @@
 		filter/source/svg/svgwriter) \
 ))
 
+$(eval $(call gb_Library_add_libs,svgfilter,-lboost_system))
+
 # vim: set noet sw=4 ts=4:
diff -ur libreoffice-5.0.2.1/sdext/CppunitTest_sdext_pdfimport.mk libreoffice-5.0.2.1-boost/sdext/CppunitTest_sdext_pdfimport.mk
--- libreoffice-5.0.2.1/sdext/CppunitTest_sdext_pdfimport.mk	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1-boost/sdext/CppunitTest_sdext_pdfimport.mk	2015-11-20 23:43:57.694492773 +0100
@@ -36,6 +36,8 @@
 	$(gb_UWINAPI) \
 ))
 
+$(eval $(call gb_CppunitTest_add_libs,sdext_pdfimport,-lboost_system))
+
 $(eval $(call gb_CppunitTest_use_static_libraries,sdext_pdfimport,\
     pdfimport_s \
 ))
diff -ur libreoffice-5.0.2.1/sdext/Executable_pdf2xml.mk libreoffice-5.0.2.1-boost/sdext/Executable_pdf2xml.mk
--- libreoffice-5.0.2.1/sdext/Executable_pdf2xml.mk	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1-boost/sdext/Executable_pdf2xml.mk	2015-11-20 23:44:04.531218795 +0100
@@ -41,4 +41,6 @@
     sdext/source/pdfimport/test/pdf2xml \
 ))
 
+$(eval $(call gb_Executable_add_libs,pdf2xml,-lboost_system))
+
 # vim:set noet sw=4 ts=4:
diff -ur libreoffice-5.0.2.1/sdext/Executable_pdfunzip.mk libreoffice-5.0.2.1-boost/sdext/Executable_pdfunzip.mk
--- libreoffice-5.0.2.1/sdext/Executable_pdfunzip.mk	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1-boost/sdext/Executable_pdfunzip.mk	2015-11-20 23:44:09.571262711 +0100
@@ -33,4 +33,6 @@
     sdext/source/pdfimport/test/pdfunzip \
 ))
 
+$(eval $(call gb_Executable_add_libs,pdfunzip,-lboost_system))
+
 # vim:set noet sw=4 ts=4:
diff -ur libreoffice-5.0.2.1/sdext/Library_pdfimport.mk libreoffice-5.0.2.1-boost/sdext/Library_pdfimport.mk
--- libreoffice-5.0.2.1/sdext/Library_pdfimport.mk	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1-boost/sdext/Library_pdfimport.mk	2015-11-20 23:44:24.911397184 +0100
@@ -40,4 +40,6 @@
     sdext/source/pdfimport/services \
 ))
 
+$(eval $(call gb_Library_add_libs,pdfimport,-lboost_system))
+
 # vim:set noet sw=4 ts=4:
