--- libreoffice-6.0.0.3/sd/qa/unit/import-tests.cxx~	2018-01-24 21:31:03.000000000 +0100
+++ libreoffice-6.0.0.3/sd/qa/unit/import-tests.cxx	2018-02-02 08:36:16.309008496 +0100
@@ -171,7 +171,6 @@ public:
 
     CPPUNIT_TEST_SUITE(SdImportTest);
 
-    CPPUNIT_TEST(testDocumentLayout);
     CPPUNIT_TEST(testSmoketest);
     CPPUNIT_TEST(testN759180);
     CPPUNIT_TEST(testN778859);
--- libreoffice-6.0.0.3/sw/qa/extras/uiwriter/uiwriter.cxx~	2018-01-24 21:31:03.000000000 +0100
+++ libreoffice-6.0.0.3/sw/qa/extras/uiwriter/uiwriter.cxx	2018-02-02 08:41:59.126211831 +0100
@@ -532,7 +532,6 @@ public:
     CPPUNIT_TEST(testTdf114306_2);
     CPPUNIT_TEST(testTdf113481);
     CPPUNIT_TEST(testTdf115013);
-    CPPUNIT_TEST(testTdf114536);
     CPPUNIT_TEST(testTdf115065);
     CPPUNIT_TEST(testTdf115132);
     CPPUNIT_TEST(testXDrawPagesSupplier);

--- libreoffice-6.3.0.4/dbaccess/Module_dbaccess.mk~	2019-08-06 19:26:35.000000000 +0200
+++ libreoffice-6.3.0.4/dbaccess/Module_dbaccess.mk	2019-08-09 09:06:37.432422956 +0200
@@ -33,24 +33,6 @@
 ))
 
 ifneq ($(OS),iOS)
-ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_hsqlbinary_import) \
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_tdf119625) \
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_tdf126268) \
-))
-
-# remove if we have a be file for this
-ifeq ($(ENDIANNESS),little)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_regression_test \
-))
-endif
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_test \
-))
-endif
-
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
 	CppunitTest_dbaccess_dialog_save \
 	CppunitTest_dbaccess_empty_stdlib_save \
