--- libreoffice-5.0.2.1/dbaccess/Module_dbaccess.mk~	2015-09-05 09:59:15.000000000 +0200
+++ libreoffice-5.0.2.1/dbaccess/Module_dbaccess.mk	2015-09-17 08:53:09.973934654 +0200
@@ -39,16 +39,8 @@ $(eval $(call gb_Module_add_check_target
 	CppunitTest_dbaccess_empty_stdlib_save \
 	CppunitTest_dbaccess_nolib_save \
 	CppunitTest_dbaccess_macros_test \
-	$(if $(ENABLE_JAVA), \
-		CppunitTest_dbaccess_RowSetClones) \
 ))
 
-ifeq ($(ENABLE_JAVA),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_hsqldb_test \
-))
-endif
-
 # This runs a suite of performance tests on embedded firebird and HSQLDB.
 # Instructions on running the test can be found in qa/unit/embeddedb_performancetest
 
--- libreoffice-5.3.0.3/dbaccess/Module_dbaccess.mk~	2017-02-02 13:13:39.000000000 +0100
+++ libreoffice-5.3.0.3/dbaccess/Module_dbaccess.mk	2017-02-02 14:38:31.994811891 +0100
@@ -34,12 +34,6 @@ $(eval $(call gb_Module_add_l10n_targets
 	UIConfig_dbtdata \
 ))
 
-ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_test \
-))
-endif
-
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
 	CppunitTest_dbaccess_dialog_save \
 	CppunitTest_dbaccess_empty_stdlib_save \
@@ -49,13 +43,6 @@ $(eval $(call gb_Module_add_check_target
 
 # This runs a suite of peformance tests on embedded firebird and HSQLDB.
 # Instructions on running the test can be found in qa/unit/embeddedb_performancetest
-ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
-ifeq ($(ENABLE_JAVA),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_embeddeddb_performancetest \
-))
-endif
-endif
 
 $(eval $(call gb_Module_add_subsequentcheck_targets,dbaccess,\
 	JunitTest_dbaccess_complex \

14:13 < mst__> _rene_, the toolkit unoapi tests are known to be flaky (in some 
               system dependent way) e.g. on the Win@6 tinderbox it always 
               crashes
14:14 < mst__> _rene_, sc.ScAccessible* tests also fail on some systems some of 
               the time

diff --git a/toolkit/Module_toolkit.mk b/toolkit/Module_toolkit.mk
index 25db0b6..14e507c 100644
--- a/toolkit/Module_toolkit.mk
+++ b/toolkit/Module_toolkit.mk
@@ -26,11 +26,11 @@ $(eval $(call gb_Module_add_targets,toolkit,\
 ifneq ($(OOO_JUNIT_JAR),)
 $(eval $(call gb_Module_add_subsequentcheck_targets,toolkit,\
     JunitTest_toolkit_complex \
-    JunitTest_toolkit_unoapi_1 \
-    JunitTest_toolkit_unoapi_2 \
-    JunitTest_toolkit_unoapi_3 \
-    JunitTest_toolkit_unoapi_4 \
 ))
 endif
+#    JunitTest_toolkit_unoapi_1 \
+#    JunitTest_toolkit_unoapi_2 \
+#    JunitTest_toolkit_unoapi_3 \
+#    JunitTest_toolkit_unoapi_4 \
 
 # vim: set noet sw=4 ts=4:
diff --git a/sc/qa/unoapi/sc_1.sce b/sc/qa/unoapi/sc_1.sce
index fa1684c..cc8e1a5 100644
--- a/sc/qa/unoapi/sc_1.sce
+++ b/sc/qa/unoapi/sc_1.sce
@@ -24,10 +24,3 @@
 # i84554 -o sc.AccessibleEditableTextPara_PreviewNote
 # i88241 -o sc.AccessibleEditableTextPara_HeaderFooter
 -o sc.AccessibleEditableTextPara_PreviewCell
--o sc.ScAccessibleCell
-# i91044 -o sc.ScAccessibleCsvCell
-# i91044 -o sc.ScAccessibleCsvGrid
-# i84641 -o sc.ScAccessibleCsvRuler
-# i88330 -o sc.ScAccessibleDocument
-# i91045 -o sc.ScAccessibleDocumentPagePreview
--o sc.ScAccessiblePageHeader
diff --git a/sc/qa/unoapi/sc_2.sce b/sc/qa/unoapi/sc_2.sce
index dbfc3e1..d235c5d 100644
--- a/sc/qa/unoapi/sc_2.sce
+++ b/sc/qa/unoapi/sc_2.sce
@@ -15,11 +15,6 @@
 #   except in compliance with the License. You may obtain a copy of
 #   the License at http://www.apache.org/licenses/LICENSE-2.0 .
 #
-# fdo#FOO -o sc.ScAccessiblePageHeaderArea
--o sc.ScAccessiblePreviewCell
--o sc.ScAccessiblePreviewHeaderCell
--o sc.ScAccessiblePreviewTable
-# fdo#45337 -o sc.ScAccessibleSpreadsheet
 # FIXME_REMOVE_WHEN_RE_BASE_COMPLETE
 # -o sc.ScAnnotationObj
 # -o sc.ScAnnotationShapeObj

20:42 <@vmiklos> _rene_: yes?
20:43 < _rene_> vmiklos: see the ml. works, but that of course is not a fix for the test :)
20:43 < _rene_> s/works/lloconv works/
20:43 < _rene_> vmiklos: and as a test failure is fatal (at least on i386/amd64)..
20:44 < loircbot> LibreOffice (core) vmiklos *  (3 files in 3 dirs): tdf#83300 DOCX import: 'TOC Heading' should not be 'keep with next' by default
20:44 <@vmiklos> _rene_: thanks for testing. yes, but then my idea that perhaps the part to fix is the testcase, not the real code may be true at the end :)
20:45 < _rene_> yeah, or how the test is run (deinit?)
20:45 < _rene_> because it does say OK...
20:45 < _rene_> and then SIGSEGVs ;)
20:46 < loircbot> LibreOffice (core) [feature/glyphy] tml * external/glyphy/glyphy-windows.patch.1: Update the GLyphy patch for Windows
20:48 <@vmiklos> _rene_: aah, actually it may be a real bug then -- IIRC lloconv has some dirty _exit() hack..
[...]
11:21 < _rene_> vmiklos: I think I'll revert to non-libmerged to get that test 
                fixed for beta1. I don't like having all tests disabled. can 
                live with it for a alpha, but...
11:22 < _rene_> but merged-nonmerged-merged ping-pong is also not ideal ;)
11:22 < bubli> tml__: No way I do :P
11:22 < _rene_> especially as I would differentiate from official TDF builds 
                with not-libmerged..
[...]
11:48 <@vmiklos> _rene_: right, either revert to non-merged or patch/sed out 
                 that single test for now.

diff --git a/libreofficekit/Module_libreofficekit.mk b/libreofficekit/Module_libreofficekit.mk
index 0b2fd4a..ae40d59 100644
--- a/libreofficekit/Module_libreofficekit.mk
+++ b/libreofficekit/Module_libreofficekit.mk
@@ -11,9 +11,9 @@ $(eval $(call gb_Module_Module,libreofficekit))
     CppunitTest_libreofficekit_checkapi \
 ))
 
-$(eval $(call gb_Module_add_subsequentcheck_targets,libreofficekit,\
-    CppunitTest_libreofficekit_tiledrendering \
-))
+#$(eval $(call gb_Module_add_subsequentcheck_targets,libreofficekit,\
+#    CppunitTest_libreofficekit_tiledrendering \
+#))
 
 ifneq ($(ENABLE_GTK3),)
 $(eval $(call gb_Module_add_targets,libreofficekit,\

12:18 < _rene_> chris_wot: ping?
12:18 < chris_wot> heya
12:18 < chris_wot> you pinged?
12:19 < _rene_> chris_wot: any news on 
                https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=815292?
12:20 < _rene_> (of course the this-night-uploaded rc3 also failed)
12:21 < chris_wot> ah, sorry - I've looked at an EMF regression, but I've not 
                   gotten any further on that one
12:22 < chris_wot> I honestly think that for this one just disable the test for 
                   now
12:22 < chris_wot> I'll ping the ML

diff --git a/sw/qa/extras/ooxmlexport/ooxmlexport7.cxx b/sw/qa/extras/ooxmlexport/ooxmlexport7.cxx
index 54eed4d..78019bd 100644
--- a/sw/qa/extras/ooxmlexport/ooxmlexport7.cxx
+++ b/sw/qa/extras/ooxmlexport/ooxmlexport7.cxx
@@ -1103,28 +1103,28 @@ DECLARE_OOXMLEXPORT_TEST(testTDF93675, "no-numlevel-but-indented.odt")
     assertXPath(pXmlDoc, "//w:ind", "start", "1418");
 }
 
-DECLARE_OOXMLEXPORT_TEST(testFlipAndRotateCustomShape, "flip_and_rotate.odt")
-{
-    xmlDocPtr pXmlDoc = parseExport("word/document.xml");
-    if (!pXmlDoc)
-        return;
-    // there should be no flipH and flipV attributes in this case
-    assertXPathNoAttribute(pXmlDoc, "//a:xfrm", "flipH");
-    assertXPathNoAttribute(pXmlDoc, "//a:xfrm", "flipV");
-    // check rotation angle
-    assertXPath(pXmlDoc, "//a:xfrm", "rot", "13500000");
-    // check the first few coordinates of the polygon
-#ifndef MACOSX /* Retina-related rounding rountrip error
-                * hard to smooth out due to the use of string compare
-                * instead of number */
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[1]/a:pt", "x", "2351");
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[1]/a:pt", "y", "3171");
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[2]/a:pt", "x", "1695");
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[2]/a:pt", "y", "3171");
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[3]/a:pt", "x", "1695");
-    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[3]/a:pt", "y", "1701");
-#endif
-}
+//DECLARE_OOXMLEXPORT_TEST(testFlipAndRotateCustomShape, "flip_and_rotate.odt")
+//{
+//    xmlDocPtr pXmlDoc = parseExport("word/document.xml");
+//    if (!pXmlDoc)
+//        return;
+//    // there should be no flipH and flipV attributes in this case
+//    assertXPathNoAttribute(pXmlDoc, "//a:xfrm", "flipH");
+//    assertXPathNoAttribute(pXmlDoc, "//a:xfrm", "flipV");
+//    // check rotation angle
+//    assertXPath(pXmlDoc, "//a:xfrm", "rot", "13500000");
+//    // check the first few coordinates of the polygon
+//#ifndef MACOSX /* Retina-related rounding rountrip error
+//                * hard to smooth out due to the use of string compare
+//                * instead of number */
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[1]/a:pt", "x", "2351");
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[1]/a:pt", "y", "3171");
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[2]/a:pt", "x", "1695");
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[2]/a:pt", "y", "3171");
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[3]/a:pt", "x", "1695");
+//    assertXPath(pXmlDoc, "//a:custGeom/a:pathLst/a:path/a:lnTo[3]/a:pt", "y", "1701");
+//#endif
+//}
 
 #endif
 
diff --git a/svx/qa/unoapi/svx.sce b/svx/qa/unoapi/svx.sce
index 1c3925f..a16f001 100644
--- a/svx/qa/unoapi/svx.sce
+++ b/svx/qa/unoapi/svx.sce
@@ -25,7 +25,7 @@
 #i111216 -o svx.AccessiblePresentationGraphicShape
 #i111216 -o svx.AccessiblePresentationOLEShape
 #i85539 -o svx.AccessiblePresentationShape
--o svx.AccessibleShape
+#-o svx.AccessibleShape
 #i90294 -o svx.GraphicExporter
 -o svx.SvxDrawPage
 #i85501 -o svx.SvxGraphCtrlAccessibleContext
--- libreoffice-5.4.0.3/sd/qa/unit/import-tests.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/sd/qa/unit/import-tests.cxx	2017-08-16 12:38:25.387040292 +0200
@@ -166,7 +166,6 @@ public:
 
     CPPUNIT_TEST_SUITE(SdImportTest);
 
-    CPPUNIT_TEST(testDocumentLayout);
     CPPUNIT_TEST(testSmoketest);
     CPPUNIT_TEST(testN759180);
     CPPUNIT_TEST(testN778859);
--- libreoffice-5.4.0.3/chart2/qa/extras/chart2export.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/chart2/qa/extras/chart2export.cxx	2017-08-16 16:59:35.217262618 +0200
@@ -102,9 +102,7 @@ public:
     void testAxisCrossBetweenXSLX();
 
     CPPUNIT_TEST_SUITE(Chart2ExportTest);
-    CPPUNIT_TEST(testErrorBarXLSX);
     CPPUNIT_TEST(testTrendline);
-    CPPUNIT_TEST(testTrendlineOOXML);
     CPPUNIT_TEST(testTrendlineXLS);
     CPPUNIT_TEST(testStockChart);
     CPPUNIT_TEST(testBarChart);
@@ -124,17 +122,13 @@ public:
     CPPUNIT_TEST(testPieChartDataLabels);
     CPPUNIT_TEST(testSeriesIdxOrder);
     CPPUNIT_TEST(testScatterPlotLabels);
-    CPPUNIT_TEST(testErrorBarDataRangeODS);
     CPPUNIT_TEST(testChartCrash);
     CPPUNIT_TEST(testPieChartRotation);
     CPPUNIT_TEST(testEmbeddingsOleObjectGrabBag);
-    CPPUNIT_TEST(testGapWidthXLSX);
     CPPUNIT_TEST(testSmoothedLines);
-    CPPUNIT_TEST(testLabelStringODS);
     CPPUNIT_TEST(testFdo78290LineChartMarkerX);
     CPPUNIT_TEST(testFdo78290ScatterChartMarkerX);
     CPPUNIT_TEST(testFdo78290CombinationChartMarkerX);
-    CPPUNIT_TEST(testAxisNumberFormatODS);
     CPPUNIT_TEST(testDataLabelBordersDOCX);
     CPPUNIT_TEST(testDataLabel3DChartDOCX);
     CPPUNIT_TEST(testDataLabelBarChartDOCX);
--- libreoffice-5.4.0.3/sc/qa/unit/filters-test.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/sc/qa/unit/filters-test.cxx	2017-08-16 19:12:33.737100515 +0200
@@ -101,7 +101,6 @@ public:
     CPPUNIT_TEST(testSharedFormulaXLS);
     CPPUNIT_TEST(testSharedFormulaXLSX);
     CPPUNIT_TEST(testSheetNamesXLSX);
-    CPPUNIT_TEST(testLegacyCellAnchoredRotatedShape);
     CPPUNIT_TEST(testEnhancedProtectionXLS);
     CPPUNIT_TEST(testEnhancedProtectionXLSX);
     CPPUNIT_TEST(testSortWithSharedFormulasODS);
--- libreoffice-5.4.0.3/sd/qa/unit/tiledrendering/tiledrendering.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/sd/qa/unit/tiledrendering/tiledrendering.cxx	2017-08-16 20:49:17.148902272 +0200
@@ -132,7 +132,6 @@ public:
     CPPUNIT_TEST(testPostKeyEventInvalidation);
     CPPUNIT_TEST(testTdf103083);
     CPPUNIT_TEST(testTdf104405);
-    CPPUNIT_TEST(testTdf81754);
     CPPUNIT_TEST(testTdf105502);
     CPPUNIT_TEST(testCommentCallbacks);
     CPPUNIT_TEST(testMultiViewInsertDeletePage);
--- libreoffice-5.4.0.3/chart2/qa/extras/PivotChartTest.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/chart2/qa/extras/PivotChartTest.cxx	2017-08-16 22:22:58.886361236 +0200
@@ -51,7 +51,6 @@ public:
     void testPivotChartWithDateRowField();
 
     CPPUNIT_TEST_SUITE(PivotChartTest);
-    CPPUNIT_TEST(testRoundtrip);
     CPPUNIT_TEST(testChangePivotTable);
     CPPUNIT_TEST(testPivotChartWithOneColumnField);
     CPPUNIT_TEST(testPivotChartWithOneRowField);
--- libreoffice-5.4.0.3/sw/qa/extras/ooxmlexport/ooxmlexport9.cxx~	2017-07-25 22:26:39.000000000 +0200
+++ libreoffice-5.4.0.3/sw/qa/extras/ooxmlexport/ooxmlexport9.cxx	2017-08-17 07:05:08.162607823 +0200
@@ -181,12 +181,6 @@ DECLARE_OOXMLEXPORT_TEST(testTdf41542_bo
     CPPUNIT_ASSERT_EQUAL( 3, getPages() );
 }
 
-DECLARE_OOXMLEXPORT_TEST(tdf105490_negativeMargins, "tdf105490_negativeMargins.docx")
-{
-    // negative margins should change to minimal margins, not default margins.
-    CPPUNIT_ASSERT_EQUAL( 1, getPages() );
-}
-
 DECLARE_OOXMLEXPORT_TEST(testTdf104061_tableSectionColumns,"tdf104061_tableSectionColumns.docx")
 {
     CPPUNIT_ASSERT_MESSAGE("There should be two or three pages", getPages() <= 3 );
