--- libreoffice-7.5.9.2/sd/qa/unit/import-tests.cxx.orig	2024-08-27 19:24:27.557188992 +0200
+++ libreoffice-7.5.9.2/sd/qa/unit/import-tests.cxx	2024-08-27 19:26:12.767773974 +0200
@@ -82,7 +82,6 @@ public:
     {
     }
 
-    void testDocumentLayout();
     void testTdf154363();
     void testTdf152434();
     void testStandardConnectors();
--- libreoffice-7.5.1.2/dbaccess/Module_dbaccess.mk.orig	2023-03-09 20:09:43.192750947 +0100
+++ libreoffice-7.5.1.2/dbaccess/Module_dbaccess.mk	2023-03-09 20:12:12.233650210 +0100
@@ -35,18 +35,6 @@ $(eval $(call gb_Module_add_targets,dbac
 ))
 
 ifneq ($(OS),iOS)
-ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_hsqlbinary_import) \
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_tdf119625) \
-    $(if $(ENABLE_JAVA),CppunitTest_dbaccess_tdf126268) \
-))
-
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_test \
-))
-endif
-
 ifneq ($(filter SCRIPTING,$(BUILD_TYPE)),)
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
 	CppunitTest_dbaccess_dialog_save \
