--- libreoffice-6.0.0.3/dbaccess/Module_dbaccess.mk~	2018-01-24 21:31:03.000000000 +0100
+++ libreoffice-6.0.0.3/dbaccess/Module_dbaccess.mk	2018-02-02 08:33:38.287577102 +0100
@@ -32,12 +32,6 @@ $(eval $(call gb_Module_add_l10n_targets
 ))
 
 ifneq ($(OS),IOS)
-ifeq ($(ENABLE_FIREBIRD_SDBC),TRUE)
-$(eval $(call gb_Module_add_check_targets,dbaccess,\
-    CppunitTest_dbaccess_firebird_test \
-))
-endif
-
 $(eval $(call gb_Module_add_check_targets,dbaccess,\
 	CppunitTest_dbaccess_dialog_save \
 	CppunitTest_dbaccess_empty_stdlib_save \
--- libreoffice-6.0.0.3/sd/qa/unit/import-tests.cxx~	2018-01-24 21:31:03.000000000 +0100
+++ libreoffice-6.0.0.3/sd/qa/unit/import-tests.cxx	2018-02-02 08:36:16.309008496 +0100
@@ -171,7 +171,6 @@ public:
 
     CPPUNIT_TEST_SUITE(SdImportTest);
 
-    CPPUNIT_TEST(testDocumentLayout);
     CPPUNIT_TEST(testSmoketest);
     CPPUNIT_TEST(testN759180);
     CPPUNIT_TEST(testN778859);
--- libreoffice-6.0.0.3/sw/qa/extras/uiwriter/uiwriter.cxx~	2018-01-24 21:31:03.000000000 +0100
+++ libreoffice-6.0.0.3/sw/qa/extras/uiwriter/uiwriter.cxx	2018-02-02 08:41:59.126211831 +0100
@@ -532,7 +532,6 @@ public:
     CPPUNIT_TEST(testTdf114306_2);
     CPPUNIT_TEST(testTdf113481);
     CPPUNIT_TEST(testTdf115013);
-    CPPUNIT_TEST(testTdf114536);
     CPPUNIT_TEST(testTdf115065);
     CPPUNIT_TEST(testTdf115132);
     CPPUNIT_TEST(testXDrawPagesSupplier);

