--- libreoffice-7.2.0.3/configure.ac~	2021-08-10 19:27:48.000000000 +0200
+++ libreoffice-7.2.0.3/configure.ac	2022-03-29 13:09:30.740802677 +0200
@@ -12759,7 +12759,8 @@
         kf5_libdirs="$kf5_libdirs /usr/lib64 /usr/lib64/kf5 /usr/lib64/kf5/devel"
     fi
 
-    kf5_test_include="KF5/kcoreaddons_version.h"
+    kf5_test_include_old="KF5/kcoreaddons_version.h"
+    kf5_test_include="KF5/KCoreAddons/kcoreaddons_version.h"
     kf5_test_library="libKF5CoreAddons.so"
     kf5_libdirs="$qt5_libdir $kf5_libdirs"
 
@@ -12774,6 +12775,10 @@
     AC_MSG_CHECKING([for KF5 headers])
     kf5_incdir="no"
     for kf5_check in $kf5_incdirs; do
+        if test -r "$kf5_check/$kf5_test_include_old"; then
+            kf5_incdir="$kf5_check/KF5"
+            break
+        fi
         if test -r "$kf5_check/$kf5_test_include"; then
             kf5_incdir="$kf5_check/KF5"
             break
