--- libreoffice-4.1.0.4/configure.ac~	2013-07-26 08:27:33.000000000 +0200
+++ libreoffice-4.1.0.4/configure.ac	2013-07-26 13:22:18.812129639 +0200
@@ -11308,6 +11308,8 @@
     AC_MSG_ERROR([neither wget nor curl found!])
 fi
 
+fi
+
 for i in md5 md5sum /usr/local/bin/md5sum gmd5sum /usr/sfw/bin/md5sum /opt/sfw/bin/gmd5sum /opt/local/bin/md5sum; do
     if test "$i" = "md5"; then
         eval "$i -x" > /dev/null 2>&1
@@ -11333,8 +11335,6 @@
     AC_MSG_ERROR([no md5sum: found!])
 fi
 
-fi
-
 AC_SUBST(WGET)
 AC_SUBST(CURL)
 AC_SUBST(MD5SUM)
