--- libreoffice-4.3.2.2/configure.ac.orig	2014-10-12 14:42:16.760723674 +0200
+++ libreoffice-4.3.2.2/configure.ac	2014-10-12 14:59:04.195877183 +0200
@@ -12315,6 +12315,10 @@
     if test "$_os" = "WINNT"; then
         OOO_JUNIT_JAR=`cygpath -m "$OOO_JUNIT_JAR"`
     fi
+    # hamcrest is a dependency of junit, add it to the classpath if found
+    if test -e /usr/share/java/hamcrest-core.jar; then
+        OOO_JUNIT_JAR=$OOO_JUNIT_JAR:/usr/share/java/hamcrest-core.jar
+    fi
     printf 'import org.junit.Before; import org.hamcrest.BaseDescription;' \
         > conftest.java
     if "$JAVACOMPILER" -classpath "$OOO_JUNIT_JAR" conftest.java >&5 2>&5; then
