--- extensions/source/nsplugin/source/npshell.cxx.old	2012-02-29 00:13:19.000000000 +0000
+++ extensions/source/nsplugin/source/npshell.cxx	2012-02-29 00:18:31.000000000 +0000
@@ -361,7 +361,7 @@
 MIMETYPE_OASIS_OPENDOCUMENT_PRESENTATION_TEMPLATE_ASCII ":otp:OpenDocument Presentation Template;"
 MIMETYPE_OASIS_OPENDOCUMENT_FORMULA_ASCII ":odf:OpenDocument Formula;" );
 
-char*
+const char*
 NPP_GetMIMEDescription(void)
 {
     debug_fprintf(NSP_LOG_APPEND, "print by Netscape Plugin,  NPP_GetMIMEDescription:%s.\n", pMimeTypes);
