--- configure.ac.orig	2012-02-01 13:56:49.000000000 +0100
+++ configure.ac	2012-02-23 18:57:32.000000000 +0100
@@ -7561,6 +7561,11 @@
                     [AC_MSG_ERROR(commons-logging.jar not found.)], [])
             fi
         fi
+        # hamcrest is a dependency of junit, add it to the classpath if
+        # found
+        if test -e /usr/share/java/hamcrest-core.jar; then
+            OOO_JUNIT_JAR=$OOO_JUNIT_JAR:/usr/share/java/hamcrest-core.jar
+        fi
     else
         AC_MSG_RESULT([internal])
         SYSTEM_APACHE_COMMONS=NO

