Index: source/dialog/selector.cxx
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/selector.cxx,v
retrieving revision 1.17
diff -u -r1.17 selector.cxx
--- svx/source/dialog/selector.cxx	8 Sep 2005 22:01:09 -0000	1.17
+++ svx/source/dialog/selector.cxx	31 Jul 2006 10:03:28 -0000
@@ -586,6 +586,8 @@
 							node is a first level child of the Root and is NOT
 							either the current document, user or share */
 							Reference< browse::XBrowseNode >& theChild = children[n];
+							if (!theChild.is())
+								continue;
                             ::rtl::OUString uiName = theChild->getName();
 							BOOL bDisplay = TRUE;
 
@@ -905,6 +907,8 @@
 
 					for ( ULONG n = 0; n < children.getLength(); n++ )
 					{
+         				if (!children[n].is())
+							continue;
 						if (children[n]->getType() == browse::BrowseNodeTypes::SCRIPT)
 						{
 							OUString uri;
@@ -1061,6 +1065,8 @@
 							node is a first level child of the Root and is NOT
 							either the current document, user or share */
 							Reference< browse::XBrowseNode >& theChild = children[n];
+                            if (!theChild.is())
+                                continue;
                             ::rtl::OUString uiName = theChild->getName();
 							BOOL bDisplay = TRUE;
 
