--- solenv/inc/unxlngppc.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngppc.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
--- solenv/inc/unxlngppc4.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngppc4.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
--- solenv/inc/unxlngi4.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngi4.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
--- solenv/inc/unxlngi6.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngi6.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
--- solenv/inc/unxlngx6.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngx6.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
--- solenv/inc/unxlngs.mk.old	2006-01-20 11:50:50.000000000 +0100
+++ solenv/inc/unxlngs.mk	2006-02-28 15:06:48.000000000 +0100
@@ -36,7 +36,7 @@
 ASM=
 AFLAGS=
 
-SOLAR_JAVA*=TRUE
+SOLAR_JAVA*=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
