diff -urN ooo-build-1.3.8.orig/patches/OOO_1_1_3/apply ooo-build-1.3.8/patches/OOO_1_1_3/apply
--- ooo-build-1.3.8.orig/patches/OOO_1_1_3/apply	2005-01-27 12:14:34.000000000 +0100
+++ ooo-build-1.3.8/patches/OOO_1_1_3/apply	2005-02-09 18:50:22.254698368 +0100
@@ -30,7 +30,7 @@
 NovellBase : LinuxCommon, FilePickers, GnomeVFS, GlobalDesktopFiles, NonArkCommon, Evo2, NoRpath, NovellOnly
 SUSE : NovellBase, SUSEOnly
 NLD  : NovellBase, NLDOnly
-PLD  : PLDOnly, LinuxCommon, FilePickers, KDECommon, GlobalDesktopFiles, NonArkCommon, NoRpath
+PLD  : PLDOnly, LinuxCommon, FilePickers, KDECommon, GlobalDesktopFiles, NonArkCommon, NoRpath, Evo2, GnomeVFS
 KDE  : LinuxCommon, FilePickers, KDEOnly, NoRpath
 Ark : ArkOnly, LinuxCommon, FilePickers, GlobalDesktopFiles, Gcc34, NoRpath
 Gentoo : LinuxCommon, FilePickers, GlobalDesktopFiles, NonArkCommon, Gcc34, GentooOnly, NoRpath
diff -urN ooo-build-1.3.8.orig/patches/OOO_1_1_4/apply ooo-build-1.3.8/patches/OOO_1_1_4/apply
--- ooo-build-1.3.8.orig/patches/OOO_1_1_4/apply	2005-01-27 12:10:22.000000000 +0100
+++ ooo-build-1.3.8/patches/OOO_1_1_4/apply	2005-02-09 18:50:41.043841984 +0100
@@ -30,7 +30,7 @@
 NovellBase : LinuxCommon, FilePickers, GnomeVFS, GlobalDesktopFiles, NonArkCommon, Evo2, NoRpath, NovellOnly
 SUSE : NovellBase, SUSEOnly
 NLD  : NovellBase, NLDOnly
-PLD  : PLDOnly, LinuxCommon, FilePickers, KDECommon, GlobalDesktopFiles, NonArkCommon, NoRpath
+PLD  : PLDOnly, LinuxCommon, FilePickers, KDECommon, GlobalDesktopFiles, NonArkCommon, NoRpath, Evo2, GnomeVFS
 KDE  : LinuxCommon, FilePickers, KDEOnly, NoRpath
 Ark : ArkOnly, LinuxCommon, FilePickers, GlobalDesktopFiles, Gcc34, NoRpath, IndicFixes, MSCodec
 Gentoo : LinuxCommon, FilePickers, GlobalDesktopFiles, NonArkCommon, Gcc34, GentooOnly, NoRpath
