--- oo/sw/source/filter/xml/xmltexti.cxx.orig	Sat May 24 14:38:53 2003
+++ oo/sw/source/filter/xml/xmltexti.cxx	Sat May 24 14:39:29 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -827,7 +827,7 @@
 void SwXMLTextImportHelper::RedlineAdjustStartNodeCursor(
 	sal_Bool bStart)
 {
-	OUString& rId = GetOpenRedlineId();
+	const OUString& rId = GetOpenRedlineId();
 	if ((NULL != pRedlineHelper) && (rId.getLength() > 0))
 	{
         Reference<XTextRange> xTextRange( GetCursor()->getStart() ); 
--- oo/sw/source/ui/dbui/dbtree.cxx.orig	Sat May 24 15:32:41 2003
+++ oo/sw/source/ui/dbui/dbtree.cxx	Sat May 24 15:33:01 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -349,8 +349,8 @@
 	const OUString* pDBNames = aDBNames.getConstArray();
 	long nCount = aDBNames.getLength();
 
-    Image& rImg = aImageList.GetImage(IMG_DB);
-    Image& rHCImg = aImageListHC.GetImage(IMG_DB);
+    const Image& rImg = aImageList.GetImage(IMG_DB);
+    const Image& rHCImg = aImageListHC.GetImage(IMG_DB);
     for(long i = 0; i < nCount; i++)
 	{
 		String sDBName(pDBNames[i]);
@@ -499,8 +499,8 @@
                         String sTableName;
                         long nCount = aTblNames.getLength();
                         const OUString* pTblNames = aTblNames.getConstArray();
-                        Image& rImg = aImageList.GetImage(IMG_DBTABLE);
-                        Image& rHCImg = aImageListHC.GetImage(IMG_DBTABLE);
+                        const Image& rImg = aImageList.GetImage(IMG_DBTABLE);
+                        const Image& rHCImg = aImageListHC.GetImage(IMG_DBTABLE);
                         for (long i = 0; i < nCount; i++)
                         {
                             sTableName = pTblNames[i];
@@ -520,8 +520,8 @@
                         String sQueryName;
                         long nCount = aQueryNames.getLength();
                         const OUString* pQueryNames = aQueryNames.getConstArray();
-                        Image& rImg = aImageList.GetImage(IMG_DBQUERY);
-                        Image& rHCImg = aImageListHC.GetImage(IMG_DBQUERY);
+                        const Image& rImg = aImageList.GetImage(IMG_DBQUERY);
+                        const Image& rHCImg = aImageListHC.GetImage(IMG_DBQUERY);
                         for (long i = 0; i < nCount; i++)
                         {
                             sQueryName = pQueryNames[i];
--- oo/sw/source/ui/uno/SwXDocumentSettings.cxx.orig	Sat May 24 16:01:31 2003
+++ oo/sw/source/ui/uno/SwXDocumentSettings.cxx	Sat May 24 16:03:07 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -471,21 +471,21 @@
 		break;
 		case HANDLE_CURRENT_DATABASE_DATA_SOURCE:
 		{
-			SwDBData& rData = mpDoc->GetDBData();
+			const SwDBData& rData = mpDoc->GetDBData();
 			if ( rValue >>= rData.sDataSource )
 				mpDoc->ChgDBData( rData );
 		}
 		break;
 		case HANDLE_CURRENT_DATABASE_COMMAND:
 		{
-			SwDBData& rData = mpDoc->GetDBData();
+			const SwDBData& rData = mpDoc->GetDBData();
 			if ( rValue >>= rData.sCommand )
 				mpDoc->ChgDBData( rData );
 		}
 		break;
 		case HANDLE_CURRENT_DATABASE_COMMAND_TYPE:
 		{
-			SwDBData& rData = mpDoc->GetDBData();
+			const SwDBData& rData = mpDoc->GetDBData();
 			if ( rValue >>= rData.nCommandType )
 				mpDoc->ChgDBData( rData );
 		}
--- oo/sw/source/ui/shells/basesh.cxx.orig	Sat May 24 16:53:00 2003
+++ oo/sw/source/ui/shells/basesh.cxx	Sat May 24 16:53:28 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -752,7 +752,7 @@
 	{
 		case FN_REPAGINATE:
 			{
-				Reference < XModel > & xModel = GetView().GetDocShell()->GetModel();
+				const Reference < XModel > & xModel = GetView().GetDocShell()->GetModel();
 				Reference < XUnoTunnel > xDocTunnel ( xModel, UNO_QUERY );
 				SwXTextDocument *pDoc = reinterpret_cast < SwXTextDocument * > ( xDocTunnel->getSomething ( SwXTextDocument::getUnoTunnelId() ) );
 				pDoc->notifyRefreshListeners();
--- oo/sw/source/ui/fldui/changedb.cxx.orig	Sat May 24 17:13:49 2003
+++ oo/sw/source/ui/fldui/changedb.cxx	Sat May 24 17:14:22 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -258,14 +258,14 @@
 	USHORT nParent = 0;
 	USHORT nChild = 0;
 
-    Image& rTableImg = aImageList.GetImage(IMG_DBTABLE);
-    Image& rDBImg = aImageList.GetImage(IMG_DB);
-    Image& rQueryImg = aImageList.GetImage(IMG_DBQUERY);
-    Image& rHCTableImg = aImageListHC.GetImage(IMG_DBTABLE);
-    Image& rHCDBImg = aImageListHC.GetImage(IMG_DB);
-    Image& rHCQueryImg = aImageListHC.GetImage(IMG_DBQUERY);
-    Image& rToInsert = nCommandType ? rQueryImg : rTableImg;
-    Image& rHCToInsert = nCommandType ? rHCQueryImg : rHCTableImg;
+    const Image& rTableImg = aImageList.GetImage(IMG_DBTABLE);
+    const Image& rDBImg = aImageList.GetImage(IMG_DB);
+    const Image& rQueryImg = aImageList.GetImage(IMG_DBQUERY);
+    const Image& rHCTableImg = aImageListHC.GetImage(IMG_DBTABLE);
+    const Image& rHCDBImg = aImageListHC.GetImage(IMG_DB);
+    const Image& rHCQueryImg = aImageListHC.GetImage(IMG_DBQUERY);
+    const Image& rToInsert = nCommandType ? rQueryImg : rTableImg;
+    const Image& rHCToInsert = nCommandType ? rHCQueryImg : rHCTableImg;
     while ((pParent = aUsedDBTLB.GetEntry(nParent++)) != NULL)
 	{
 		if (sDBName == aUsedDBTLB.GetEntryText(pParent))
