--- vcl/util/makefile.mk.orig	2003-12-16 15:48:03.000000000 +0000
+++ vcl/util/makefile.mk	2003-12-16 15:51:24.000000000 +0000
@@ -294,7 +294,7 @@
 SHL1STDLIBS += -lXp -lXext -lSM -lICE -lX11
 .ELSE
 .IF "$(CPU)" == "I"
-SHL1STDLIBS += -Wl,-Bstatic -lXinerama -Wl,-Bdynamic 
+SHL1STDLIBS += -Wl,-Bdynamic -lXinerama -Wl,-Bdynamic 
 .ENDIF
 SHL1STDLIBS += -lXext -lSM -lICE -lX11
 .ENDIF          # "$(USE_XPRINT)" == "TRUE"
