Index: desktop/source/splash/splash.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/splash/splash.cxx,v
retrieving revision 1.4
diff -u -p -u -r1.4 splash.cxx
--- desktop/source/splash/splash.cxx	15 Jul 2003 14:06:11 -0000	1.4
+++ desktop/source/splash/splash.cxx	1 Oct 2003 12:49:29 -0000
@@ -227,7 +227,7 @@ void SplashScreen::initBitmap()
 			// any language specific information, we have to search for the correct resource
 			// file. The bitmap resource is language independent.
 			const USHORT nResId = RID_DEFAULTINTRO;
-			LanguageType aLanguageType;
+			LanguageType aLanguageType = LANGUAGE_DONTKNOW;
 			String		 aMgrName = String::CreateFromAscii( "iso" );
 			aMgrName += String::CreateFromInt32(SUPD); // current build version
 			ResMgr* pLabelResMgr = ResMgr::SearchCreateResMgr( U2S( aMgrName ), aLanguageType );
Index: desktop/source/app/app.cxx
===================================================================
RCS file: /cvs/framework/desktop/source/app/app.cxx,v
retrieving revision 1.121.10.1
diff -u -p -u -r1.121.10.1 app.cxx
--- desktop/source/app/app.cxx	15 Aug 2003 11:56:47 -0000	1.121.10.1
+++ desktop/source/app/app.cxx	1 Oct 2003 12:49:30 -0000
@@ -1335,7 +1336,7 @@ void Desktop::Main()
 
     // create title string
     sal_Bool    bCheckOk = sal_False;
-    LanguageType aLanguageType;
+    LanguageType aLanguageType = LANGUAGE_DONTKNOW;
     String aMgrName = String::CreateFromAscii( "iso" );
     aMgrName += String::CreateFromInt32(SUPD); // current build version
     ResMgr* pLabelResMgr = ResMgr::SearchCreateResMgr( U2S( aMgrName ), aLanguageType );
