--- ooo-build-2.0.2.orig/bin/build-galleries	2006-03-13 00:17:29.000000000 +0100
+++ ooo-build-2.0.2/bin/build-galleries	2006-03-13 00:21:47.000000000 +0100
@@ -123,7 +123,7 @@
 	if [ ! -f /tmp/.X$XDISPLAY-lock ]; then
 		echo "    trying display :$XDISPLAY ..."
 		sleep 2s
-		$XVFB -ac :$XDISPLAY >& /dev/null &
+		$XVFB -ac :$XDISPLAY &
 		trap "kill $! > /dev/null 2>&1 || true" EXIT
 		# let server to start
 		sleep 10s 
