Index: sfx2/source/dialog/cfg.src
===================================================================
RCS file: /cvs/framework/sfx2/source/dialog/cfg.src,v
retrieving revision 1.32
diff -u -p -u -r1.32 cfg.src
--- sfx2/source/dialog/cfg.src	17 Jul 2003 13:16:18 -0000	1.32
+++ sfx2/source/dialog/cfg.src	6 Oct 2003 13:55:26 -0000
@@ -441,14 +441,28 @@ TabPage TP_CONFIG_MENU
 		Pos = MAP_APPFONT ( 244 , 77 ) ;
 		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TabStop = TRUE ;
-		Symbol = IMAGEBUTTON_ARROW_UP ;
+		ButtonImage = Image
+ 		{
+			ImageBitmap = Bitmap
+			{
+				File = "lc10150.bmp" ;
+			};
+            MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+        };
 	};
 	ImageButton BTN_MN_DOWN
 	{
 		Pos = MAP_APPFONT ( 244 , 94 ) ;
 		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TabStop = TRUE ;
-		Symbol = IMAGEBUTTON_ARROW_DOWN ;
+		ButtonImage = Image
+		{
+			ImageBitmap = Bitmap
+			{
+				File = "lc10151.bmp" ;
+			};
+            MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+        };
 	};
 	PushButton BTN_RESET
 	{
Index: svx/source/dialog/hyphen.src
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/hyphen.src,v
retrieving revision 1.18
diff -u -p -u -r1.18 hyphen.src
--- svx/source/dialog/hyphen.src	17 Jul 2003 11:18:32 -0000	1.18
+++ svx/source/dialog/hyphen.src	6 Oct 2003 13:55:41 -0000
@@ -98,14 +98,28 @@ ModalDialog RID_SVXDLG_HYPHENATE
         Pos = MAP_APPFONT ( 6 , 33 ) ;
 		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TABSTOP = TRUE ;
-		SYMBOL = IMAGEBUTTON_ARROW_LEFT ;
+		ButtonImage = Image
+ 		{
+			ImageBitmap = Bitmap
+			{
+				File = "lc10152.bmp" ;
+			};
+            MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+	    };
 	};
 	ImageButton BTN_RIGHT
 	{
         Pos = MAP_APPFONT ( 22 , 33 ) ;
 		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TABSTOP = TRUE ;
-		Symbol = IMAGEBUTTON_ARROW_RIGHT ;
+		ButtonImage = Image
+ 		{
+			ImageBitmap = Bitmap
+			{
+				File = "lc10153.bmp" ;
+			};
+            MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+	    };
 	};
     FixedText FT_WORD
 	{
Index: sw/source/ui/envelp/envlop.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/envelp/envlop.src,v
retrieving revision 1.32
diff -u -r1.32 envlop.src
--- sw/source/ui/envelp/envlop.src	17 Jul 2003 11:53:20 -0000	1.32
+++ sw/source/ui/envelp/envlop.src	28 Oct 2003 12:10:53 -0000
@@ -413,8 +413,16 @@
 	ImageButton BTN_INSERT
 	{
         Pos = MAP_APPFONT ( 137 , 71 ) ;
-        Size = MAP_APPFONT ( 18 , 12 ) ;
-		SYMBOL = IMAGEBUTTON_ARROW_LEFT ;
+        Size = MAP_APPFONT ( 18 , 14 ) ;
+		ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10152.bmp" ;
+                        };
+                MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
+
 		TabStop = TRUE ;
 	};
 	FixedText FT_DBFIELD
Index: sw/source/ui/frmdlg/olumn.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/frmdlg/column.src,v
retrieving revision 1.48
diff -u -r1.48 column.src
--- sw/source/ui/frmdlg/column.src	17 Jul 2003 11:57:05 -0000	1.48
+++ sw/source/ui/frmdlg/column.src	28 Oct 2003 12:09:57 -0000
@@ -247,9 +247,17 @@
 	{
 		Disable = TRUE ;
         Pos = MAP_APPFONT ( 62 , 69 ) ;
-		Size = MAP_APPFONT ( 10 , 10 ) ;
+		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TabStop = TRUE ;
-		SYMBOL = IMAGEBUTTON_ARROW_LEFT ;
+		ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10152.bmp" ;
+                        };
+                MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
+
 	};
 	FixedText FT_1
 	{
@@ -276,9 +284,17 @@
 	{
 		Disable = TRUE ;
         Pos = MAP_APPFONT ( 238 , 69 ) ;
-		Size = MAP_APPFONT ( 10 , 10 ) ;
+		Size = MAP_APPFONT ( 14 , 14 ) ;
 		TabStop = TRUE ;
-		SYMBOL = IMAGEBUTTON_ARROW_RIGHT ;
+		ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10153.bmp" ;
+                        };
+                MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
+
 	};
 	FixedText FT_WIDTH
 	{
Index: sw/source/ui/index/cnttab.src
===================================================================
RCS file: /cvs/sw/sw/source/ui/index/cnttab.src,v
retrieving revision 1.69
diff -u -r1.69 cnttab.src
--- sw/source/ui/index/cnttab.src	22 Jul 2003 16:51:59 -0000	1.69
+++ sw/source/ui/index/cnttab.src	28 Oct 2003 12:12:30 -0000
@@ -2902,8 +2902,16 @@
 		ImageButton WIN_LEFT_SCROLL
 		{
 			Pos = MAP_APPFONT (  0, 0 ) ;
-			Size = MAP_APPFONT ( 10, 12 ) ;
-			Symbol = IMAGEBUTTON_ARROW_LEFT;
+			Size = MAP_APPFONT ( 14, 14 );
+	                ButtonImage = Image
+        	        {
+                	        ImageBitmap = Bitmap
+                        	{
+	                                File = "lc10152.bmp" ;
+        	                };
+	                	MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+        	        };
+
             DialogControl        = FALSE;
             TabStop              = FALSE;
         };
@@ -2916,8 +2924,16 @@
 		ImageButton WIN_RIGHT_SCROLL
 		{
             Pos = MAP_APPFONT (  126, 0 ) ;
-			Size = MAP_APPFONT ( 10, 12 ) ;
-			Symbol = IMAGEBUTTON_ARROW_RIGHT;
+			Size = MAP_APPFONT ( 14, 14 ) ;
+	                ButtonImage = Image
+        	        {
+                	        ImageBitmap = Bitmap
+                        	{
+                                	File = "lc10153.bmp" ;
+	                        };
+        		        MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+	                };
+
             DialogControl        = FALSE;
             TabStop              = FALSE;
         };
Index: svx/source/dialog/textanim.src
===================================================================
RCS file: /cvs/graphics/svx/source/dialog/textanim.src,v
retrieving revision 1.30
diff -u -r1.30 textanim.src
--- svx/source/dialog/textanim.src	17 Jul 2003 11:24:13 -0000	1.30
+++ svx/source/dialog/textanim.src	30 Oct 2003 13:36:59 -0000
@@ -469,7 +469,15 @@
         Pos = MAP_APPFONT ( MA_TOTAL_WIDTH - MA_MOVEBUT_GRACE_X - 2*MA_BUTTON_WIDTH - RSC_SP_TBPG_INNERBORDER_RIGHT - RSC_SP_FLGR_INNERBORDER_RIGHT,
 							RSC_SP_TBPG_INNERBORDER_TOP + RSC_CD_FIXEDLINE_HEIGHT + RSC_SP_FLGR_INNERBORDER_TOP );
 		Size = BUTTON_SIZE ;
-		Symbol = IMAGEBUTTON_ARROW_UP ;
+	        ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+        		        File = "lc10150.bmp" ;
+                        };
+		        MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
+
 		QuickHelpText = "Nach Oben" ;
 		QuickHelpText [ ENGLISH ] = "To top" ;
 		QuickHelpText [ english_us ] = "To Top" ;
@@ -505,7 +513,14 @@
         Pos = MAP_APPFONT ( MA_TOTAL_WIDTH - MA_MOVEBUT_GRACE_X - 3*MA_BUTTON_WIDTH - RSC_SP_TBPG_INNERBORDER_RIGHT - RSC_SP_FLGR_INNERBORDER_RIGHT,
 							RSC_SP_TBPG_INNERBORDER_TOP + RSC_CD_FIXEDLINE_HEIGHT + RSC_SP_FLGR_INNERBORDER_TOP + MA_BUTTON_HEIGHT);
 		Size = BUTTON_SIZE ;
-		Symbol = IMAGEBUTTON_ARROW_LEFT ;
+                ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10152.bmp" ;
+                        };
+                        MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
 		QuickHelpText = "Nach Links" ;
 		QuickHelpText [ ENGLISH ] = "To left" ;
 		QuickHelpText [ english_us ] = "To Left" ;
@@ -541,7 +556,14 @@
         Pos = MAP_APPFONT ( MA_TOTAL_WIDTH - MA_MOVEBUT_GRACE_X - MA_BUTTON_WIDTH - RSC_SP_TBPG_INNERBORDER_RIGHT - RSC_SP_FLGR_INNERBORDER_RIGHT,
 							RSC_SP_TBPG_INNERBORDER_TOP + RSC_CD_FIXEDLINE_HEIGHT + RSC_SP_FLGR_INNERBORDER_TOP + MA_BUTTON_HEIGHT ) ;
 		Size = BUTTON_SIZE ;
-		Symbol = IMAGEBUTTON_ARROW_RIGHT ;
+                ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10153.bmp" ;
+                        };
+                        MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
 		QuickHelpText = "Nach Rechts" ;
 		QuickHelpText [ ENGLISH ] = "To left" ;
 		QuickHelpText [ english_us ] = "To Right" ;
@@ -577,7 +599,14 @@
         Pos = MAP_APPFONT ( MA_TOTAL_WIDTH - MA_MOVEBUT_GRACE_X - 2*MA_BUTTON_WIDTH - RSC_SP_TBPG_INNERBORDER_RIGHT - RSC_SP_FLGR_INNERBORDER_RIGHT,
 							RSC_SP_TBPG_INNERBORDER_TOP + RSC_CD_FIXEDLINE_HEIGHT + RSC_SP_FLGR_INNERBORDER_TOP + 2*MA_BUTTON_HEIGHT ) ;
 		Size = BUTTON_SIZE ;
-		Symbol = IMAGEBUTTON_ARROW_DOWN ;
+                ButtonImage = Image
+                {
+                        ImageBitmap = Bitmap
+                        {
+                                File = "lc10151.bmp" ;
+                        };
+                        MaskColor = Color { Red = 0xFFFF ; Green = 0x0000 ; Blue = 0xFFFF ; };
+                };
 		QuickHelpText = "Nach Unten" ;
 		QuickHelpText [ ENGLISH ] = "To bottom" ;
 		QuickHelpText [ english_us ] = "To Bottom" ;
