--- oo_1.0_src/config_office/configure.in~~	Thu May  2 16:48:20 2002
+++ oo_1.0_src/config_office/configure.in	Thu May  2 16:50:49 2002
@@ -998,10 +998,10 @@
    CFLAGS=$X_CFLAGS
    LDFLAGS="$X_LDFLAGS $X_LIBS"
    AC_CHECK_LIB(X11, XOpenDisplay, x_libs="-lX11 $X_EXTRA_LIBS", [AC_MSG_ERROR([X Development libraries not found])])
-else
-   x_includes="no_x_includes"
-   x_libraries="no_x_libraries"
-fi
+dnl else
+dnl   x_includes="no_x_includes"
+dnl   x_libraries="no_x_libraries"
+dnl fi
 dnl ***************************************
 dnl testing libc version for Linux...
 dnl ***************************************
