--- oo_641c_src/solenv/inc/_tg_zip.mk.zipdep	Fri Nov  2 18:59:44 2001
+++ oo_641c_src/solenv/inc/_tg_zip.mk	Sat Feb 16 16:59:57 2002
@@ -163,7 +163,8 @@
 	+-zipdep $(ZIP2FLAGS) $@ $(foreach,j,$(ZIP2LIST) $(subst,LANGDIR,$(longlang_{$(subst,$(BIN)$/$(ZIP2TARGET), $(@:db))}) $j )) >> $(MISC)$/$(TARGET).$(PWD:f).$(@:b).dpzz
 .ENDIF			# "$(ZIP2DIR)" != ""
 .ENDIF			# "$(common_build_zip)"!=""
-	@+echo $@ : makefile.mk >> $(MISC)$/$(TARGET).$(PWD:f).$(@:b).dpzz
+# (gb) workaround for setup2/script/makefile.mk
+#	@+echo $@ : makefile.mk >> $(MISC)$/$(TARGET).$(PWD:f).$(@:b).dpzz
 .ENDIF			# "$(make_zip_deps)" != ""
 .ENDIF
 
--- oo_641c_src/tools/bootstrp/makefile.mk.zipdep	Tue Oct 16 16:21:13 2001
+++ oo_641c_src/tools/bootstrp/makefile.mk	Fri Feb 15 17:02:24 2002
@@ -147,7 +147,7 @@
 APP4LIBS=       $(LB)$/bootstrp.lib
 APP4DEPN=   $(LB)$/atools.lib $(LB)$/bootstrp.lib
 
-#APP5TARGET= zipdep
+APP5TARGET= zipdep
 APP5OBJS=	$(OBJ)$/zipdep.obj
 APP5LIBS=	$(LB)$/bootstrp.lib
 APP5STDLIBS=$(STATIC_LIBS) 
