--- oo_1.0_src/freetype/prj/d.lst.systemft	Fri May 31 17:48:49 2002
+++ oo_1.0_src/freetype/prj/d.lst	Fri May 31 17:51:22 2002
@@ -2,14 +2,13 @@
 mkdir: %_DEST%\inc%_EXT%\freetype\config
 mkdir: %_DEST%\inc%_EXT%\freetype\internal
 
-..\%__SRC%\inc\ft2build.h %_DEST%\inc%_EXT%\ft2build.h
-..\%__SRC%\inc\*.h %_DEST%\inc%_EXT%\freetype\*.h
-..\%__SRC%\inc\config\*.h %_DEST%\inc%_EXT%\freetype\config\*.h
-..\%__SRC%\inc\internal\*.h %_DEST%\inc%_EXT%\freetype\internal\*.h
+\usr\include\ft2build.h %_DEST%\inc%_EXT%\ft2build.h
+\usr\include\freetype2\freetype\*.h %_DEST%\inc%_EXT%\freetype\*.h
+\usr\include\freetype2\freetype\config\*.h %_DEST%\inc%_EXT%\freetype\config\*.h
+\usr\include\freetype2\freetype\internal\*.h %_DEST%\inc%_EXT%\freetype\internal\*.h
 
-..\%__SRC%\lib\*.so.* %_DEST%\lib%_EXT%\*.so.*
+\usr\lib\libfreetype.so.* %_DEST%\lib%_EXT%\libfreetype.so.*
 ..\%__SRC%\lib\*.dylib %_DEST%\lib%_EXT%\*.dylib
-..\%__SRC%\lib\*.a %_DEST%\lib%_EXT%\*.a
 
 ..\%__SRC%\lib\freetype.lib %_DEST%\lib%_EXT%\freetype.lib
 
--- oo_1.0_src/freetype/makefile.mk.systemft	Tue Feb 26 12:21:37 2002
+++ oo_1.0_src/freetype/makefile.mk	Fri May 31 17:48:37 2002
@@ -65,6 +65,17 @@
 PRJNAME=so_freetype
 TARGET=so_freetype
 
+# --- Hack to get system freetype ----------------------------------
+.IF "$(GUI)"=="UNX"
+all:
+	@echo "         An already available installation of freetype should exist on your system."
+	@echo "         Therefore the version provided here does not need to be built in addition."
+
+	+$(COPY) /usr$/lib$/libfreetype.so.6* $(DLLDEST)
+	+$(COPY) -r /usr$/include$/freetype2$/freetype $(DLLDEST)$/..$/inc
+	+$(COPY) -r /usr$/include$/ft2build.h $(DLLDEST)$/..$/inc$/freetype
+.ENDIF
+
 # --- Settings -----------------------------------------------------
 
 .INCLUDE :	settings.mk
@@ -103,12 +103,10 @@
 BUILD_DIR=$(CONFIGURE_DIR)
 BUILD_ACTION=$(GNUMAKE)
 
-OUT2LIB= \
-	objs$/.libs$/libfreetype.a
 .IF "$(OS)"=="MACOSX"
-OUT2LIB+=objs$/.libs$/libfreetype.*.dylib
+OUT2LIB=objs$/.libs$/libfreetype.*.dylib
 .ELSE
-OUT2LIB+=objs$/.libs$/libfreetype.so.6*
+OUT2LIB=objs$/.libs$/libfreetype.so.6*
 .ENDIF          # "$(OS)"=="MACOSX"
 
 .ELSE			# "$(GUI)"=="UNX"
--- oo_1.0.2_src.orig/solenv/inc/libs.mk	Tue Jan  7 15:22:02 2003
+++ oo_1.0.2_src/solenv/inc/libs.mk	Thu Feb 13 17:29:26 2003
@@ -131,7 +131,7 @@
 VCLLIBST=-lvcl
 VCLLIB=$(SVLIB)
 FREETYPELIB=-lfreetype
-FREETYPELIBST=$(STATIC) -lfreetype $(DYNAMIC)
+FREETYPELIBST=$(DYNAMIC) -lfreetype $(DYNAMIC)
 XPLIB=-l_xp
 TKLIB=-ltk$(OFFICEUPD)$(DLLPOSTFIX)
 SVTOOLLIB=-lsvt$(OFFICEUPD)$(DLLPOSTFIX)

