Index: bin/setup.in
===================================================================
RCS file: /cvs/gnome/ooo-build/bin/setup.in,v
retrieving revision 1.66
retrieving revision 1.68
diff -u -u -r1.66 -r1.68
--- openoffice/bin/setup.in	3 Jul 2004 02:14:36 -0000	1.66
+++ openoffice/bin/setup.in	7 Jul 2004 17:26:23 -0000	1.68
@@ -72,13 +72,19 @@
 
 else
     if test "z$CONFIGURE_OPTIONS" = "z"; then
-	CONFIGURE_OPTIONS="--with-lang=$LANGS \
-		       --enable-libart --enable-libsn --enable-crashdump=no \
-		       --without-fonts --disable-rpath \
-		       --with-system-zlib --enable-fontconfig \
-		       --with-system-freetype --disable-mozilla \
-		       --with-widget-sets=original,kde,gtk";
+	if [ -s ./setup.configure ]; then
+		. ./setup.configure
+		echo "Using configure options from the ./setup.configure file";
+	else
+		CONFIGURE_OPTIONS="--with-lang=$LANGS \
+			       --enable-libart --enable-libsn --enable-crashdump=no \
+			       --without-fonts --disable-rpath \
+			       --with-system-zlib --enable-fontconfig \
+			       --with-system-freetype --disable-mozilla \
+			       --with-widget-sets=original,kde,gtk";
+	fi
     else
+	echo "CONFIGURE_OPTIONS=\"`echo \"$CONFIGURE_OPTIONS\" | xargs`\"" > ./setup.configure
 	echo "Using configure options from the environment";
     fi
 fi
diff -u -u -r1.30 -r1.31
--- openoffice/bin/package-ooo	6 Jul 2004 13:37:53 -0000	1.30
+++ openoffice/bin/package-ooo	7 Jul 2004 23:24:25 -0000	1.31
@@ -155,7 +155,7 @@
 echo "Done";
 
 # install vcl module for more widget sets if needed
-if test "${OOO_WIDGET_SETS/*,*/comma}" = "comma" ; then
+if (echo "${OOO_WIDGET_SETS}" | grep -q ",") ; then
     # Must remove the ./instsetoo/$ARCHITECTURE/*/normal/install scripts
     # as they conflict with the normal system install binary
     rm -f $OOBUILDDIR/instsetoo/$ARCHITECTURE/*/normal/install
