diff -uNr ooold/autodoc/inc/ary_i/uidl/speci_ce.hxx oonew/autodoc/inc/ary_i/uidl/speci_ce.hxx
--- ooold/autodoc/inc/ary_i/uidl/speci_ce.hxx	Fri Mar  8 15:45:16 2002
+++ oonew/autodoc/inc/ary_i/uidl/speci_ce.hxx	Fri Aug  2 12:54:25 2002
@@ -65,6 +65,7 @@
 
 // USED SERVICES
 	// BASE CLASSES
+#include <ary_i/uidl/cenamesp.hxx>
 #include <ary_i/ce2.hxx>
 	// COMPONENTS
 	// PARAMETERS
diff -uNr ooold/autodoc/prj/build.lst oonew/autodoc/prj/build.lst
--- ooold/autodoc/prj/build.lst	Tue May 14 11:02:10 2002
+++ oonew/autodoc/prj/build.lst	Thu Aug  1 18:09:58 2002
@@ -23,7 +23,7 @@
 ar	autodoc\source\inc\display  		get		-	all	ar_si_display NULL
 ar	autodoc\source\inc\oodb       		get		-	all	ar_si_oodb2 NULL
 ar	autodoc\source\inc\tools     		get		-	all	ar_si_tools NULL
-ar	autodoc\source\inc\utility     		get		-	all	ar_si_util2 NULL
+#ar	autodoc\source\inc\utility     		get		-	all	ar_si_util2 NULL
 ar	autodoc\source\ary					get		-	all	ar_sr NULL
 ar	autodoc\source\ary\inc				get		-	all	ar_sri NULL
 ar	autodoc\source\ary\inc\cpp			get		-	all	ar_sri_cpp NULL
diff -uNr ooold/autodoc/source/displa_i/idoc/api_file.cxx oonew/autodoc/source/displa_i/idoc/api_file.cxx
--- ooold/autodoc/source/displa_i/idoc/api_file.cxx	Thu May  2 14:36:16 2002
+++ oonew/autodoc/source/displa_i/idoc/api_file.cxx	Thu Aug  1 17:47:21 2002
@@ -59,6 +59,7 @@
  *
  ************************************************************************/
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "api_file.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/dht_impl.cxx oonew/autodoc/source/displa_i/idoc/dht_impl.cxx
--- ooold/autodoc/source/displa_i/idoc/dht_impl.cxx	Fri Mar  8 15:45:25 2002
+++ oonew/autodoc/source/displa_i/idoc/dht_impl.cxx	Thu Aug  1 17:52:51 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "dht_impl.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_compl.cxx oonew/autodoc/source/displa_i/idoc/di_compl.cxx
--- ooold/autodoc/source/displa_i/idoc/di_compl.cxx	Fri Mar  8 15:45:25 2002
+++ oonew/autodoc/source/displa_i/idoc/di_compl.cxx	Thu Aug  1 17:54:50 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_compl.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_docu.cxx oonew/autodoc/source/displa_i/idoc/di_docu.cxx
--- ooold/autodoc/source/displa_i/idoc/di_docu.cxx	Fri Mar  8 15:45:25 2002
+++ oonew/autodoc/source/displa_i/idoc/di_docu.cxx	Thu Aug  1 17:55:32 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_docu.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_glbix.cxx oonew/autodoc/source/displa_i/idoc/di_glbix.cxx
--- ooold/autodoc/source/displa_i/idoc/di_glbix.cxx	Fri Mar  8 15:45:26 2002
+++ oonew/autodoc/source/displa_i/idoc/di_glbix.cxx	Thu Aug  1 17:55:44 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_glbix.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_head.cxx oonew/autodoc/source/displa_i/idoc/di_head.cxx
--- ooold/autodoc/source/displa_i/idoc/di_head.cxx	Thu May  2 14:36:16 2002
+++ oonew/autodoc/source/displa_i/idoc/di_head.cxx	Thu Aug  1 17:56:03 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_head.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_linkg.cxx oonew/autodoc/source/displa_i/idoc/di_linkg.cxx
--- ooold/autodoc/source/displa_i/idoc/di_linkg.cxx	Fri Mar  8 15:45:26 2002
+++ oonew/autodoc/source/displa_i/idoc/di_linkg.cxx	Thu Aug  1 17:56:17 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_linkg.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_linkl.cxx oonew/autodoc/source/displa_i/idoc/di_linkl.cxx
--- ooold/autodoc/source/displa_i/idoc/di_linkl.cxx	Fri Mar  8 15:45:26 2002
+++ oonew/autodoc/source/displa_i/idoc/di_linkl.cxx	Thu Aug  1 17:59:20 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_linkl.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_summy.cxx oonew/autodoc/source/displa_i/idoc/di_summy.cxx
--- ooold/autodoc/source/displa_i/idoc/di_summy.cxx	Fri Mar  8 15:45:26 2002
+++ oonew/autodoc/source/displa_i/idoc/di_summy.cxx	Thu Aug  1 17:57:01 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_summy.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_syntx.cxx oonew/autodoc/source/displa_i/idoc/di_syntx.cxx
--- ooold/autodoc/source/displa_i/idoc/di_syntx.cxx	Fri Mar  8 15:45:26 2002
+++ oonew/autodoc/source/displa_i/idoc/di_syntx.cxx	Thu Aug  1 17:57:07 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_syntx.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/di_whole.cxx oonew/autodoc/source/displa_i/idoc/di_whole.cxx
--- ooold/autodoc/source/displa_i/idoc/di_whole.cxx	Thu May  2 14:36:16 2002
+++ oonew/autodoc/source/displa_i/idoc/di_whole.cxx	Thu Aug  1 17:57:14 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "di_whole.hxx"
 
diff -uNr ooold/autodoc/source/displa_i/idoc/didl_def.cxx oonew/autodoc/source/displa_i/idoc/didl_def.cxx
--- ooold/autodoc/source/displa_i/idoc/didl_def.cxx	Fri Mar  8 15:45:25 2002
+++ oonew/autodoc/source/displa_i/idoc/didl_def.cxx	Thu Aug  1 17:57:20 2002
@@ -59,6 +59,7 @@
  *
  ************************************************************************/
 
+#include "/usr/include/stlport/utility"
 #include <precomp.h>
 #include "didl_def.hxx"
 
diff -uNr ooold/autodoc/source/parser_i/idl/cx_idlco.cxx oonew/autodoc/source/parser_i/idl/cx_idlco.cxx
--- ooold/autodoc/source/parser_i/idl/cx_idlco.cxx	Fri Mar  8 15:45:34 2002
+++ oonew/autodoc/source/parser_i/idl/cx_idlco.cxx	Fri Aug  2 12:15:29 2002
@@ -60,6 +60,7 @@
  ************************************************************************/
 
 #include <precomp.h>
+#include <s2_luidl/tk_ident.hxx>
 #include <s2_luidl/cx_idlco.hxx>
 
 
@@ -68,7 +69,6 @@
 #include <s2_dsapi/cx_dsapi.hxx>
 #include <tools/tkpchars.hxx>
 #include <tokens/tkpstam2.hxx>
-#include <s2_luidl/tk_ident.hxx>
 #include <s2_luidl/tk_keyw.hxx>
 #include <s2_luidl/tk_punct.hxx>
 #include <s2_luidl/tokrecv.hxx>
diff -uNr ooold/autodoc/source/parser_i/inc/s2_luidl/cx_idlco.hxx oonew/autodoc/source/parser_i/inc/s2_luidl/cx_idlco.hxx
--- ooold/autodoc/source/parser_i/inc/s2_luidl/cx_idlco.hxx	Fri Mar  8 15:45:35 2002
+++ oonew/autodoc/source/parser_i/inc/s2_luidl/cx_idlco.hxx	Fri Aug  2 12:20:27 2002
@@ -64,9 +64,10 @@
 
 // USED SERVICES
 	// BASE CLASSES
-#include <tokens/tkpcont2.hxx>
+#include <s2_luidl/tk_ident.hxx>
+#include "../tokens/tkpcont2.hxx"
 	// COMPONENTS
-#include <tokens/tkpstam2.hxx>
+#include "../tokens/tkpstam2.hxx"
 	// PARAMETERS
 
 
diff -uNr ooold/autodoc/source/parser_i/inc/s2_luidl/cx_sub.hxx oonew/autodoc/source/parser_i/inc/s2_luidl/cx_sub.hxx
--- ooold/autodoc/source/parser_i/inc/s2_luidl/cx_sub.hxx	Fri Mar  8 15:45:35 2002
+++ oonew/autodoc/source/parser_i/inc/s2_luidl/cx_sub.hxx	Fri Aug  2 12:01:43 2002
@@ -64,7 +64,8 @@
 
 // USED SERVICES
 	// BASE CLASSES
-#include <tokens/tkpcont2.hxx>
+#include "cx_idlco.hxx"
+#include "../tokens/tkpcont2.hxx"
 	// COMPONENTS
 	// PARAMETERS
 
diff -uNr ooold/autodoc/source/parser_i/inc/s2_luidl/parsenv2.hxx oonew/autodoc/source/parser_i/inc/s2_luidl/parsenv2.hxx
--- ooold/autodoc/source/parser_i/inc/s2_luidl/parsenv2.hxx	Fri Mar  8 15:45:35 2002
+++ oonew/autodoc/source/parser_i/inc/s2_luidl/parsenv2.hxx	Fri Aug  2 12:33:48 2002
@@ -66,6 +66,7 @@
 
 // USED SERVICES
 	// BASE CLASSES
+#include <ary_i/codeinf2.hxx>
 #include <s2_luidl/tokproct.hxx>
 	// COMPONENTS
 #include <s2_luidl/semnode.hxx>
