Index: ooo-wrapper.in
===================================================================
RCS file: /cvs/gnome/ooo-build/bin/ooo-wrapper.in,v
retrieving revision 1.30
retrieving revision 1.34
diff -u -u -r1.30 -r1.34
--- ooo-build.org/bin/ooo-wrapper.in	15 Jul 2004 02:24:03 -0000	1.30
+++ ooo-build/bin/ooo-wrapper.in	31 Aug 2004 09:48:10 -0000	1.34
@@ -394,7 +394,7 @@
 
     unlink $SetupConfig;
 
-    check_user_installation() || die "Installation of OpenOffice.org $Version was not successfull\n";
+    check_user_installation() || die "Installation of OpenOffice.org $Version was not successfull.\nCheck ~/.openoffice-install-log log file for details.\n";
 
     rewrite_paths();
 }
@@ -422,10 +422,29 @@
 
 # select the preferred icons set by the running windowmanager
 unless (defined $icons_set) {
-    $icons_set="kde" if (-d "$SystemInstallDir/program/resource.kde" &&
-                         "$ENV{WINDOWMANAGER}" =~ ".*kde\$");
-    $icons_set="gnome" if (-d "$SystemInstallDir/program/resource.gnome" &&
-                         "$ENV{WINDOWMANAGER}" =~ ".*gnome\$");
+    if ((exists $ENV{GNOME_DESKTOP_SESSION_ID}) && (-d "$SystemInstallDir/program/resource.gnome")) {
+        $icons_set="gnome";
+    } elsif ((exists $ENV{KDE_FULL_SESSION}) && (-d "$SystemInstallDir/program/resource.kde")) {
+	$icons_set="kde";
+    }
+
+    if (not defined $icons_set) {
+        if ((exists $ENV{WINDOWMANAGER}) && (-d "$SystemInstallDir/program/resource.kde") &&
+            ("$ENV{WINDOWMANAGER}" =~ ".*kde\$")) {
+                $icons_set="kde";
+        } elsif ((exists $ENV{WINDOWMANAGER}) && (-d "$SystemInstallDir/program/resource.gnome") &&
+            ("$ENV{WINDOWMANAGER}" =~ ".*gnome\$")) {
+                $icons_set="gnome";
+        } elsif (-d "$SystemInstallDir/program/resource.default") {
+            $icons_set="default";
+        } elsif (-d "$SystemInstallDir/program/resource.kde") {
+            $icons_set="kde";
+        } elsif (-d "$SystemInstallDir/program/resource.gnome") {
+            $icons_set="gnome";
+        } else {
+            $icons_set="default";
+        }
+    }
 }
 
 if (defined $icons_set) {
