diff -urN ooo-build-1.1.61.org/patches/OOO_1_1_2/apply ooo-build-1.1.61/patches/OOO_1_1_2/apply
--- ooo-build-1.1.61.org/patches/OOO_1_1_2/apply	2004-07-06 13:21:38.283310176 +0200
+++ ooo-build-1.1.61/patches/OOO_1_1_2/apply	2004-07-06 13:22:31.649197328 +0200
@@ -777,7 +777,6 @@
 #######################################################
 
 [ NativeWidgets ]
-non-existant.diff
 # Native Widget Framework
 nativewidgets-config_office.diff
 nativewidgets-vcl.diff
