Index: vcl/source/control/fixed.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/control/fixed.cxx,v
retrieving revision 1.10
diff -u -p -u -r1.10 fixed.cxx
--- vcl/source/control/fixed.cxx	17 Oct 2002 14:47:00 -0000	1.10
+++ vcl/source/control/fixed.cxx	12 Sep 2003 14:38:28 -0000
@@ -73,6 +73,9 @@
 #ifndef _VCL_CONTROLLAYOUT_HXX
 #include <controllayout.hxx>
 #endif
+#ifndef _SV_BITMAPEX_HXX
+#include <bitmapex.hxx>
+#endif
 
 #include <rc.h>
 
@@ -810,14 +820,16 @@ void FixedBitmap::ImplDraw( OutputDevice
 	else
 	{
 		// Haben wir ueberhaupt eine Bitmap
-		if ( !(!(*pBitmap)) )
+        if ( !!(*pBitmap) )
 		{
+			BitmapEx aEx( *pBitmap );
+
 			if ( GetStyle() & WB_SCALE )
-				pDev->DrawBitmap( rPos, rSize, *pBitmap );
+				pDev->DrawBitmapEx( rPos, rSize, aEx );
 			else
 			{
 				Point aPos = ImplCalcPos( GetStyle(), rPos, pBitmap->GetSizePixel(), rSize );
-				pDev->DrawBitmap( aPos, *pBitmap );
+				pDev->DrawBitmapEx( aPos, aEx );
 			}
 		}
 	}
