--- ./build/oof680-m14/ucb/source/ucp/ftp/makefile.mk.org	2007-09-28 18:42:46.882942137 +0200
+++ ./build/oof680-m14/ucb/source/ucp/ftp/makefile.mk	2007-09-28 18:42:57.171497260 +0200
@@ -48,10 +48,6 @@
 
 .INCLUDE: settings.mk
 
-.IF "$(SYSTEM_CURL)" == "YES"
-CFLAGS+=-DCURL_NO_OLDIES
-.ENDIF
-
 # --- General -----------------------------------------------------
 
 # first target ( shared library )
