Index: vcl/unx/source/gdi/dtint.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/dtint.cxx,v
retrieving revision 1.16
diff -u -p -u -r1.16 dtint.cxx
--- vcl/unx/source/gdi/dtint.cxx	28 May 2003 12:33:47 -0000	1.16
+++ vcl/unx/source/gdi/dtint.cxx	10 Jun 2003 14:17:38 -0000
@@ -229,19 +229,8 @@ Font DtIntegrator::parseFont( const Byte
         xub_StrLen nIndex = nPos2;
         ByteString aToken = rLine.GetToken( 1, ',', nIndex );
         if( aToken.Len() )
-        {
-            int nPixelHeight = aToken.ToInt32();
-            long nDPIX, nDPIY;
-            long nDispDPIY = mpSalDisplay->GetResolution().B();
-            mpSalDisplay->GetScreenFontResolution( nDPIX, nDPIY );
-            int nHeight = nPixelHeight * nDispDPIY / nDPIY;
-            // allow for rounding in back conversion (at SetFont)
-            while( (nHeight * nDPIY / nDispDPIY) > nPixelHeight )
-                nHeight--;
-            while( (nHeight * nDPIY / nDispDPIY) < nPixelHeight )
-                nHeight++;
-            aFont.SetHeight( nHeight );
-        }
+	    aFont.SetHeight( (long) OString(aToken).toDouble() );
+
         while( nIndex != STRING_NOTFOUND )
         {
             aToken = rLine.GetToken( 0, ',', nIndex );
