diff -Nur oo.orig/crashrep/source/unx/makefile.mk oo.chng/crashrep/source/unx/makefile.mk
--- oo.orig/crashrep/source/unx/makefile.mk	Wed Jul 30 19:57:00 2003
+++ oo.chng/crashrep/source/unx/makefile.mk	Thu Jul 31 10:44:25 2003
@@ -115,7 +115,7 @@
 APP2TARGET=$(TARGET2)
 APP2OBJS=$(OBJFILES)
 
-APP2STDLIBS=$(STATIC) `pkg-config --only-mod-libs --libs gtk+-2.0` -lpng -lz -ljpeg -ltiff $(DYNAMIC) -lXext -lX11 -ldl -lnsl
+APP2STDLIBS=$(STATIC) `pkg-config --libs gtk+-2.0` -lexpat `pkg-config --libs gtk+-2.0` -lexpat -lpng -lz -ljpeg -ltiff $(DYNAMIC) -lXext -lX11 -ldl -lnsl
 .IF "$(OS)" == "SOLARIS"
 APP2STDLIBS+=-lsocket
 .ENDIF
