--- ooo-build-2.0.2.7.orig/patches/src680/apply	2006-04-21 15:21:46.640348000 +0200
+++ ooo-build-2.0.2.7/patches/src680/apply	2006-04-21 16:10:55.324630250 +0200
@@ -864,16 +864,6 @@
 # Db 4.3/4.4 needs -lpthread
 system-db-4.3-use-lpthread.diff, i#58474, rengelha
 
-
-[PLDOnly]
-SectionOwner => arekm
-
-# fix build of sal project
-solenv-unxlngi4-link.diff, i#66982
-# fix for macro browser crash
-sfx2-badscript.diff, i#67976
-
-
 [ BuildBits ]
 # add automatical detection of serializer.jar into the configure
 # add it to classpath on one more place
