--- ./patches/src680/apply~	2005-07-28 00:28:15.000000000 +0300
+++ ./patches/src680/apply	2005-07-28 00:29:27.000000000 +0300
@@ -31,7 +31,7 @@
 # Ark Linux
 Ark: LinuxCommon, NotDebian, ArkOnly
 # www.pld-linux.org
-PLDBase: LinuxCommon, NotDebian
+PLDBase: LinuxCommon, NotDebian, PLDOnly
 PLD: PLDBase
 PLD64: PLDBase, 64bit
 # Mandriva Linux
--- ./patches/src680/apply~	2005-07-28 00:28:15.000000000 +0300
+++ ./patches/src680/apply	2005-07-28 00:29:27.000000000 +0300
@@ -1086,3 +1086,6 @@
 vba-basic-globals.diff
 vba-build.diff
 vba-sc-docuno-hook.diff
+
+[ PLDOnly ]
+openoffice.org-STL-lib64.diff
