--- ooo-build-1.3.5/patches/OOO_1_1_3/apply.orig	2004-10-15 06:58:12.000000000 +0200
+++ ooo-build-1.3.5/patches/OOO_1_1_3/apply	2004-11-03 19:43:40.630493584 +0100
@@ -24,7 +24,7 @@
 NovellBase : LinuxCommon, GnomeVFS, GlobalDesktopFiles, NonDebianCommon, Evo2, NovellOnly
 SUSE : NovellBase, SUSEOnly
 NLD  : NovellBase, NLDOnly
-PLD  : PLDOnly, LinuxCommon, KDECommon, GlobalDesktopFiles, NonDebianCommon
+PLD  : PLDOnly, LinuxCommon, KDECommon, GlobalDesktopFiles, NonDebianCommon, Evo2, GnomeVFS
 KDE  : LinuxCommon, KDEOnly
 
 # Odd one, very few patches:
