--- desktop/scripts/soffice.sh.orig	2003-09-22 12:07:01.000000000 -0700
+++ desktop/scripts/soffice.sh	2003-09-22 12:07:47.000000000 -0700
@@ -144,25 +144,27 @@
 esac
 
 # extend the ld_library_path for java: javaldx checks the sofficerc for us
-java_ld_library_path=`"$sd_prog/javaldx"`
-if [ "$java_ld_library_path" != "" ] ; then
-    case $sd_platform in
-        AIX)
-            LIBPATH=${java_ld_library_path}:${LIBPATH}
-            ;;
-        Darwin)
-            DYLD_LIBRARY_PATH=${java_ld_library_path}:${DYLD_LIBRARY_PATH}
-            ;;
-        HP-UX)
-            SHLIB_PATH=${java_ld_library_path}:${SHLIB_PATH}
-            ;;
-        IRIX*)
-            LD_LIBRARYN32_PATH=${java_ld_library_path}:${LD_LIBRARYN32_PATH}
-            ;;
-        *)
-            LD_LIBRARY_PATH=${java_ld_library_path}:${LD_LIBRARY_PATH}
-            ;;
-    esac
+if [ -x "$sd_prog/javaldx" ] ; then
+    java_ld_library_path=`"$sd_prog/javaldx"`
+    if [ "$java_ld_library_path" != "" ] ; then
+        case $sd_platform in
+            AIX)
+                LIBPATH=${java_ld_library_path}:${LIBPATH}
+                ;;
+            Darwin)
+                DYLD_LIBRARY_PATH=${java_ld_library_path}:${DYLD_LIBRARY_PATH}
+                ;;
+            HP-UX)
+                SHLIB_PATH=${java_ld_library_path}:${SHLIB_PATH}
+                ;;
+            IRIX*)
+                LD_LIBRARYN32_PATH=${java_ld_library_path}:${LD_LIBRARYN32_PATH}
+                ;;
+            *)
+                LD_LIBRARY_PATH=${java_ld_library_path}:${LD_LIBRARY_PATH}
+                ;;
+        esac
+    fi
 fi
 
 # set java environment variables
--- desktop/source/javaldx/makefile.mk.orig	2003-09-22 12:07:12.000000000 -0700
+++ desktop/source/javaldx/makefile.mk	2003-09-22 12:11:42.000000000 -0700
@@ -77,6 +77,8 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)"!=""
+
 OBJFILES=$(OBJ)$/javaldx.obj
 
 APP1TARGET=javaldx
@@ -86,6 +88,8 @@
 APP1STDLIBS+= -lstdc++
 .ENDIF
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
