diff -Nur oo/berkeleydb/makefile.mk oo.new/berkeleydb/makefile.mk
--- oo/berkeleydb/makefile.mk	Fri Feb  8 16:18:17 2002
+++ oo.new/berkeleydb/makefile.mk	Sun Apr 14 23:56:12 2002
@@ -117,12 +117,12 @@
 
 .INCLUDE : set_ext.mk
 .INCLUDE :	target.mk
-.INCLUDE :	tg_ext.mk
+#.INCLUDE :	tg_ext.mk
 
-TG_DELIVER : $(INPATH)$/misc$/build$/so_predeliver
-		$(DELIVER)
+#TG_DELIVER : $(INPATH)$/misc$/build$/so_predeliver
+#		$(DELIVER)
 
-.IF "$(BUILD_SOSL)"!=""
-ALLTAR : TG_DELIVER
-.ENDIF			# "$(BUILD_SOSL)"!=""
+#.IF "$(BUILD_SOSL)"!=""
+#ALLTAR : TG_DELIVER
+#.ENDIF			# "$(BUILD_SOSL)"!=""
 
diff -Nur oo/config_office/makefile.rc oo.new/config_office/makefile.rc
--- oo/config_office/makefile.rc	Fri Oct 19 17:25:23 2001
+++ oo.new/config_office/makefile.rc	Sun Apr 14 23:56:12 2002
@@ -155,7 +155,6 @@
 	so3 \
 	soltools \
 	sot \
-	berkeleydb \
 	starmath \
 	stoc \
 	store \
diff -Nur oo/solenv/inc/libs.mk oo.new/solenv/inc/libs.mk
--- oo/solenv/inc/libs.mk	Fri Oct 26 15:14:41 2001
+++ oo.new/solenv/inc/libs.mk	Mon Apr 15 12:10:25 2002
@@ -169,8 +169,8 @@
 ZLIB3RDLIB=-lzlib
 JPEG3RDLIB=-ljpeglib
 NEON3RDLIB=-lneon
-BERKELEYLIB=-ldb-3.2
-BERKELEYCPPLIB=-ldb_cxx-3.2
+BERKELEYLIB=-ldb-3
+BERKELEYCPPLIB=-ldb_cxx-3
 SFX2LIB=-lsfx$(UPD)$(DLLPOSTFIX)
 SFXLIB=-lsfx$(UPD)$(DLLPOSTFIX)
 SFXDEBUGLIB=
diff -Nur oo/xmlhelp/source/cxxhelp/provider/databases.cxx oo.new/xmlhelp/source/cxxhelp/provider/databases.cxx
--- oo/xmlhelp/source/cxxhelp/provider/databases.cxx	Fri Nov 23 17:51:01 2001
+++ oo.new/xmlhelp/source/cxxhelp/provider/databases.cxx	Mon Apr 15 10:59:50 2002
@@ -60,7 +60,7 @@
  ************************************************************************/
 
 
-#include <berkeleydb/db_cxx.h>
+#include <db_cxx.h>
 #ifndef _VOS_DIAGNOSE_HXX_
 #include <vos/diagnose.hxx>
 #endif
diff -Nur oo/xmlhelp/source/cxxhelp/provider/urlparameter.cxx oo.new/xmlhelp/source/cxxhelp/provider/urlparameter.cxx
--- oo/xmlhelp/source/cxxhelp/provider/urlparameter.cxx	Wed Oct 31 14:08:14 2001
+++ oo.new/xmlhelp/source/cxxhelp/provider/urlparameter.cxx	Mon Apr 15 11:05:50 2002
@@ -88,7 +88,7 @@
 #include <sablot/shandler.h>
 #endif
 #ifndef _DB_CXX_H_
-#include <berkeleydb/db_cxx.h>
+#include <db_cxx.h>
 #endif
 #ifndef _URLPARAMETER_HXX_
 #include <provider/urlparameter.hxx>
