--- ooo-build-1.3.0/bin/install-help.orig	2004-07-14 10:22:38.000000000 +0200
+++ ooo-build-1.3.0/bin/install-help	2004-07-24 16:09:52.000000000 +0200
@@ -7,24 +7,27 @@
     exit 1;
 fi
 
-# the available extra localizations
-LANGNUMS="07 33 34 36 39 42 43 46 49 81 82 86 88"
+# the tested localizations
+LANGNUMS=`./openoffice-xlate-lang -p all`
 
 for LANGNUM in $LANGNUMS ; do
   LANGCODE=`$TOOLSDIR/bin/openoffice-xlate-lang -i $LANGNUM`
   install -m 755 -d $HELPDIR/$LANGCODE || exit 1;
-  if test -f $SRCDIR/helpcontent_${LANGNUM}_unix.tar.bz2 ; then
+  if test -f $SRCDIR/helpcontent_${LANGNUM}_unix.tar.bz2 -o -f $SRCDIR/helpcontent_${LANGNUM}_unix.tgz ; then
     echo "Unpacking $LANGCODE help content..."
-    tar -xjf $SRCDIR/helpcontent_${LANGNUM}_unix.tar.bz2 -C $HELPDIR/$LANGCODE || exit 1;
-  
-    pushd $HELPDIR/$LANGCODE
+    if test -f $SRCDIR/helpcontent_${LANGNUM}_unix.tar.bz2 ; then
+      tar -xjf $SRCDIR/helpcontent_${LANGNUM}_unix.tar.bz2 -C $HELPDIR/$LANGCODE || exit 1;
+    else
+      tar -xzf $SRCDIR/helpcontent_${LANGNUM}_unix.tgz -C $HELPDIR/$LANGCODE || exit 1;
+    fi
+    cd $HELPDIR/$LANGCODE
     for arch in *.zip ; do
       if test "z$arch" != "zhelpxsl.zip" ; then
         unzip -o $arch
       fi
       rm $arch
     done
-    popd
+    cd -
 
     # fix or add highcontrast*.css
     for config in custom.css default.css \
--- ooo-build-1.3.0/bin/package-ooo.orig	2004-07-21 20:30:10.000000000 +0200
+++ ooo-build-1.3.0/bin/package-ooo	2004-07-24 16:13:24.000000000 +0200
@@ -49,7 +49,7 @@
 	chmod +x $PREFIX/bin/ooo-wrapper$BINSUFFIX
 	for app in calc draw fromtemplate impress html math web writer ffice; do
 		ln -sf ooo-wrapper${BINSUFFIX} $PREFIX/bin/oo${app}${BINSUFFIX}
-		if test "z$VENDORNAME" = "zNovell" -o "z$VENDORNAME" = "zSUSE" ; then
+		if test "z$VENDORNAME" = "zNovell" -o "z$VENDORNAME" = "zSUSE" -o "z$VENDORNAME" = "zPLD"; then
 			echo ".so man1/openoffice$BINSUFFIX.1" >| $MANDIR/man1/oo${app}$BINSUFFIX.1;
 		fi
 	done
