--- vcl/source/window/menu-checkbox-off.xpm	Thu Apr 11 09:25:15 2002
+++ vcl/source/window/menu-checkbox-off.xpm	Thu Apr 10 19:47:09 2003
@@ -0,0 +1,12 @@
+/* XPM */
+static char * menu_checkbox_off_xpm[] = {
+"8 8 1 1",
+" 	c None",
+"        ",
+"        ",
+"        ",
+"        ",
+"        ",
+"        ",
+"        ",
+"        "};
--- vcl/source/window/menu-checkbox-on.xpm	Thu Apr 11 09:25:15 2002
+++ vcl/source/window/menu-checkbox-on.xpm	Thu Apr 10 19:47:09 2003
@@ -0,0 +1,16 @@
+/* XPM */
+static char * menu_checkbox_on_xpm[] = {
+"8 8 5 1",
+" 	c None",
+"+	c #DEDEDE",
+"@	c #000000",
+"#	c #989898",
+"$	c #C9C9C9",
+"        ",
+"        ",
+"     +@#",
+"+@+ +@@+",
+"@@@$@@+ ",
+"+@@@@+  ",
+" +@@+   ",
+"        "};
--- vcl/source/window/menu-radio-off.xpm	Thu Apr 11 09:25:15 2002
+++ vcl/source/window/menu-radio-off.xpm	Thu Apr 10 19:47:09 2003
@@ -0,0 +1,18 @@
+/* XPM */
+static char * menu_radio_off_xpm[] = {
+"8 8 7 1",
+" 	c None",
+".	c #FDFDFD",
+"+	c #CDCDCD",
+"@	c #4F4F4F",
+"#	c #0A0A0A",
+"$	c #D3D3D3",
+"%	c #FEFEFE",
+" +@##@+ ",
+"+#$%%$#+",
+"@$....$@",
+"#%....%#",
+"#%....%#",
+"@$....$@",
+"+#$%%$#+",
+" +@##@+ "};
--- vcl/source/window/menu-radio-on.xpm	Thu Apr 11 09:25:15 2002
+++ vcl/source/window/menu-radio-on.xpm	Thu Apr 10 19:47:09 2003
@@ -0,0 +1,20 @@
+/* XPM */
+static char * menu_radio_on_xpm[] = {
+"8 8 9 1",
+" 	c None",
+"+	c #CDCDCD",
+"@	c #4F4F4F",
+"#	c #0A0A0A",
+"$	c #D3D3D3",
+"%	c #FEFEFE",
+"&	c #9F9F9F",
+"*	c #151515",
+"=	c #000000",
+" +@##@+ ",
+"+#$%%$#+",
+"@$&**&$@",
+"#%*==*%#",
+"#%*==*%#",
+"@$&**&$@",
+"+#$%%$#+",
+" +@##@+ "};
