diff -urbB ooo-build-2.0.0.org/patches/OOO_2_0/apply ooo-build-2.0.0/patches/OOO_2_0/apply
--- ooo-build-2.0.0.org/patches/OOO_2_0/apply	2005-10-24 21:12:15.000000000 +0000
+++ ooo-build-2.0.0/patches/OOO_2_0/apply	2005-10-26 18:00:40.136587088 +0000
@@ -38,7 +38,7 @@
 # Ark Linux
 Ark: LinuxCommon, NotDebian, ArkOnly
 # www.pld-linux.org
-PLDBase: LinuxCommon, NotDebian
+PLDBase: LinuxCommon, NotDebian, PLDOnly
 PLD: PLDBase
 PLD64: PLDBase, 64bit
 # Mandriva Linux
@@ -371,6 +371,9 @@
 ooo-vendor-debian.diff
 debian-splash.diff
 
+[ PLDOnly ]
+ooo-longnames-54730.diff
+
 [ MandrivaOnly ]
 mandriva-splash.diff
 default-system-fpicker.diff
