Index: sw/source/ui/inc/swmn.hrc
===================================================================
RCS file: /cvs/sw/sw/source/ui/inc/swmn.hrc,v
retrieving revision 1.133.34.1
diff -u -r1.133.34.1 swmn.hrc
--- sw/source/ui/inc/swmn.hrc	25 Aug 2003 13:34:46 -0000	1.133.34.1
+++ sw/source/ui/inc/swmn.hrc	14 Nov 2003 11:13:24 -0000
@@ -5207,6 +5207,8 @@
 						{
                             Identifier = FN_EDIT_FORMULA ;
                             HelpID = FN_EDIT_FORMULA ;
+			    _MenuItemFlags = MIB_CHECKABLE;
+			    Checkable = TRUE; 
                             Text = "~Rechenleiste" ;
                             Text [ english_us ] = "Fo~rmula Bar" ;
 							Text[ portuguese ] = "Fo~rmularbar";
@@ -5390,6 +5392,8 @@
 			{
 				Identifier = FN_VIEW_MARKS ;
 				HelpID = FN_VIEW_MARKS ;
+				_MenuItemFlags = MIB_CHECKABLE;
+				Checkable = TRUE;
 				Text = "Ma~rkierungen" ;
 				Text [ English ] = "~Marks" ;
 				Text [ norwegian ] = "~Merker" ;
@@ -5427,6 +5431,8 @@
 			{
 				Identifier = FN_VIEW_FIELDNAME ;
 				HelpID = FN_VIEW_FIELDNAME ;
+				_MenuItemFlags = MIB_CHECKABLE;
+				Checkable = TRUE;
 				Text = "~Feldbefehle" ;
 				Text [ English ] = "~Fields" ;
 				Text [ norwegian ] = "~Felter" ;
@@ -5464,6 +5470,8 @@
 			{
 				Identifier = FN_VIEW_META_CHARS ;
 				HelpID = FN_VIEW_META_CHARS ;
+				_MenuItemFlags = MIB_CHECKABLE;
+				Checkable = TRUE;
 				Text = "Steuer~zeichen" ;
 				Text [ English ] = "~Control codes" ;
 				Text [ norwegian ] = "~Kontroller koder" ;
@@ -5545,6 +5553,8 @@
 			{
 				Identifier = FN_PRINT_LAYOUT ;
 				HelpID = FN_PRINT_LAYOUT ;
+			    	_MenuItemFlags = MIB_CHECKABLE;
+			    	Checkable = TRUE;
 				Text = "~Druck - Layout" ;
 				Text [ English ] = "~Print layout" ;
 				Text [ portuguese ] = "~Configuração da impressão" ;
@@ -7574,6 +7584,8 @@
 						{
 							Identifier = SID_AUTOSPELL_CHECK ;
 							HelpID = SID_AUTOSPELL_CHECK ;
+							_MenuItemFlags = MIB_CHECKABLE;
+							Checkable = TRUE;
 							/* ### ACHTUNG: Neuer Text in Resource? ~Automatisch prüfen : ~Automatisch pr³fen */
 							Text = "~Automatisch prüfen" ;
 							Text [ English ] = "~Online spelling" ;
Index: sd/source/ui/app/menuids.src
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/menuids.src,v
retrieving revision 1.49
diff -u -r1.49 menuids.src
--- sd/source/ui/app/menuids.src	17 Jul 2003 12:39:55 -0000	1.49
+++ sd/source/ui/app/menuids.src	14 Nov 2003 11:13:25 -0000
@@ -3739,6 +3739,8 @@
     {\
         Identifier = SID_GRID_VISIBLE ; \
         HelpID = SID_GRID_VISIBLE ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "~Raster sichtbar" ; \
         Text [ ENGLISH ] = "Grid visible" ; \
         Text [ dutch ] = "Raster ~zichtbaar" ; \
@@ -3774,6 +3776,8 @@
     {\
         Identifier = SID_GRID_USE ; \
         HelpID = SID_GRID_USE; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
 		Text = "~Am Raster fangen"; \
 		Text [ ENGLISH ] =  "Snap to grid"; \
 		Text[ english_us ] = "Snap to Grid"; \
@@ -3888,6 +3892,8 @@
     {\
         Identifier = SID_HELPLINES_VISIBLE ; \
         HelpID = SID_HELPLINES_VISIBLE ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         /* ### ACHTUNG: Neuer Text in Resource? ~Fanglinien sichtbar : ~Hilfslinien sichtbar */\
         Text = "~Fanglinien sichtbar" ; \
         Text [ ENGLISH ] = "Helplines visible" ; \
@@ -3924,6 +3930,8 @@
     {\
         Identifier = SID_HELPLINES_USE ; \
         HelpID = SID_HELPLINES_USE ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
 		Text = "~An Fanglinien fangen"; \
 		Text [ english ] =  "Snap to Guides"; \
 		Text[ english_us ] = "Snap to Snap Lines"; \
@@ -4460,6 +4468,8 @@
     {\
 		Identifier = SID_OBJECT_ROTATE ; \
 		HelpID = SID_OBJECT_ROTATE ; \
+		_MenuItemFlags = MIB_CHECKABLE; \
+		Checkable = TRUE; \
         Text = "~Drehen" ; \
         Text [ ENGLISH ] = "~Rotate" ; \
 		Text[ english_us ] = "~Rotate";\
@@ -4635,6 +4645,8 @@
     {\
         Identifier = SID_BEZIER_EDIT ; \
         HelpID = SID_BEZIER_EDIT ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "Pu~nkte" ; \
         Text [ ENGLISH ] = "Poi~nts" ; \
 		Text[ english_us ] = "Poi~nts";\
@@ -4669,6 +4681,8 @@
     {\
 		Identifier = SID_GLUE_EDITMODE ; \
 		HelpID = SID_GLUE_EDITMODE ; \
+		_MenuItemFlags = MIB_CHECKABLE; \
+		Checkable = TRUE; \
 		Text = "~Klebepunkte"; \
 		Text[ ENGLISH ] =  "Glue Points"; \
 		Text[ english_us ] = "~Glue Points";\
Index: sd/source/ui/app/menuids2.src
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/menuids2.src,v
retrieving revision 1.62
diff -u -r1.62 menuids2.src
--- sd/source/ui/app/menuids2.src	17 Jul 2003 12:40:07 -0000	1.62
+++ sd/source/ui/app/menuids2.src	14 Nov 2003 11:13:25 -0000
@@ -618,6 +618,8 @@
     {\
         Identifier = SID_TOGGLECOMMONTASKBAR ; \
         HelpID = SID_TOGGLECOMMONTASKBAR ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+        Checkable = TRUE; \
         Text = "Prä~sentation" ; \
         Text [ ENGLISH ] = "~Presentation" ; \
         Text [ portuguese ] = "Apre~sentação" ; \
@@ -2375,6 +2377,8 @@
     {\
         Identifier = SID_AUTOSPELL_CHECK ; \
         HelpID = SID_AUTOSPELL_CHECK ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "~Automatisch prüfen" ; \
         Text [ English ] = "~Online spelling" ; \
         Text [ dutch ] = "~Automatisch controleren" ; \
@@ -2409,6 +2413,8 @@
     {\
         Identifier = SID_THESAURUS ; \
         HelpID = SID_THESAURUS ; \
+    	_MenuItemFlags = MIB_CHECKABLE; \
+    	Checkable = TRUE; \
         Text = "T~hesaurus..." ; \
         Text [ English ] = "~Thesaurus..." ; \
         Text [ norwegian ] = "~Synonymordlister..." ; \
@@ -2444,6 +2450,8 @@
     {\
         Identifier = SID_HYPHENATION ; \
         HelpID = SID_HYPHENATION ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "Silben~trennung" ; \
         Text [ English ] = "~Hyphenation" ; \
         Text [ english_us ] = "~Hyphenation" ; \
@@ -2582,6 +2590,8 @@
     {\
         Identifier = SID_ANIMATION_OBJECTS ; \
         HelpID = SID_ANIMATION_OBJECTS ; \
+    	_MenuItemFlags = MIB_CHECKABLE; \
+    	Checkable = TRUE; \
         Text = "~Animation" ; \
         Text [ ENGLISH ] = "~Animation" ; \
         Text [ norwegian ] = "~Animation" ; \
@@ -2617,6 +2627,8 @@
     {\
         Identifier = SID_EFFECT_WIN ; \
         HelpID = SID_EFFECT_WIN ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+    	Checkable = TRUE; \
         Text = "E~ffekt" ; \
         Text [ ENGLISH ] = "~Effect" ; \
         Text [ english_us ] = "~Effects" ; \
@@ -2651,6 +2663,8 @@
     {\
         Identifier = SID_ANIMATION_EFFECTS ; \
         HelpID = SID_ANIMATION_EFFECTS ; \
+    	_MenuItemFlags = MIB_CHECKABLE; \
+    	Checkable = TRUE; \
         Text = "~Interaktion..." ; \
         Text [ ENGLISH ] = "~Action..." ; \
         Text [ english_us ] = "~Interaction..." ; \
@@ -2968,6 +2982,8 @@
     {\
         Identifier = SID_GALLERY; \
         HelpID = SID_GALLERY; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "~Gallery" ; \
         Text [ ENGLISH ] = "~Gallery" ; \
         Text [ english_us ] = "~Gallery" ; \
Index: sd/source/ui/app/menuids3.src
===================================================================
RCS file: /cvs/graphics/sd/source/ui/app/menuids3.src,v
retrieving revision 1.41
diff -u -r1.41 menuids3.src
--- sd/source/ui/app/menuids3.src	17 Jul 2003 12:40:18 -0000	1.41
+++ sd/source/ui/app/menuids3.src	14 Nov 2003 11:13:25 -0000
@@ -491,6 +491,8 @@
     {\
         Identifier = SID_HIDE_SLIDE ; \
         HelpID = SID_HIDE_SLIDE ; \
+	_MenuItemFlags = MIB_CHECKABLE; \
+	Checkable = TRUE; \
         Text = "~Dia anzeigen" ; \
         Text [ ENGLISH ] = "Show slide" ; \
         Text [ norwegian ] = "Show slide" ; \
Index: sc/source/ui/src/menue.src
===================================================================
RCS file: /cvs/sc/sc/source/ui/src/menue.src,v
retrieving revision 1.104
diff -u -r1.104 menue.src
--- sc/source/ui/src/menue.src	17 Jul 2003 12:56:00 -0000	1.104
+++ sc/source/ui/src/menue.src	14 Nov 2003 11:13:26 -0000
@@ -2847,6 +2847,8 @@
 								{
 									Identifier = SID_AUTOSPELL_CHECK ;
 									HelpID = SID_AUTOSPELL_CHECK ;
+									_MenuItemFlags = MIB_CHECKABLE;
+									Checkable = TRUE;
 									Text = "~Automatisch prüfen" ;
 									Text [ English ] = "~Online spelling" ;
 									Text [ english_us ] = "~AutoSpellcheck" ;
@@ -2899,6 +2901,8 @@
 					{
 						Identifier = SID_THESAURUS ;
 						HelpId = SID_THESAURUS ;
+					    	_MenuItemFlags = MIB_CHECKABLE;
+					    	Checkable = TRUE;
 						Text = "T~hesaurus..." ;
 						Text [ English ] = "~Thesaurus..." ;
 						Text [ norwegian ] = "~Synonymordlister..." ;
@@ -7319,6 +7323,8 @@
 								{
 									Identifier = SID_AUTOSPELL_CHECK ;
 									HelpID = SID_AUTOSPELL_CHECK ;
+									_MenuItemFlags = MIB_CHECKABLE;
+									Checkable = TRUE;
 									Text = "~Automatisch prüfen" ;
 									Text [ English ] = "~Online spelling" ;
 									Text [ english_us ] = "~AutoSpellcheck" ;
Index: svx/inc/globlmn.hrc
===================================================================
RCS file: /cvs/graphics/svx/inc/globlmn.hrc,v
retrieving revision 1.123
diff -u -r1.123 globlmn.hrc
--- svx/inc/globlmn.hrc	22 Jul 2003 16:35:38 -0000	1.123
+++ svx/inc/globlmn.hrc	14 Nov 2003 11:13:27 -0000
@@ -208,6 +208,8 @@
     Identifier = SID_STYLE_DESIGNER ; \
 	Command = ".uno:DesignerDialog" ; \
     HelpID = SID_STYLE_DESIGNER ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE; \
     Text = "St~ylist" ; \
     Text [ English ] = "St~ylist" ; \
     Text [ english_us ] = "St~ylist" ; \
@@ -2783,6 +2785,8 @@
     Identifier = SID_NAVIGATOR ; \
 	Command = ".uno:Navigator" ; \
     HelpID = SID_NAVIGATOR ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE; \
     TEXT = "~Navigator" ; \
     TEXT [ English ] = "~Navigator" ; \
     TEXT [ norwegian ] = "~Navigeringsvindu" ; \
@@ -2852,6 +2856,8 @@
     Identifier = SID_PLUGINS_ACTIVE ; \
 	Command = ".uno:PlugInsActive" ; \
     HelpID = SID_PLUGINS_ACTIVE ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE; \
     Text = "~PlugIn" ; \
     Text [ English ] = "PlugIn" ; \
     Text [ english_us ] = "P~lug-in" ; \
@@ -2887,6 +2893,8 @@
     Identifier = SID_IMAP ; \
 	Command = ".uno:ImageMapDialog" ; \
     HelpID = SID_IMAP ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE;  \
     Text = "I~mageMap" ; \
     Text [ English ] = "ImageMap" ; \
     Text [ english_us ] = "ImageMap" ; \
@@ -3315,6 +3323,8 @@
     Identifier = SID_BROWSER_MODE ; \
 	Command = ".uno:BrowseView" ; \
     HelpID = SID_BROWSER_MODE ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE; \
     Text = "~Online-Layout" ; \
     Text [ ENGLISH ] = "~Online layout" ; \
     Text [ english_us ] = "~Online Layout" ; \
@@ -3420,6 +3430,8 @@
     Identifier = SID_HYPERLINK_INSERT ; \
 	Command = ".uno:InsertHyperlink" ; \
     HelpID = SID_HYPERLINK_INSERT ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+     Checkable = TRUE; \
     Text = "H~yperlinkleiste" ; \
     Text [ English ] = "H~yperlinkleiste" ; \
     Text [ english_us ] = "~Hyperlink Bar" ; \
@@ -3647,6 +3659,8 @@
     Identifier = SID_GALLERY ; \
 	Command = ".uno:Gallery" ; \
     HelpID = SID_GALLERY ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+     Checkable = TRUE; \
     Text = "Ga~llery" ; \
     Text [ ENGLISH ] = "Ga~llery" ; \
     Text [ english_us ] = "~Gallery" ; \
@@ -5151,6 +5165,8 @@
     Identifier = SID_HELP_PI ; \
 	Command = ".uno:HelperDialog" ; \
     HelpID = SID_HELP_PI ; \
+    _MenuItemFlags = MIB_CHECKABLE; \
+    Checkable = TRUE; \
     TEXT = "He~lp Agent" ; \
     TEXT [ English ] = "He~lp Agent" ; \
     Text [ english_us ] = "Help ~Agent" ; \
@@ -5723,6 +5739,8 @@
 	Identifier = SID_VIEW_DATA_SOURCE_BROWSER; \
 	Command = ".uno:ViewDataSourceBrowser" ; \
 	HelpID = SID_VIEW_DATA_SOURCE_BROWSER; \
+        _MenuItemFlags = MIB_CHECKABLE; \
+        Checkable = TRUE; \
 	Text = "~Datenquellen" ; \
 	Text [ english ] = "~Data source browser" ; \
 	Text [ english_us ] = "~Data Sources" ; \
