--- oo/scp/source/office/files.scp.orig	Fri May  9 14:18:37 2003
+++ oo/scp/source/office/files.scp	Sat May 24 21:59:07 2003
@@ -429,16 +429,6 @@
     Dir             = GID_DIR_PROGRAM;
 	Styles			= (PACKED);
 End
-File GID_FILE_LIB_MOZABDRV
-	TXT_FILE_BODY;
-	#ifdef UNX
-    Name            = "libmozabdrv2.so";
-	#else
-    Name            = "mozabdrv2.dll";
-	#endif
-    Dir             = GID_DIR_PROGRAM;
-	Styles			= (PACKED);
-End
 
 
 STD_UNO_LIB_FILE( GID_FILE_LIB_DBU, dbu )
@@ -790,18 +780,6 @@
         #endif
 End
 
-File GID_FILE_LIB_MOZAB_2
-        TXT_FILE_BODY;
-        Styles          = (PACKED,UNO_COMPONENT);
-        RegistryID      = GID_STARREGISTRY_SERVICES_RDB;
-        Dir             = GID_DIR_PROGRAM;
-        #ifdef UNX
-        Name            = "libmozab2.so";
-        #else
-        Name            = "mozab2.dll";
-        #endif
-End
-
 #ifdef UNX
 
 File GID_FILE_LIB_EVOAB_1
@@ -3214,10 +3192,3 @@
 #endif
 
 #endif                                  // of #ifdef SOLAR_JAVA
-
-File GID_FILE_MOZILLA_RUNTIME
-        TXT_FILE_BODY;
-        Styles = (ARCHIVE);
-        Dir    = GID_DIR_PROGRAM;
-        Name	= "mozruntime.zip";
-End
