--- oo/i18nutil/source/utility/widthfolding.cxx.orig	Sat Jul 12 16:44:09 2003
+++ oo/i18nutil/source/utility/widthfolding.cxx	Sat Jul 12 16:43:27 2003
@@ -60,7 +60,7 @@
  ************************************************************************/
 
 // prevent internal compiler error with MSVC6SP3
-#include <stl/utility>
+//#include <stl/utility>
 #include <i18nutil/widthfolding.hxx>
 #include <i18nutil/x_rtl_ustring.h>
 #include "widthfolding_data.h"
