diff -Nur oo/connectivity/source/drivers/mozab/makefile.mk oo.new/connectivity/source/drivers/mozab/makefile.mk
--- oo/connectivity/source/drivers/mozab/makefile.mk	Mon Mar 11 16:07:41 2002
+++ oo.new/connectivity/source/drivers/mozab/makefile.mk	Sun Apr 21 14:35:50 2002
@@ -85,12 +85,12 @@
 ENABLE_EXCEPTIONS=TRUE
 LDUMP=ldump2.exe
 # --- Settings ----------------------------------
-.IF "$(DBGUTIL_OJ)"!=""
-ENVCFLAGS+=/FR$(SLO)$/
-.ENDIF
+#.IF "$(DBGUTIL_OJ)"!=""
+#ENVCFLAGS+=/FR$(SLO)$/
+#.ENDIF
 
-.INCLUDE : settings.mk
-.INCLUDE :  $(PRJ)$/version.mk
+#.INCLUDE : settings.mk
+#.INCLUDE :  $(PRJ)$/version.mk
 
 
 # --- Recursiveness  ---------------------------------------------------
@@ -106,10 +106,10 @@
 
 # --- MOZAB BASE Library -----------------------------------
 
-SHL1VERSIONMAP= $(TARGET).map
-.IF "$(OS)"!="FREEBSD"
-SHL1TARGET=	$(TARGET)$(MOZAB_MAJOR)
-.ENDIF
+#SHL1VERSIONMAP= $(TARGET).map
+#.IF "$(OS)"!="FREEBSD"
+#SHL1TARGET=	$(TARGET)$(MOZAB_MAJOR)
+#.ENDIF
 SHL1OBJS=$(SLOFILES)
 SHL1STDLIBS=\
 	$(CPPULIB)					\
@@ -180,27 +180,27 @@
 
 # --- Targets ----------------------------------
 
-.INCLUDE : target.mk
+#.INCLUDE : target.mk
 
 # --- filter file ------------------------------
 
-.IF "$(depend)"==""
+#.IF "$(depend)"==""
 
-$(MISC)$/$(SHL1TARGET).flt: makefile.mk
-	@echo ------------------------------
-    @echo CLEAR_THE_FILE	> $@
-	@echo _TI				>>$@
-	@echo _real				>>$@
-.ENDIF
-
-.IF "$(depend)"==""
-
-$(MISC)$/$(SHL2TARGET).flt: makefile.mk
-	@echo ------------------------------
-    @echo CLEAR_THE_FILE	> $@
-	@echo _TI				>>$@
-	@echo _real				>>$@
-.ENDIF
+#$(MISC)$/$(SHL1TARGET).flt: makefile.mk
+#	@echo ------------------------------
+#    @echo CLEAR_THE_FILE	> $@
+#	@echo _TI				>>$@
+#	@echo _real				>>$@
+#.ENDIF
+
+#.IF "$(depend)"==""
+
+#$(MISC)$/$(SHL2TARGET).flt: makefile.mk
+#	@echo ------------------------------
+#    @echo CLEAR_THE_FILE	> $@
+#	@echo _TI				>>$@
+#	@echo _real				>>$@
+#.ENDIF
 
 killdpc: 
 	-+$(RM) $(DPCTARGET)
