--- dictionaries/en_US/makefile.mk-old	2003-09-24 13:09:06.000000000 +0200
+++ dictionaries/en_US/makefile.mk	2003-09-24 13:09:24.000000000 +0200
@@ -107,8 +107,8 @@
 #unzip the text input files
 $(MISC)$/th_en_US_words.txt:	th_en_US_words.zip
 # cd is a 4nt internal command, we need a '+'
-	+cd $(MISC) && $(WRAPCMD) jar xf $(PWD)$/th_en_US_words.zip
+	+cd $(MISC) && $(WRAPCMD) unzip -o $(PWD)$/th_en_US_words.zip
 
 $(MISC)$/th_en_US_thes.txt:	th_en_US_thes.zip
 # ditto
-	+cd $(MISC) && $(WRAPCMD) jar xf $(PWD)$/th_en_US_thes.zip
+	+cd $(MISC) && $(WRAPCMD) unzip -o $(PWD)$/th_en_US_thes.zip
