Index: vcl/unx/source/window/salframe.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/window/salframe.cxx,v
retrieving revision 1.163
diff -u -p -u -r1.163 salframe.cxx
--- vcl/unx/source/window/salframe.cxx	1 Jul 2003 14:49:01 -0000	1.163
+++ vcl/unx/source/window/salframe.cxx	9 Sep 2003 14:24:38 -0000
@@ -3304,8 +3304,8 @@ long SalFrameData::HandleReparentEvent( 
      *  note: this works because hWM_Parent is direct child of root,
      *  not necessarily parent of GetShellWindow()
      */
-    pFrame_->maGeometry.nX		= xp + pFrame_->maGeometry.nLeftDecoration;
-    pFrame_->maGeometry.nY		= yp + pFrame_->maGeometry.nTopDecoration;
+    pFrame_->maGeometry.nX		= xp + nLeft;
+    pFrame_->maGeometry.nY		= yp + nTop;
     bool bResized = w != pFrame_->maGeometry.nWidth || h != pFrame_->maGeometry.nHeight;
     pFrame_->maGeometry.nWidth	= w;
     pFrame_->maGeometry.nHeight = h;
