diff -urN ooo-build-1.1.55.org/patches/OOO_1_1_1/apply ooo-build-1.1.55/patches/OOO_1_1_1/apply
--- ooo-build-1.1.55.org/patches/OOO_1_1_1/apply	2004-05-06 14:39:57.902700112 +0200
+++ ooo-build-1.1.55/patches/OOO_1_1_1/apply	2004-05-06 14:40:43.252805848 +0200
@@ -23,7 +23,7 @@
 KDEBase : LinuxCommon, KDECommon, NativeWidgets, StartupNotification, ODKFixes
 KDE : KDEBase, KDEOnly
 SuSE : SuSEOnly, LinuxCommon, KDEBase, GlobalDesktopFiles, VBABits
-PLD : PLDOnly, LinuxCommon, KDEBase, OOoCUPS, GlobalDesktopFiles, StartupNotification, ODKFixes, VBABits
+PLD : PLDOnly, LinuxCommon, GnomeVFS, GnomeCUPS, GlobalDesktopFiles, StartupNotification, ODKFixes, VBABits
 Ark : ArkOnly, LinuxCommon, NativeWidgets, GlobalDesktopFiles, ODKFixes, VBABits, OOoCUPS
 # somewhat odder:
 Win32 : Common, ODKFixes, VBABits, Win32Only
@@ -318,7 +318,7 @@
 # Disable broken crashdialog bits
 module-buildset-crashrep.diff
 
-[ XimianOnly, RedHatOnly ]
+[ XimianOnly, RedHatOnly, PLDOnly ]
 # Disable custom staroffice printer admin thing
 module-buildset.diff
 
