Index: vcl/source/glyphs/gcach_ftyp.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/glyphs/gcach_ftyp.cxx,v
retrieving revision 1.90
diff -u -p -u -r1.90 gcach_ftyp.cxx
--- vcl/source/glyphs/gcach_ftyp.cxx	24 Apr 2003 10:29:11 -0000	1.90
+++ vcl/source/glyphs/gcach_ftyp.cxx	13 May 2003 12:55:20 -0000
@@ -113,6 +113,13 @@
 #include <settings.hxx>
 #include <tools/lang.hxx>
 
+#if defined(FT_NEXT_SHORT) && !defined(NEXT_Short)
+// Freetype 2.1 isn't fully compatible with the *internal(!)* API used by OOo
+#  define NEXT_Short(x)  FT_NEXT_SHORT(x)
+#  define NEXT_UShort(x) FT_NEXT_USHORT(x)
+#  define NEXT_Long(x)	 FT_NEXT_LONG(x)
+#endif
+
 // -----------------------------------------------------------------------
 
 static FT_Library aLibFT = 0;
