diff -Nur oo/solenv/inc/unxlngi3.mk oo.new/solenv/inc/unxlngi3.mk
--- oo/solenv/inc/unxlngi3.mk	Fri Apr 19 18:59:59 2002
+++ oo.new/solenv/inc/unxlngi3.mk	Sun Apr 21 20:13:28 2002
@@ -85,9 +85,9 @@
 .ENDIF 
 
 # name of C++ Compiler
-CC=g++
+CC=g++2
 # name of C Compiler
-cc=gcc
+cc=gcc2
 # flags for C and C++ Compiler
 # do not use standard header search paths
 # if installed elsewhere
@@ -135,7 +135,7 @@
 DYNAMIC		= -Wl,-Bdynamic
 
 # name of linker
-LINK=gcc
+LINK=gcc2
 # default linker flags
 LINKFLAGS=
 
