--- scp/source/global/udk_base.scp.orig	2003-08-11 11:22:39.000000000 -0700
+++ scp/source/global/udk_base.scp	2003-08-11 11:38:05.000000000 -0700
@@ -342,6 +342,7 @@
 	#endif
 End
 
+#ifdef SOLAR_JAVA
 File gid_File_Lib_Jvmaccess
     TXT_FILE_BODY;
     Dir = GCFG_BINARY_DIR;
@@ -368,3 +369,4 @@
     Styles = (NETWORK,RELATIVE);
 End
 #endif
+#endif
--- scp/source/global/setupzip.scp.orig	2003-08-11 11:23:38.000000000 -0700
+++ scp/source/global/setupzip.scp	2003-08-11 11:28:02.000000000 -0700
@@ -393,8 +393,6 @@
 
 #if !defined(IPL_PRODUCT) && !defined(OPL_PRODUCT) || defined(UNX)
 
-#ifdef SOLAR_JAVA
-
 File GID_FILE_LIB_J
 	TXT_FILE_BODY;
 	#ifdef UNX
@@ -408,8 +406,6 @@
 
 #endif
 
-#endif
-
 File GID_FILE_LIB_COMPHELPER2
 	TXT_FILE_BODY;
 	#ifdef UNX
@@ -527,12 +523,14 @@
 
 #endif
 
+#ifdef SOLAR_JAVA
 File GID_FILE_JAR_UNOIL
 	TXT_FILE_BODY;
 	Name            = "unoil.jar";
 	Dir             = GCFG_CLASSES_DIR;
 	Styles          = (PACKED,DONT_OVERWRITE,SETUPZIP);
 End
+#endif
 
 #ifdef INCLUDE_JAVA_ACCESSBRIDGE
 
@@ -546,14 +544,14 @@
 #endif
 
 #ifdef UNX
-
+#ifdef SOLAR_JAVA
 File GID_FILE_BIN_JAVALDX
         BIN_FILE_BODY;
         Dir             = GCFG_BINARY_DIR;
         Styles          = (PACKED, SETUPZIP);
         Name            = EXENAME(javaldx);
 End
-
+#endif
 #endif
 
 #ifdef UNX
--- scp/source/global/udk_complete.scp.orig	2003-08-11 11:28:16.000000000 -0700
+++ scp/source/global/udk_complete.scp	2003-08-11 11:29:30.000000000 -0700
@@ -75,6 +75,7 @@
     #endif
 End
 
+#ifdef SOLAR_JAVA
 #ifndef MACOSX
 File gid_File_Lib_OfficeBean
     TXT_FILE_BODY;
@@ -94,6 +95,8 @@
      Styles          = (PACKED); 
 End
 #endif
+#endif
+
 /*
 File gid_File_Lib_Stm
     TXT_FILE_BODY;
--- scp/source/office/files.scp.orig	2003-08-11 11:29:45.000000000 -0700
+++ scp/source/office/files.scp	2003-08-11 11:30:15.000000000 -0700
@@ -715,6 +715,7 @@
 
 #endif
 
+#ifdef SOLAR_JAVA
 File GID_FILE_LIB_JDBC_2
         TXT_FILE_BODY;
         Styles          = (PACKED,UNO_COMPONENT);
@@ -726,6 +727,7 @@
         Name            = "jdbc2.dll";
         #endif
 End
+#endif
 
 File GID_FILE_LIB_ODBC_2
         TXT_FILE_BODY;
--- scp/util/makefile.mk.orig	2003-08-11 11:30:57.000000000 -0700
+++ scp/util/makefile.mk	2003-08-11 11:34:00.000000000 -0700
@@ -88,7 +88,6 @@
                 module_math.par         \
                 module_grafik.par       \
                 module_lingu.par        \
-                module_javafilter.par	\
                 profile.par             \
                 profile_calc.par        \
                 profile_impress.par     \
@@ -96,13 +95,10 @@
                 profile_math.par        \
                 profile_chart.par       \
                 profile_grafik.par      \
-                profile_javafilter.par	\
                 setupzip.par            \
                 readme.par	            \
                 udk_base.par            \
                 udk_bridges.par         \
-                udk_java.par			\
-                udk_pure_java.par		\
                 udk_remote.par			\
                 udk_complete.par		\
                 files.par				\
@@ -113,10 +109,15 @@
                 files_chart.par         \
                 files_grafik.par        \
                 files_lingu.par         \
-                files_javafilter.par	\
                 dir.par					\
                 python.par
-
+.IF "$(SOLAR_JAVA)"!=""
+SCP1FILES +=    module_javafilter.par	\
+                profile_javafilter.par	\
+                files_javafilter.par	\
+                udk_java.par			\
+                udk_pure_java.par
+.ENDIF
 
 #python.par
 
