--- config_office/configure.in~	2005-07-28 00:12:10.000000000 +0300
+++ config_office/configure.in	2005-07-28 00:12:50.000000000 +0300
@@ -1751,10 +1751,14 @@
 				   fi
 			   fi
 		   else
-			   if test -f "$STLPORT4/lib/libstlport_gcc.so"; then
+			   if test -f "$STLPORT4/lib64/libstlport_gcc.so"; then
 				   AC_MSG_RESULT([checked])
 			   else
-				   AC_MSG_ERROR([STLport4 libraries not found])
+               if test -f "$STLPORT4/lib/libstlport_gcc.so"; then
+                  AC_MSG_RESULT([checked])
+               else
+                  AC_MSG_ERROR([STLport4 libraries not found])
+               fi
 			   fi
 		   fi
 	   fi


# this one is ugly, it will break if you have STLPORT installed for both (32/64) archs
# but i didn't want to make this patch arch specific either. perhaps test for
# glob would be cleaner solution. glen 2005-07-28
--- stlport/makefile.mk~	2005-07-28 01:22:23.000000000 +0300
+++ stlport/makefile.mk	2005-07-28 01:22:24.000000000 +0300
@@ -207,7 +207,7 @@
 all :
        @echo "         An already available installation of STLport has been chosen in the configure process."
        @echo "         Therefore the version provided here does not need to be built in addition."
-       +$(COPY) $(STLPORT4)$/lib$/*stlport*$(DLLPOST) $(DLLDEST)
+       +$(COPY) $(STLPORT4)$/lib*$/*stlport*$(DLLPOST) $(DLLDEST)
 .ENDIF
 
 .INCLUDE : 	set_ext.mk
