--- oo_1.1rc4_src/psprint_config/configuration/letter/psprint.conf.psprint-pdf-drivers	2002-12-10 06:59:31.000000000 -0500
+++ oo_1.1rc4_src/psprint_config/configuration/letter/psprint.conf	2003-09-11 13:12:02.000000000 -0400
@@ -96,3 +96,83 @@ SubstFont_Times New Roman=Times
 SubstFont_Thorndale=Times
 SubstFont_Timmons=Times
 SubstFont_Cumberland=Courier
+
+; [gghibo] Entries to enable a PDF writer with PDF 1.4 output and
+; bitmap not downsampled (useful for press).
+[PDF 1.4 Converter Screen]
+Printer=ADISTILL/PDF 1.4 Converter Screen
+DefaultPrinter=0
+Location=
+Comment=
+Command=/usr/bin/gs -q -dNOPAUSE -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -sOutputFile="(OUTFILE)" -
+Features=pdf=
+Copies=1
+Scale=0
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=Letter
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
+
+[PDF 1.4 Converter Press]
+Printer=ADISTILL/PDF 1.4 Converter Press
+DefaultPrinter=0
+Location=
+Comment=
+Command=/usr/bin/gs -q -dNOPAUSE -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/prepress -dAutoFilterColorImages=false -sColorImageFilter=FlateEncode -dAutoFilterGrayImages=false -sGrayImageFilter=FlateEncode -sOutputFile="(OUTFILE)" -
+Features=pdf=
+Copies=1
+Scale=0
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=Letter
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
+
+; [gghibo] entries to produce distillable PostScript output
+;
+[Generic PostScript output (distillable)]
+Printer=ADISTILL/Generic PostScript output (distillable)
+DefaultPrinter=0
+Location=
+Comment=
+Command=
+Features=
+Copies=1
+Scale=1
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=Letter
+PPD_Resolution=600dpi
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
--- oo_1.1rc4_src/psprint_config/configuration/psprint.conf.psprint-pdf-drivers	2002-02-28 11:05:08.000000000 -0500
+++ oo_1.1rc4_src/psprint_config/configuration/psprint.conf	2003-09-09 17:39:45.000000000 -0400
@@ -96,3 +96,83 @@ SubstFont_Times New Roman=Times
 SubstFont_Thorndale=Times
 SubstFont_Timmons=Times
 SubstFont_Cumberland=Courier
+
+; [gghibo] Entries to enable a PDF writer with PDF 1.4 output and
+; bitmap not downsampled (useful for press).
+[PDF 1.4 Converter Screen]
+Printer=ADISTILL/PDF 1.4 Converter Screen
+DefaultPrinter=0
+Location=
+Comment=
+Command=/usr/bin/gs -q -dNOPAUSE -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -sOutputFile="(OUTFILE)" -
+Features=pdf=
+Copies=1
+Scale=0
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=A4
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
+
+[PDF 1.4 Converter Press]
+Printer=ADISTILL/PDF 1.4 Converter Press
+DefaultPrinter=0
+Location=
+Comment=
+Command=/usr/bin/gs -q -dNOPAUSE -sDEVICE=pdfwrite -dCompatibilityLevel=1.4 -dPDFSETTINGS=/prepress -dAutoFilterColorImages=false -sColorImageFilter=FlateEncode -dAutoFilterGrayImages=false -sGrayImageFilter=FlateEncode -sOutputFile="(OUTFILE)" -
+Features=pdf=
+Copies=1
+Scale=0
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=A4
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
+
+; [gghibo] entries to produce distillable PostScript output
+;
+[Generic PostScript output (distillable)]
+Printer=ADISTILL/Generic PostScript output (distillable)
+DefaultPrinter=0
+Location=
+Comment=
+Command=
+Features=
+Copies=1
+Scale=1
+Orientation=Portrait
+PSLevel=0
+ColorDevice=0
+ColorDepth=24
+MarginAdjust=0,0,0,0
+PPD_PageSize=A4
+PPD_Resolution=600dpi
+PerformFontSubstitution=true
+SubstFont_Arial=Helvetica
+SubstFont_Times New Roman=Times
+SubstFont_Cumberland=Courier
+SubstFont_Timmons=Times
+SubstFont_Helmet=Helvetica
+SubstFont_Thorndale=Times
+SubstFont_Albany=Helvetica
+SubstFont_Courier New=Courier
