--- oo_1.1_src/offmgr/source/offapp/intro/ooo.src.orig	2003-09-22 20:48:59.000000000 +0200
+++ oo_1.1_src/offmgr/source/offapp/intro/ooo.src	2003-11-30 16:04:31.000000000 +0100
@@ -81,14 +81,14 @@
 #define OOO_LICENSE "LGPL"
 #define OOO_INTRO "openintro_sun.bmp"
 #else
-//#define OOO_VENDOR "my company"
+#define OOO_VENDOR "PLD Linux Distribution"
 #define OOO_LICENSE "LGPL"
-#define OOO_INTRO "openintro.bmp"
+#define OOO_INTRO "openintro_pld.bmp"
 #endif
 
 Bitmap RID_DEFAULT_ABOUT_BMP_LOGO
 {
-	File = "openabout.bmp" ;
+	File = "openabout_pld.bmp" ;
 };
 
 Bitmap RID_DEFAULTINTRO
