--- config_office/configure.in~	2007-03-28 00:03:45.805243465 +0300
+++ config_office/configure.in	2007-03-28 00:21:10.154123488 +0300
@@ -2235,9 +2235,9 @@
 				   fi
 			   fi
 		   else
-			   if test -f "$STLPORT4/lib/libstlport_gcc.so"; then
+			   if test -f "$STLPORT4/lib64/libstlport_gcc.so"; then
 				   AC_MSG_RESULT([checked])
-			   elif test -f "$STLPORT4/lib/libstlport.so"; then
+			   elif test -f "$STLPORT4/lib64/libstlport.so"; then
 				   AC_MSG_RESULT([checked])
 				   STLPORT_VER=500
 			   else
