--- ooo-build-1.1.52/patches/OOO_1_1_1/apply.rh	2004-03-31 03:46:36.000000000 -0500
+++ ooo-build-1.1.52/patches/OOO_1_1_1/apply	2004-04-18 17:42:15.880895629 -0400
@@ -45,6 +45,10 @@
 icu-build-no-extras.diff
 
 [ Misc ]
+# disable "Check in all languages" in the UI and set the
+# value to false in preferences
+gui-disable-spellcheck-all-langs.diff
+
 # fix atomic dec/inc for Sparc #15502 patch_to_boost_patch (also #14848)
 build-sparc-boost.diff
 

