diff -urN ooo-build-1.1.54.org/patches/OOO_1_1_1/apply ooo-build-1.1.54/patches/OOO_1_1_1/apply
--- ooo-build-1.1.54.org/patches/OOO_1_1_1/apply	2004-05-05 23:34:31.946500792 +0200
+++ ooo-build-1.1.54/patches/OOO_1_1_1/apply	2004-05-05 23:35:05.973327928 +0200
@@ -22,7 +22,7 @@
 RHShrike : RedHatBase, RHShrikeOnly, OldLibArt
 KDE : LinuxCommon, NativeWidgets, StartupNotification, ODKFixes
 SuSE : SuSEOnly, LinuxCommon, KDE, GlobalDesktopFiles, VBABits
-PLD: PLDOnly, LinuxCommon, KDE, OOoCUPS, GlobalDesktopFiles, StartupNotification, ODKFixes, VBABits
+PLD: PLDOnly, LinuxCommon, GnomeVFS, GnomeCUPS, GlobalDesktopFiles, StartupNotification, ODKFixes, VBABits
 # somewhat odder:
 Win32 : Common, ODKFixes, VBABits, Win32Only
 
@@ -306,7 +306,7 @@
 gui-ft-pt-not-pixel.diff
 gui-ft-use-system-font.diff
 
-[ XimianOnly, RedHatOnly ]
+[ XimianOnly, RedHatOnly, PLDOnly ]
 # Disable custom staroffice printer admin thing & broken crashdialog bits
 module-buildset.diff
 
