Index: svtools/source/dialogs/printdlg.cxx
===================================================================
RCS file: /cvs/util/svtools/source/dialogs/printdlg.cxx,v
retrieving revision 1.16
diff -u -p -u -r1.16 printdlg.cxx
--- svtools/source/dialogs/printdlg.cxx	11 Apr 2003 17:01:36 -0000	1.16
+++ svtools/source/dialogs/printdlg.cxx	29 May 2003 14:07:34 -0000
@@ -431,6 +431,9 @@
         {
             try
             {
+		INetURLObject aLocation( maFiPrintFile.GetText(), INET_PROT_FILE );
+		xFilePicker->setDefaultName( aLocation.GetLastName() );
+		xFilePicker->setDisplayDirectory ( aLocation.GetPartBeforeLastName() );
 #ifdef UNX
                 // add PostScript and PDF
                 if( ! Application::IsRemoteServer() )
@@ -455,6 +458,10 @@
 #endif
                 // add arbitrary files
                 xFilterMgr->appendFilter( maAllFilterStr, OUString( RTL_CONSTASCII_USTRINGPARAM( "*.*" ) ) );
+
+		String aFilter = pPrinter->GetJobValue( String::CreateFromAscii( "Filter#" ) );
+		if (aFilter.Len())
+			xFilterMgr->setCurrentFilter( aFilter );
             }
             catch( IllegalArgumentException rExc )
             {
@@ -465,6 +472,13 @@
                 Sequence< OUString > aPathSeq( xFilePicker->getFiles() );
                 INetURLObject aObj( aPathSeq[0] );
                 maFiPrintFile.SetText( aObj.PathToFileName() );
+
+#ifdef UNX
+		Printer* pPrinter = TEMPPRINTER() ? TEMPPRINTER() : mpPrinter;
+		if ( pPrinter )
+			pPrinter->SetJobValue( String::CreateFromAscii( "Filter#" ),
+					xFilterMgr->getCurrentFilter() );
+#endif
                 return true;
             }
             return false;

