--- offapi/com/sun/star/ucb/GnomeVFSContentProvider.idl
+++ offapi/com/sun/star/ucb/GnomeVFSContentProvider.idl
@@ -0,0 +1,21 @@
+#ifndef __com_sun_star_ucb_GnomeVFSContentProvider_idl__
+#define __com_sun_star_ucb_GnomeVFSContentProvider_idl__
+
+#ifndef __com_sun_star_ucb_XContentProvider_idl__
+#include <com/sun/star/ucb/XContentProvider.idl>
+#endif
+
+//=============================================================================
+
+module com { module sun { module star { module ucb {
+
+service GnomeVFSContentProvider
+{
+	interface com::sun::star::ucb::XContentProvider;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
--- offapi/com/sun/star/ucb/GnomeVFSDocumentContent.idl
+++ offapi/com/sun/star/ucb/GnomeVFSDocumentContent.idl
@@ -0,0 +1,97 @@
+#ifndef __com_sun_star_ucb_GnomeVFSDocumentContent_idl__
+#define __com_sun_star_ucb_GnomeVFSDocumentContent_idl__
+
+#ifndef __com_sun_star_lang_XComponent_idl__
+#include <com/sun/star/lang/XComponent.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XContent_idl__
+#include <com/sun/star/ucb/XContent.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandProcessor_idl__
+#include <com/sun/star/ucb/XCommandProcessor.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandProcessor2_idl__
+#include <com/sun/star/ucb/XCommandProcessor2.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertiesChangeNotifier_idl__
+#include <com/sun/star/beans/XPropertiesChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertyContainer_idl__
+#include <com/sun/star/beans/XPropertyContainer.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertySetInfoChangeNotifier_idl__
+#include <com/sun/star/beans/XPropertySetInfoChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandInfoChangeNotifier_idl__
+#include <com/sun/star/ucb/XCommandInfoChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_container_XChild_idl__
+#include <com/sun/star/container/XChild.idl>
+#endif
+
+//=============================================================================
+
+module com { module sun { module star { module ucb {
+
+service GnomeVFSDocumentContent
+{
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::lang::XComponent;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::ucb::XContent;
+
+	interface com::sun::star::ucb::XCommandProcessor;
+
+	[optional] interface com::sun::star::ucb::XCommandProcessor2;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertiesChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertyContainer;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertySetInfoChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::ucb::XCommandInfoChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::container::XChild;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
--- offapi/com/sun/star/ucb/GnomeVFSFolderContent.idl
+++ offapi/com/sun/star/ucb/GnomeVFSFolderContent.idl
@@ -0,0 +1,109 @@
+#ifndef __com_sun_star_ucb_GnomeVFSFolderContent_idl__
+#define __com_sun_star_ucb_GnomeVFSFolderContent_idl__
+
+#ifndef __com_sun_star_lang_XComponent_idl__
+#include <com/sun/star/lang/XComponent.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XContent_idl__
+#include <com/sun/star/ucb/XContent.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandProcessor_idl__
+#include <com/sun/star/ucb/XCommandProcessor.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandProcessor2_idl__
+#include <com/sun/star/ucb/XCommandProcessor2.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertiesChangeNotifier_idl__
+#include <com/sun/star/beans/XPropertiesChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertyContainer_idl__
+#include <com/sun/star/beans/XPropertyContainer.idl>
+#endif
+
+#ifndef __com_sun_star_beans_XPropertySetInfoChangeNotifier_idl__
+#include <com/sun/star/beans/XPropertySetInfoChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XCommandInfoChangeNotifier_idl__
+#include <com/sun/star/ucb/XCommandInfoChangeNotifier.idl>
+#endif
+
+#ifndef __com_sun_star_container_XChild_idl__
+#include <com/sun/star/container/XChild.idl>
+#endif
+
+#ifndef __com_sun_star_ucb_XContentCreator_idl__
+#include <com/sun/star/ucb/XContentCreator.idl>
+#endif
+
+//=============================================================================
+
+module com { module sun { module star { module ucb {
+
+//=============================================================================
+/** A DCP Folder is a container for other DCP Folders or Documents.
+
+	@see com::sun::star::ucb::GnomeVFSContentProvider
+	@see com::sun::star::ucb::GnomeVFSDocumentContent
+*/
+service GnomeVFSFolderContent
+{
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::lang::XComponent;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::ucb::XContent;
+
+	interface com::sun::star::ucb::XCommandProcessor;
+
+	[optional] interface com::sun::star::ucb::XCommandProcessor2;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertiesChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertyContainer;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::beans::XPropertySetInfoChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::ucb::XCommandInfoChangeNotifier;
+
+	//-------------------------------------------------------------------------
+	/** This interface is implemented according to the specification of
+		service <type>Content</type>.
+	 */
+	interface com::sun::star::container::XChild;
+
+	interface com::sun::star::ucb::XContentCreator;
+};
+
+//=============================================================================
+
+}; }; }; };
+
+#endif
Index: offapi/com/sun/star/ucb/makefile.mk
===================================================================
RCS file: /cvs/api/offapi/com/sun/star/ucb/makefile.mk,v
retrieving revision 1.41
diff -u -p -u -r1.41 makefile.mk
--- offapi/com/sun/star/ucb/makefile.mk	27 May 2003 10:29:04 -0000	1.41
+++ offapi/com/sun/star/ucb/makefile.mk	29 Aug 2003 14:29:02 -0000
@@ -129,6 +129,9 @@ IDLFILES=\
 	FTPContent.idl\
 	FTPContentProvider.idl\
 	GlobalTransferCommandArgument.idl\
+	GnomeVFSContentProvider.idl\
+	GnomeVFSDocumentContent.idl\
+	GnomeVFSFolderContent.idl\
 	HandleCookiesRequest.idl\
 	HelpContent.idl\
 	HelpContentProvider.idl\
