--- oo/tools/bootstrp/static.mk.orig	Wed Jan 22 22:54:04 2003
+++ oo/tools/bootstrp/static.mk	Wed Jan 22 22:54:27 2003
@@ -2,9 +2,9 @@
 #
 #   $RCSfile$
 #
-#   $Revision$
+#   $Revision$
 #
-#   last change: $Author$ $Date$
+#   last change: $Author$ $Date$
 #
 #   The Contents of this file are made available subject to the terms of
 #   either of the following licenses
@@ -60,11 +60,21 @@
 #
 #*************************************************************************
 
+.IF "$(OFFICEUPD)"==""
+STATICUPD=$(UPD)
+.ELSE
+STATICUPD=$(OFFICEUPD)
+.ENDIF 
+
 .IF "$(GUI)"=="WNT"
 # -----------------------------------------------------------
 # --------------------- Windows32 ---------------------------
 # -----------------------------------------------------------
-SALLIBST  		=	$(SALLIB)
+#.IF "$(STATICUPD)" >= 640 
+#SALLIBST  		=	$(SALLIB)
+#.ELSE
+SALLIBST  		=	asal.lib
+#.ENDIF
 BOOTSTRP		=	bootstrp.lib
 TRANSEX			=	transex.lib
 VERMAP			= 	bsvermap.lib
@@ -81,20 +91,10 @@
 .ELSE
 STATIC_LIBS		+= 	atools.lib
 .ENDIF
-.IF "$(UPD)" >= "590"
-STATIC_LIBS		+=  kernel9x.lib 						\
-					advapi9x.lib 						\
-					shell9x.lib
-.IF "$(UPD)" >= "604"
-STATIC_LIBS		+=  tools32.lib
-.ENDIF
-.ENDIF
-.IF "$(UPD)" >= "546"
-STATIC_LIBS		+= 	$(SALLIB)
-.ELSE
-STATIC_LIBS		+= 	aosl.lib							\
-					artl.lib
-.ENDIF
+STATIC_LIBS		+= 	uwinapi.lib	\
+					ascii_expat_xmlparse.lib			\
+					expat_xmltok.lib			\
+					$(SALLIB)
 .IF "$(LIBSTLPORT)" != ""
 STATIC_LIBS		+= 	$(LIBSTLPORT)
 .ENDIF
@@ -123,16 +123,9 @@
 					$(LIBSTLPORTST)						\
 					$(STATIC)
 
-.IF "$(UPD)" >= "546"
-.IF "$(UPD)" >= "604"
-STATIC_LIBS		+= 	-lsal$(SAL_MAJOR)$(DLLPOSTFIX)
-.ELSE
-STATIC_LIBS     +=  -lsal$(UPD)$(DLLPOSTFIX)
-.ENDIF
-.ELSE
-STATIC_LIBS		+= 	-losl$(UPD)$(DLLPOSTFIX)			\
-					-lrtl$(UPD)$(DLLPOSTFIX)
-.ENDIF
+STATIC_LIBS		+= 	-lsal$(SAL_MAJOR)$(DLLPOSTFIX)		\
+					-lascii_expat_xmlparse				\
+					-lexpat_xmltok
 
 .IF "$(COM)"=="C52"
 .IF "$(BUILD_SOSL)"==""
@@ -148,8 +141,14 @@
 STATIC_LIBS		+= -lnsl -lsocket -lposix4 -lc
 .ENDIF
 .IF "$(OS)"=="LINUX"
-STATIC_LIBS		+= -lsupc++	-lX11 -lXext 
-.ENDIF
+.IF "$(COM)"=="GCC"
+.IF "$(CVER)"!="C295"
+STATIC_LIBS		+=	-lsupc++ -lX11 -lXext
+.ELSE 
+STATIC_LIBS		+=	-lX11 -lXext
+.ENDIF # CVER
+.ENDIF # GCC 
+.ENDIF # LINUX
 
 # -----------------------------------------------------------
 
@@ -162,12 +161,7 @@
 VERMAP			= 	bsvermap.lib
 STATIC_LIBS		=	$(L)$/atools.lib					\
 					$(L)$/avos.lib
-.IF "$(UPD)" >= "546"
 STATIC_LIBS		+= 	$(L)$/asal.lib
-.ELSE
-STATIC_LIBS		+= 	$(L)$/aosl.lib						\
-					$(L)$/artl.lib
-.ENDIF
 # -----------------------------------------------------------
 
 .ENDIF
