Index: sd/source/ui/view/viewshel.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/view/viewshel.cxx,v
retrieving revision 1.22
diff -u -p -u -r1.22 viewshel.cxx
--- sd/source/ui/view/viewshel.cxx	4 Jun 2003 11:06:02 -0000	1.22
+++ sd/source/ui/view/viewshel.cxx	30 Sep 2003 14:41:36 -0000
@@ -121,13 +121,6 @@ SO2_DECL_REF(SvInPlaceObject)
 
 #define DELTA_ZOOM  10
 
-// Use magenta as transparency color instead of the standard gray that is
-// imported from vcl/image.hxx
-#undef IMAGE_STDBTN_COLOR
-#undef IMAGE_STDBTN_COLOR_HC
-#define IMAGE_STDBTN_COLOR Color(0xff,0x00,0xff)
-#define IMAGE_STDBTN_COLOR_HC Color(0xff,0x00,0xff)
-
 /*************************************************************************
 |*
 |* SFX-Slotmap und Standardinterface deklarieren
Index: sd/source/ui/view/drviewsa.cxx
===================================================================
RCS file: /cvs/graphics/sd/source/ui/view/drviewsa.cxx,v
retrieving revision 1.17
diff -u -p -u -r1.17 drviewsa.cxx
--- sd/source/ui/view/drviewsa.cxx	22 May 2003 10:32:31 -0000	1.17
+++ sd/source/ui/view/drviewsa.cxx	30 Sep 2003 14:45:12 -0000
@@ -146,13 +146,6 @@ using namespace ::com::sun::star;
 
 BOOL SdDrawViewShell::bPipette = FALSE;
 
-// Use magenta as transparency color instead of the standard gray that is
-// imported from vcl/image.hxx
-#undef IMAGE_STDBTN_COLOR
-#undef IMAGE_STDBTN_COLOR_HC
-#define IMAGE_STDBTN_COLOR Color(0xff,0x00,0xff)
-#define IMAGE_STDBTN_COLOR_HC Color(0xff,0x00,0xff)
-
 // ------------------------
 // - ScannerEventListener -
 // ------------------------
