--- scp/prj/build.lst.orig	2003-07-15 14:35:49.000000000 +0000
+++ scp/prj/build.lst	2003-12-16 13:48:20.000000000 +0000
@@ -17,5 +17,4 @@
 cp	scp\source\kde							nmake	-	u	cp_kde NULL
 cp	scp\source\gnome						nmake	-	u	cp_gnome NULL
 cp	scp\source\lingu						nmake	-	all	cp_lingu NULL
-cp	scp\source\python						nmake   -       all     cp_python NULL
-cp	scp\util								nmake	-	all	cp_util cp_crashrep cp_calc cp_chart cp_draw cp_fonts cp_gbl cp_grffilt cp_impress cp_javafilter cp_math cp_off cp_office_root cp_lingu cp_cde.u cp_kde.u cp_gnome.u cp_python NULL
+cp	scp\util								nmake	-	all	cp_util cp_crashrep cp_calc cp_chart cp_draw cp_fonts cp_gbl cp_grffilt cp_impress cp_javafilter cp_math cp_off cp_office_root cp_lingu cp_cde.u cp_kde.u cp_gnome.u NULL
--- oo_1.1_src/scp/util/makefile.mk.orig	2003-08-21 11:46:53.000000000 +0200
+++ oo_1.1_src/scp/util/makefile.mk	2004-01-15 14:04:18.719375584 +0100
@@ -114,12 +114,9 @@
                 files_grafik.par        \
                 files_lingu.par         \
                 files_javafilter.par	\
-                dir.par					\
-                python.par
+                dir.par					
 
 
-#python.par
-
 .IF "$(GUI)"=="UNX"
 SCP1FILES += \
         dir_psprint.par   \
@@ -133,8 +130,7 @@
         files_cde.par		\
         unxbasic_kde.par	\
         unxbasic_gnome.par	\
-        unxbasic_cde.par \
-        unxbasic_python.par
+        unxbasic_cde.par
 .ELSE
 SCP1FILES += \
         basic.par
@@ -147,9 +143,7 @@
         system_impress.par      \
         system_draw.par         \
         system_math.par         \
-        system_chart.par	\
-        winbasic_python.par
-
+        system_chart.par
 .ENDIF
 
 .IF "$(ENABLE_CRASHDUMP)" != "" || "$(PRODUCT)" == "" 
