--- config_office/configure.in~	2008-07-18 13:01:39.977133894 +0200
+++ config_office/configure.in	2008-07-18 20:47:51.991159521 +0200
@@ -4170,7 +4170,7 @@
 
     #e.g. http://fedoraproject.org/wiki/Releases/FeatureXULRunnerAPIChanges
     #the plugin pkg-config etc. reverts to "mozilla-plugin" with libxul
-    if test "$MOZ_FLAVOUR" == "libxul"; then
+    if test "$MOZ_FLAVOUR" == "libxul" -o "$MOZ_FLAVOUR" == "xulrunner"; then
         MOZ_FLAVOUR="mozilla"
     fi
 
