--- ooo-build-2.2.0/patches/src680/apply	2007-03-27 02:54:40.198763000 +0300
+++ ooo-build-r9097/patches/src680/apply	2007-03-27 16:17:52.011699303 +0200
@@ -62,7 +62,8 @@
 Ark: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, SystemJavaBits
 Ark64: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, 64bit, SystemJavaBits
 # www.pld-linux.org
-PLDBase: LinuxCommon, NotDebian, PLDOnly, NovellOnly, NovellOnlyUnix, FilePickers, GCJExperimental, SystemJavaBits, SystemJavaBitsWorkaround, BerkeleyDB43, STLport5
+PLDBase: LinuxCommon, NotDebian, PLDOnly, SystemJavaBits, BerkeleyDB43, STLport5, \
+	CalcSolver
 PLD: PLDBase
 PLD64: PLDBase, 64bit, PLD64bitfixes
 # Mandriva Linux
@@ -1746,6 +1747,12 @@
 # we will move to gcc-4.x, so it does not make sense to push it upstream
 buildfix-boost-gcc33.diff
 
+[ PLDOnly ]
+build-wizard-bitmap-scp.diff
+scp2-user-config-ooo-2.0.diff
+pushfont-psstream.diff
+xmlhelp-use-libxslt-from-system.diff
+
 [ UnUsedButNotYetRemovedFromSVN ]
 # diffs kept for reference as there might soon be a need to resurrect
 # part of the code
