--- oo_1.0_src/svtools/source/svhtml/htmlout.cxx.jj	Thu Nov 29 12:41:03 2001
+++ oo_1.0_src/svtools/source/svhtml/htmlout.cxx	Fri Jul 12 21:16:44 2002
@@ -392,6 +392,15 @@ void lcl_ConvertCharToHTML( sal_Unicode 
 		break;
 	}
 
+	// Don't uglify ISO-8859-2 output unnecessarily
+	if( pStr && pStr[0] && eDestEnc == RTL_TEXTENCODING_ISO_8859_2 )
+	{
+		if( !strcmp( pStr + 1, "acute" ) && strchr( "iIyYaAeEoOuU", pStr[0] ) )
+			pStr = 0;
+		else if( !strcmp( pStr + 1, "caron" ) && ( pStr[0] == 's' || pStr[0] == 'S' ) )
+			pStr = 0;
+	}
+
 	if( pStr )
 	{
 		((rDest += '&') += pStr) += ';';
