--- solenv/inc/unxlngs.mk~	2004-02-24 00:20:16.000000000 +0100
+++ solenv/inc/unxlngs.mk	2004-02-24 00:42:46.000000000 +0100
@@ -64,7 +64,7 @@
 ASM=gcc
 AFLAGS=-Wa,-Av8plus,-K,PIC -c $(CDEFS)
 
-SOLAR_JAVA=TRUE
+SOLAR_JAVA=
 JAVAFLAGSDEBUG=-g
 
 # filter for supressing verbose messages from linker
@@ -95,6 +95,14 @@
 CFLAGS=
 .ENDIF
 CFLAGS+=-fmessage-length=0 -c $(INCLUDE)
+
+# flags to enable build with symbols; required for crashdump feature                                                      
+.IF "$(ENABLE_SYMBOLS)"=="SMALL"                                                                                          
+CFLAGSENABLESYMBOLS=-g1                                                                                                   
+.ELSE                                                                                                                     
+CFLAGSENABLESYMBOLS=-g                                                                                                    
+.ENDIF                                                                                                                    
+
 # flags for the C++ Compiler
 CFLAGSCC= -pipe 
 # Flags for enabling exception handling
