--- oo_1.0.1_src/sal/osl/unx/signal.c.segv	2003-01-14 03:59:12.000000000 -0500
+++ oo_1.0.1_src/sal/osl/unx/signal.c	2003-01-14 04:47:54.000000000 -0500
@@ -162,7 +162,8 @@ static sal_Bool InitSignal()
 
 	char ProgFile[512];
 
-	if( (NULL != osl_impl_getExecutableName(ProgFile, sizeof(ProgFile))) &&
+	if( 0 &&
+		(NULL != osl_impl_getExecutableName(ProgFile, sizeof(ProgFile))) &&
         (NULL != strstr(ProgFile, "soffice")) )
 	{
 		char CmdLine[512];
