--- oo_1.1rc3_src/sal/osl/unx/semaphor.c.jj	2003-08-28 10:24:23.736357952 +0100
+++ oo_1.1rc3_src/sal/osl/unx/semaphor.c	2003-08-28 10:24:26.260974152 +0100
@@ -79,7 +79,7 @@
 /* osl_createSemaphore  */
 /*****************************************************************************/
 
-#ifdef LINUX
+#if 0
 /* sem_t from glibc-2.1 (/usr/include/semaphore.h) */ 
 
 typedef struct {
@@ -100,7 +100,7 @@
      * larger of both
      * XXX
      */
-#ifdef LINUX
+#if 0
 	if ( sizeof(glibc_21_sem_t) > sizeof(sem_t) )
 		Semaphore= malloc(sizeof(glibc_21_sem_t));
 	else
--- oo_1.1rc3_src/tools/source/solar/solar.c.jj	2003-08-28 11:50:57.589772216 +0100
+++ oo_1.1rc3_src/tools/source/solar/solar.c	2003-08-28 11:50:05.167741576 +0100
@@ -577,7 +577,7 @@
 
   {
 	char* p = NULL;
-	InfoMemoryAccess( p );
+	/*InfoMemoryAccess( p );*/
 	p = (char*)&p;
 	InfoMemoryAccess( p );
 	InfoMemoryTypeAccess( t_short );
--- oo_1.1_src/neon/neon.patch.jj	2003-10-04 10:32:18.341106008 +0100
+++ oo_1.1_src/neon/neon.patch	2003-10-04 10:32:49.123426384 +0100
@@ -19,7 +19,7 @@
 ! 
 ! /* Define as __inline if that's what the C compiler calls it.  */
 ! #ifdef UNX
-! #define inline
+! #define inline inline
 ! #endif
 ! 
 ! /* Define if you need to in order for stat and other things to work.  */
