--- oo_1.0.2_src/dmake/unix/runargv.c.fix-errno	2003-01-07 09:32:42.000000000 -0500
+++ oo_1.0.2_src/dmake/unix/runargv.c	2003-01-24 15:38:58.000000000 -0500
@@ -29,6 +29,9 @@
 #if defined(__CYGWIN__)
 #include <errno.h>
 #endif
+#if HAVE_ERRNO_H
+#include <errno.h>
+#endif
 
 typedef struct prp {
    char *prp_cmd;
@@ -71,7 +74,9 @@ int     shell;
 char	*cmd;
 {
 #if !defined(__CYGWIN__)
+#if !HAVE_ERRNO_H
    extern  int  errno;
+#endif
 #ifndef __APPLE__
 #ifdef arm32
    extern  const char * const sys_errlist[];
--- oo_1.0.2_src/unzip/inc/unzip.h.fix-errno	2003-01-25 01:13:19.000000000 -0500
+++ oo_1.0.2_src/unzip/inc/unzip.h	2003-01-25 01:22:43.000000000 -0500
@@ -134,8 +134,8 @@
 #endif
 #include <ctype.h>       /* skip for VMS, to use tolower() function? */
 #include <errno.h>       /* used in mapname() */
-#ifndef NO_ERRNO
-#  define DECLARE_ERRNO  /* everybody except MSC 6.0, SCO cc, Watcom C/386 */
+#if !defined(NO_ERRNO) && !defined(__GLIBC__)
+#  define DECLARE_ERRNO  /* everybody except MSC 6.0, SCO cc, Watcom C/386, glibc systems  */
 #endif /* !NO_ERRNO */
 #ifdef VMS
 #  include <types.h>     /* (placed up here instead of in VMS section below */
