--- config_office/configure.in.orig	2006-10-22 16:26:15.161119500 +0200
+++ config_office/configure.in	2006-10-22 16:27:57.079489000 +0200
@@ -3639,12 +3639,12 @@
       NSPR_LIB="-L`$PKG_CONFIG --variable=libdir nspr`"
       AC_SUBST(NSPR_LIB)
     else
-      PKG_CHECK_MODULES( MOZ_NSPR, $MOZ_FLAVOUR-nspr )
+      PKG_CHECK_MODULES( MOZ_NSPR, mozilla-nspr )
     fi
 
     PKG_CHECK_MODULES( MOZ_NSS, nss, STANDALONENSS="TRUE", STANDALONENSS="" )
     if test -z "$STANDALONENSS"; then
-        PKG_CHECK_MODULES( MOZ_NSS, $MOZ_FLAVOUR-nss )
+        PKG_CHECK_MODULES( MOZ_NSS, mozilla-nss )
     else
         NSS_LIB="-L`$PKG_CONFIG --variable=libdir nss`"
         AC_SUBST(NSS_LIB)
