--- ooo-build/patches/dev300/apply	2009-12-19 12:50:32.394024392 +0100
+++ ooo-build/patches/dev300/apply.new	2009-12-19 12:50:59.453396860 +0100
@@ -92,7 +92,7 @@
 # Ark Linux
 Ark: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, KDE4
 # www.pld-linux.org
-PLDBase: LinuxCommon, NotDebian, CairoFonts, PLDOnly
+PLDBase: LinuxCommon, NotDebian, CairoFonts, Lockdown, PLDOnly
 PLD: PLDBase
 PLD64: PLDBase
 # Mandriva Linux
@@ -1331,9 +1331,6 @@
 [ PLDOnly ]
 SectionOwner => arekm
 
-# fix for macro browser crash
-sfx2-badscript.diff, i#67976
-
 [ BuildBits ]
 # work around http://gcc.gnu.org/bugzilla/show_bug.cgi?id=22392
 svx-customshapes-NOOPTFILES.diff, martink
