diff -urN padmin-old/source/makefile.mk padmin/source/makefile.mk
--- padmin-old/source/makefile.mk	2003-10-13 02:02:59.000000000 +0200
+++ padmin/source/makefile.mk	2003-10-13 02:25:38.000000000 +0200
@@ -70,6 +70,10 @@
 
 .INCLUDE: settings.mk
 
+.IF "$(FONTCONFIG_FONTS)" != ""
+CFLAGS+=-DFONTCONFIG_FONTS
+.ENDIF
+
 # --- Files --------------------------------------------------------
 
 SRCFILES=padialog.src rtsetup.src
@@ -95,11 +99,13 @@
 	$(SLO)$/newppdlg.obj		\
 	$(SLO)$/prtsetup.obj		\
 	$(SLO)$/prgsbar.obj			\
-	$(SLO)$/fontentry.obj		\
 	$(SLO)$/helper.obj			\
 	$(SLO)$/adddlg.obj			\
 	$(SLO)$/titlectrl.obj
 
+.IF "$(FONTCONFIG_FONTS)" == ""
+SLOFILES+= $(SLO)$/fontentry.obj
+.ENDIF
 
 OBJFILES=\
 	$(OBJ)/pamain.obj
diff -urN padmin-old/source/padialog.cxx padmin/source/padialog.cxx
--- padmin-old/source/padialog.cxx	2003-10-13 02:02:59.000000000 +0200
+++ padmin/source/padialog.cxx	2003-10-13 02:08:07.000000000 +0200
@@ -89,12 +89,19 @@
 #ifndef _TOOLS_COLOR_HXX
 #include <tools/color.hxx>
 #endif
+#ifndef _URLOBJ_HXX
+#include <tools/urlobj.hxx>
+#endif
 #ifndef _PAD_PADIALOG_HRC_
 #include <padialog.hrc>
 #endif
+
+#ifndef FONTCONFIG_FONTS
 #ifndef _PAD_FONTENTRY_HXX_
 #include <fontentry.hxx>
 #endif
+#endif
+
 #ifndef _PAD_HELPER_HXX_
 #include <helper.hxx>
 #endif
@@ -135,7 +142,9 @@
 		m_aStdPB( this, PaResId( RID_PA_BTN_STD ) ),
 		m_aRemPB( this, PaResId( RID_PA_BTN_DEL ) ),
 		m_aPrintersFL( this, PaResId( RID_PA_FL_PRINTERS ) ),
+#ifndef FONTCONFIG_FONTS
 		m_aFontsPB( this, PaResId( RID_PA_BTN_FONTS ) ),
+#endif
 		m_aConfPB( this, PaResId( RID_PA_BTN_CONF ) ),
 		m_aRenamePB( this, PaResId( RID_PA_BTN_RENAME ) ),
 		m_aTestPagePB( this, PaResId( RID_PA_TESTPAGE ) ),
@@ -174,13 +183,15 @@
 	m_aConfPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aRenamePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aTestPagePB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
-	m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aAddPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	m_aDevicesLB.setDelPressedLink( LINK( this, PADialog, DelPressedHdl ) );
 
+#ifndef FONTCONFIG_FONTS
+	m_aFontsPB.SetClickHdl( LINK( this, PADialog, ClickBtnHdl ) );
 	::psp::PrintFontManager& rFontManager( ::psp::PrintFontManager::get() );
 	if( ! rFontManager.checkImportPossible() )
 		m_aFontsPB.Enable( FALSE );
+#endif
 }
 
 PADialog::~PADialog()
@@ -217,11 +228,13 @@
 		PrintTestPage();
 	else if( pButton == &m_aAddPB )
 		AddDevice();
+#ifndef FONTCONFIG_FONTS
 	else if( pButton == &m_aFontsPB )
 	{
 		FontNameDlg aDialog( this );
 		aDialog.Execute();
 	}
+#endif
 
 	return 0;
 }
diff -urN padmin-old/source/padialog.hxx padmin/source/padialog.hxx
--- padmin-old/source/padialog.hxx	2003-10-13 02:02:59.000000000 +0200
+++ padmin/source/padialog.hxx	2003-10-13 02:08:07.000000000 +0200
@@ -118,7 +118,9 @@
 
         FixedLine							m_aSepButtonFL;
 		PushButton							m_aAddPB;
+#ifndef FONTCONFIG_FONTS
 		PushButton							m_aFontsPB;
+#endif
 		CancelButton  						m_aCancelButton;
 
 		String								m_aDefPrt;
