Index: tools/source/rc/resmgr.cxx
===================================================================
RCS file: /cvs/util/tools/source/rc/resmgr.cxx,v
retrieving revision 1.24
diff -u -3 -p -r1.24 resmgr.cxx
--- tools/source/rc/resmgr.cxx	12 Jun 2003 10:31:01 -0000	1.24
+++ tools/source/rc/resmgr.cxx	6 Nov 2003 16:17:19 -0000
@@ -1336,7 +1338,8 @@ ResMgr* ResMgr::CreateResMgr( const sal_
         }
     }
 
-    return NULL;
+    // No resources found - fall back to any language
+    return SearchCreateResMgr(pPrefixName, nType);
 }
 
 // -----------------------------------------------------------------------
