--- config_office/configure.in~	2007-10-02 15:51:45.295762844 +0200
+++ config_office/configure.in	2007-10-02 15:55:50.484834254 +0200
@@ -4443,8 +4443,8 @@
     AC_CHECK_HEADER(hunspell/hunspell.hxx, [ HUNSPELL_CFLAGS=-I/usr/include/hunspell ],
         [AC_MSG_ERROR(hunspell headers not found.)], [])
        ], [])
-    AC_CHECK_LIB(hunspell, main, [],
-           [ AC_MSG_ERROR(hunspell library not found.) ], [])
+    AC_CHECK_LIB(hunspell-1.2, main, [],
+           [ AC_MSG_ERROR(hunspell-1.2 library not found.) ], [])
 else
     AC_MSG_RESULT([internal])
     SYSTEM_HUNSPELL=NO
--- lingucomponent/source/spellcheck/spell/makefile.mk~	2007-02-01 20:08:29.000000000 +0100
+++ lingucomponent/source/spellcheck/spell/makefile.mk	2007-10-02 21:43:10.511251807 +0200
@@ -47,7 +47,7 @@
 
 .IF "$(HUNSPELLLIB)"==""
 .IF "$(GUI)"=="UNX"
-HUNSPELLLIB=-lhunspell
+HUNSPELLLIB=-lhunspell-1.2
 .ENDIF # unx
 .IF "$(GUI)"=="WNT"
 HUNSPELLLIB=hunspell.lib
