Index: vcl/unx/source/gdi/dtint.cxx
===================================================================
RCS file: /cvs/gsl/vcl/unx/source/gdi/dtint.cxx,v
retrieving revision 1.16
retrieving revision 1.16.112.2
diff -u -3 -p -r1.16 -r1.16.112.2
--- vcl/unx/source/gdi/dtint.cxx	28 May 2003 12:33:47 -0000	1.16
+++ vcl/unx/source/gdi/dtint.cxx	10 Nov 2003 10:43:05 -0000	1.16.112.2
@@ -138,6 +138,27 @@ DtIntegrator* DtIntegrator::CreateDtInte
 		if( pIntegrator->mpDisplay == pDisplay )
 			return pIntegrator;
 	}
+    /*
+     *  #i22061# override desktop detection
+     *  if environment variable OOO_FORCE_DESKTOP is set
+     *  to one of "cde" "kde" "gnome" then autodetection
+     *  is overridden.
+     */
+    static const char* pOverride = getenv( "OOO_FORCE_DESKTOP" );
+    if( pOverride && *pOverride )
+    {
+        OString aOver( pOverride );
+#if USE_CDE
+        if( aOver.equalsIgnoreAsciiCase( "cde" ) == 0 )
+            return new CDEIntegrator( pFrame );
+#endif
+        if( aOver.equalsIgnoreAsciiCase( "kde" ) == 0 )
+            return new KDEIntegrator( pFrame );
+        if( aOver.equalsIgnoreAsciiCase( "gnome" ) == 0 )
+            return new GNOMEIntegrator( pFrame );
+        if( aOver.equalsIgnoreAsciiCase( "none" ) == 0 )
+            return new DtIntegrator( pFrame );
+    }
 
 	Atom nDtAtom = None;
 
