Index: bin/setup.in
===================================================================
RCS file: /cvs/gnome/ooo-build/bin/setup.in,v
retrieving revision 1.66
retrieving revision 1.68
diff -u -u -r1.66 -r1.68
--- openoffice/bin/setup.in	3 Jul 2004 02:14:36 -0000	1.66
+++ openoffice/bin/setup.in	7 Jul 2004 17:26:23 -0000	1.68
@@ -72,13 +72,19 @@
 
 else
     if test "z$CONFIGURE_OPTIONS" = "z"; then
-	CONFIGURE_OPTIONS="--with-lang=$LANGS \
-		       --enable-libart --enable-libsn --enable-crashdump=no \
-		       --without-fonts --disable-rpath \
-		       --with-system-zlib --enable-fontconfig \
-		       --with-system-freetype --disable-mozilla \
-		       --with-widget-sets=original,kde,gtk";
+	if [ -s ./setup.configure ]; then
+		. ./setup.configure
+		echo "Using configure options from the ./setup.configure file";
+	else
+		CONFIGURE_OPTIONS="--with-lang=$LANGS \
+			       --enable-libart --enable-libsn --enable-crashdump=no \
+			       --without-fonts --disable-rpath \
+			       --with-system-zlib --enable-fontconfig \
+			       --with-system-freetype --disable-mozilla \
+			       --with-widget-sets=original,kde,gtk";
+	fi
     else
+	echo "CONFIGURE_OPTIONS=\"`echo \"$CONFIGURE_OPTIONS\" | xargs`\"" > ./setup.configure
 	echo "Using configure options from the environment";
     fi
 fi
