--- ooo-build-2.2.0/patches/src680/apply~	2007-03-27 02:54:55.779113930 +0300
+++ ooo-build-2.2.0/patches/src680/apply	2007-03-27 02:54:40.198763000 +0300
@@ -62,7 +62,8 @@
 Ark: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, SystemJavaBits
 Ark64: LinuxCommon, NotDebian, ArkOnly, BerkeleyDB43, 64bit, SystemJavaBits
 # www.pld-linux.org
-PLDBase: LinuxCommon, NotDebian, PLDOnly, NovellOnly, NovellOnlyUnix, FilePickers, GCJExperimental, SystemJavaBits, SystemJavaBitsWorkaround, BerkeleyDB43, STLport5
+PLDBase: LinuxCommon, NotDebian, PLDOnly, SystemJavaBits, BerkeleyDB43, STLport5, \
+	CalcSolver
 PLD: PLDBase
 PLD64: PLDBase, 64bit, PLD64bitfixes
 # Mandriva Linux
