--- configure.in~	2010-02-01 22:04:43.000000000 +0100
+++ configure.in	2010-02-01 22:07:21.307930913 +0100
@@ -6443,7 +6443,7 @@
    fi
 
    dnl Check for Meta Object Compiler
-   AC_PATH_PROG( MOC4, moc, no, [$PATH:$QT4DIR/bin] )
+   AC_PATH_PROG( MOC4, moc, no, [$QT4DIR/bin:$PATH])
    if test "$MOC4" = "no" ; then
       AC_MSG_ERROR([Qt Meta Object Compiler not found.  Please specify
 the root of your Qt installation by exporting QT4DIR before running "configure".])
