Index: vcl/source/window/toolbox.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/toolbox.cxx,v
retrieving revision 1.61
diff -u -p -u -r1.61 toolbox.cxx
--- vcl/source/window/toolbox.cxx	28 May 2003 12:32:32 -0000	1.61
+++ vcl/source/window/toolbox.cxx	15 Sep 2003 10:03:43 -0000
@@ -3045,6 +3044,7 @@ void ToolBox::ImplDrawItem( USHORT nPos,
         {
             DrawSelectionBackground( pItem->maRect, bHighlight, pItem->meState == STATE_CHECK, TRUE, pItem->mpWindow ? TRUE : FALSE );
 
+#ifdef DRAW_1BIT_ALPHA_SHADOWS
             // draw shadows
             if( bHighlight == 2 && pItem->meState != STATE_CHECK )
             {
@@ -3063,17 +3063,15 @@ void ToolBox::ImplDrawItem( USHORT nPos,
                 nTempOffX-=2;
                 nTempOffY-=2;
             }
+#endif
 
             if( bHighlight )
             {
                 if( bHighContrastWhite )
                     nImageStyle |= IMAGE_DRAW_COLORTRANSFORM;
             }
-            DrawImage( Point( nTempOffX, nTempOffY ), *pImage, nImageStyle );
-
         }
-        else
-            DrawImage( Point( nTempOffX, nTempOffY ), *pImage, nImageStyle );
+		DrawImage( Point( nTempOffX, nTempOffY ), *pImage, nImageStyle );
     }
 
     // Text ausgeben
Index: vcl/source/window/window.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/window.cxx,v
retrieving revision 1.176
diff -u -p -u -r1.176 window.cxx
--- vcl/source/window/window.cxx	1 Jul 2003 14:48:37 -0000	1.176
+++ vcl/source/window/window.cxx	15 Sep 2003 10:04:20 -0000
@@ -8583,8 +8588,13 @@ void Window::DrawSelectionBackground( co
     Color aSelectionFillCol( aSelectionBorderCol );
     Color aSelectionMaskCol( aSelectionBorderCol );
 
+#ifdef HORRIBLE_WITH_INDUSTRIAL_THEME
 	BOOL bDark = GetSettings().GetStyleSettings().GetFaceColor().IsDark();
 	BOOL bBright = GetSettings().GetStyleSettings().GetFaceColor().IsBright();
+#else
+	BOOL bDark = FALSE;
+	BOOL bBright = FALSE;
+#endif
 
     int c1 = aSelectionBorderCol.GetLuminance();
     int c2 = GetDisplayBackground().GetColor().GetLuminance();
