Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.82
diff -u -p -u -r1.82 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	17 Jul 2003 13:07:14 -0000	1.82
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	5 Aug 2003 14:05:18 -0000
@@ -5075,9 +5075,9 @@
 					<label xml:lang="x-comment">Capitalize first letter of every sentence</label>
 					<label xml:lang="sk">Každé prvé písmeno vety veľkým</label>
 				</info>
-				<value>true</value>
-			</prop>
-			<prop oor:name="ChangeUnderlineWeight" oor:type="xs:boolean">
+				<value>false</value>
+			</prop> <!-- patch1 -->
+			<prop oor:name="ChangeUnderlineWeight" oor:type="xs:boolean">  <!-- patch2 -->
 				<!-- OldPath: AutoCorrect/Options/All -->
 				<!-- OldLocation: Soffice.cfg -->
 				<!-- UIHints: Tools  AutoCorrect/AutoFormat  Options - Automatic *bold* and _underline_ -->
