--- oo_1.1_src/vcl/source/glyphs/gcach_ftyp.cxx.orig	2004-01-15 14:16:33.000000000 +0100
+++ oo_1.1_src/vcl/source/glyphs/gcach_ftyp.cxx	2004-01-15 18:12:10.921457208 +0100
@@ -77,12 +77,13 @@
 #include <osl/file.hxx>
 #include <osl/thread.hxx>
 
-#include "freetype/freetype.h"
-#include "freetype/ftglyph.h"
-#include "freetype/ftoutln.h"
-#include "freetype/tttables.h"
-#include "freetype/tttags.h"
-#include "freetype/ttnameid.h"
+#include <ft2build.h>
+#include FT_FREETYPE_H
+#include FT_TRUETYPE_TABLES_H
+#include FT_GLYPH_H
+#include FT_OUTLINE_H
+#include FT_TRUETYPE_TAGS_H
+#include FT_TRUETYPE_IDS_H
 
 #ifndef FREETYPE_PATCH
     // VERSION_MINOR in freetype.h is too coarse

--- oo_1.1_src/vcl/source/glyphs/gcach_layout.cxx.orig	2004-01-15 18:17:08.079282352 +0100
+++ oo_1.1_src/vcl/source/glyphs/gcach_layout.cxx	2004-01-15 18:17:49.194031960 +0100
@@ -68,7 +68,8 @@
 #endif
 
 #ifndef _SV_GCACHFTYP_HXX
-#include <freetype/freetype.h>
+#include <ft2build.h>
+#include FT_FREETYPE_H
 #include <gcach_ftyp.hxx>
 #endif
 
