Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.84
diff -u -p -u -r1.84 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	24 Jul 2003 13:12:08 -0000	1.84
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	29 Aug 2003 13:32:53 -0000
@@ -13180,11 +13180,11 @@
 						<label xml:lang="th">Font Antialiasing Minimum Pixel Height</label>
 						<label xml:lang="x-comment">Font Antialiasing Minimum Pixel Height</label>
 					</info>
-					<value>8</value>
+					<value>1</value>
 				</prop>
-			</group>
+			</group>  <!-- dummy -->
 		</group>
-		<group oor:name="Undo">
+		<group oor:name="Undo"> <!-- dummy -->
 			<info>
 				<author>MBA</author>
 				<desc xml:lang="en-US">Contains the settings regarding the undo operation in the Office.</desc>
