diff -urN ooo-build-1.3.2.org/patches/OOO_1_1_2/apply ooo-build-1.3.2/patches/OOO_1_1_2/apply
--- ooo-build-1.3.2.org/patches/OOO_1_1_2/apply	2004-08-29 22:33:55.506404688 +0200
+++ ooo-build-1.3.2/patches/OOO_1_1_2/apply	2004-08-29 22:41:39.287899184 +0200
@@ -864,6 +864,9 @@
 # splash
 pld-splash.diff
 
+#
+gui-disable-spellcheck-all-langs.diff
+
 #######################################################
 [ ArkOnly ]
 
