Index: sw/source/ui/config/viewopt.cxx
===================================================================
RCS file: /cvs/sw/sw/source/ui/config/viewopt.cxx,v
retrieving revision 1.4
diff -u -p -u -r1.4 viewopt.cxx
--- sw/source/ui/config/viewopt.cxx	5 Jul 2001 13:48:54 -0000	1.4
+++ sw/source/ui/config/viewopt.cxx	25 Feb 2003 16:15:51 -0000
@@ -441,6 +441,7 @@ void SwViewOption::PaintPostIts( OutputD
 |*************************************************************************/
 
 SwViewOption::SwViewOption() :
+	sSymbolFont( RTL_CONSTASCII_STRINGPARAM( "symbol" ) ),
 	nZoom( 100 ),
 	nPagePrevRow( 1 ),
 	nPagePrevCol( 2 ),
