--- config_office/configure.in	2007-03-05 03:29:11.059872065 +0200
+++ config_office/configure.in	2007-03-05 03:33:16.395423096 +0200
@@ -4208,7 +4208,7 @@
             AGG_INCDIR="`echo $AGG_CFLAGS | $SED -e s/-I//`"
          fi
          # we could have used pkg-config --modversion but 2.4's libagg.pc.in still contains 2.3 :/
-         if $GREP -q "Version 2.4" `echo $AGG_INCDIR`/agg_basics.h; then
+         if grep -q "Version 2.[[45]]" `echo $AGG_INCDIR`/agg_basics.h; then
             AC_MSG_RESULT([2.4])
             AGG_VERSION=2400
          else
