# CLK_TCK was obsoleted by CLOCKS_PER_SEC
--- oo_641c_src/tools/source/datetime/ttime.cxx.jj	Mon Jun 11 19:17:46 2001
+++ oo_641c_src/tools/source/datetime/ttime.cxx	Mon Jan 14 20:55:07 2002
@@ -493,7 +493,7 @@ ULONG Time::GetSystemTicks()
 
 	if ( !nImplTicksPerSecond )
 	{
-		nImplTicksPerSecond = CLK_TCK;
+		nImplTicksPerSecond = CLOCKS_PER_SEC;
 		dImplTicksPerSecond = nImplTicksPerSecond;
 		dImplTicksULONGMAX	= (double)(ULONG)ULONG_MAX;
 	}
