Index: oo_cvs/offmgr/source/offapp/dialog/opthtml.src
===================================================================
RCS file: /cvs/framework/offmgr/source/offapp/dialog/opthtml.src,v
retrieving revision 1.38
diff -u -3 -p -r1.38 opthtml.src
--- offmgr/source/offapp/dialog/opthtml.src	13 Jun 2003 11:38:08 -0000	1.38
+++ offmgr/source/offapp/dialog/opthtml.src	4 Jul 2003 08:27:15 -0000
@@ -459,7 +459,7 @@ TabPage RID_OFAPAGE_HTMLOPT
 		Text [ spanish ] = "Importar HTML-Tags ~desconocidos como campos" ;
 		Text [ french ] = "Importer les balises HTML inconnues comme champs" ;
 		Text [ dutch ] = "~Onbekende HTML-tags als velden importeren" ;
-		Text [ swedish ] = "~Importera okända HTLM-taggar som fält" ;
+		Text [ swedish ] = "~Importera okända HTML-taggar som fält" ;
 		Text [ danish ] = "Importer ukendte HTML-koder som felter" ;
 		Text [ portuguese ] = "~Importar HTML-Tags desconhecidos como campos" ;
 		Text [ portuguese_brazilian ] = "~Importar Tags HTML desconhecidos como campos" ;
