Index: vcl/source/window/toolbox.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/toolbox.cxx,v
retrieving revision 1.60
diff -u -p -u -r1.60 toolbox.cxx
--- vcl/source/window/toolbox.cxx	17 Apr 2003 15:19:46 -0000	1.60
+++ vcl/source/window/toolbox.cxx	14 May 2003 08:11:12 -0000
@@ -3024,9 +3024,6 @@ void ToolBox::ImplDrawItem( USHORT nPos,
         if ( !pItem->mbEnabled || !IsEnabled() )
             nImageStyle |= IMAGE_DRAW_DISABLE;
 
-        if ( pItem->meState == STATE_DONTKNOW )
-            nImageStyle |= IMAGE_DRAW_DISABLE;
-
 
         // Image ausgeben
         nTempOffX = nOffX;
