http://www.openoffice.org/issues/show_bug.cgi?id=74751

--- ooo-build-ooe680-m6/build/ooe680-m6/config_office/set_soenv.in~	2007-02-21 13:30:12.952600580 +0200
+++ ooo-build-ooe680-m6/build/ooe680-m6/config_office/set_soenv.in	2007-02-25 00:47:11.838336410 +0200
@@ -426,6 +426,7 @@
       $OUTPATH        = "unxlngx6";
       # Blackdown.org JDK porting project uses `amd64' and `server' in JDK 1.4.2 RC1
       $JRELIBDIR      = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."amd64";
+      $JREEXTRALIBDIR = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."amd64".$ds."xawt";
       $JRETOOLKITDIR  = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."amd64".$ds."server";
       $JRETHREADDIR   = '$JAVA_HOME'.$ds."jre".$ds."lib".$ds."amd64".$ds."native_threads";
    }
