--- ooo-build-ooe680-m6/Makefile.am.orig	2006-12-13 18:06:12.351681750 +0100
+++ ooo-build-ooe680-m6/Makefile.am	2006-12-13 18:06:32.908966500 +0100
@@ -2,7 +2,7 @@
 #  MAYBE_BONOBO = bonobo
 # endif
 
-SUBDIRS = po desktop distro-configs bin etc patches man doc fonts stamp src templates www # $(MAYBE_BONOBO)
+SUBDIRS = po desktop distro-configs bin etc patches man doc stamp src templates www # $(MAYBE_BONOBO)
 
 include $(top_srcdir)/Makefile.shared
 
