--- svtools/prj/build.lst.org	2004-03-30 08:42:42.000000000 +0000
+++ svtools/prj/build.lst	2004-03-30 08:43:16.000000000 +0000
@@ -51,4 +51,4 @@
 st	svtools\source\passwordcontainer		nmake	-	all	st_passcont NULL
 st  svtools\source\syslocale                nmake   -   all st_sysloc NULL
 st	svtools\source\filepicker               nmake	-	all	st_filepick st_bmp NULL
-st  svtools\util                            nmake   -   all st_util st__brw st__access st__ctr st__dde st__item st__misc st__srch st__url st_cmdparse st_ctl st_dial st_edit st_file st_html st_papp st_rtf st_sql st_usdde.u st_ueaimp.u st_ufilt.u st_msdde.m st_msc.m st_undo st_uno st_vfilt st_vigif st_vipng st_vixbm st_vixpm st_vjpeg st_vwmf st_passcont st_sysloc st_filepick st_not st_conf st_sbx st_mem NULL
+st  svtools\util                            nmake   -   all st_util st__brw st__access st__ctr st__dde st__item st__misc st__srch st__url st_cmdparse st_ctl st_dial st_edit st_file st_html st_papp st_rtf st_sql st_usdde.u st_ueaimp.u st_ufilt.u st_msdde.m st_msc.m st_undo st_uno st_vfilt st_vigif st_vipng st_vixbm st_vixpm st_vjpeg st_vwmf st_passcont st_sysloc st_filepick st_not st_conf st_sbx NULL
