--- bridges/source/jni_uno/makefile.mk.orig	2003-09-22 12:34:56.000000000 -0700
+++ bridges/source/jni_uno/makefile.mk	2003-09-22 12:36:30.000000000 -0700
@@ -72,6 +72,8 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)"!=""
+
 SLOFILES= \
 	$(SLO)$/jni_info.obj		\
 	$(SLO)$/jni_data.obj		\
@@ -99,6 +101,8 @@
 .ENDIF
 .ENDIF
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :  target.mk
