--- psprint/source/helper/helper.cxx.orig	2003-08-08 08:41:05.000000000 -0700
+++ psprint/source/helper/helper.cxx	2003-08-08 08:43:33.000000000 -0700
@@ -72,7 +72,9 @@
 #include <tools/config.hxx>
 #include <rtl/bootstrap.hxx>
 #include <sal/config.h>
+#ifdef SOLAR_JAVA
 #include "jvmaccess/javainfo.hxx"
+#endif // SOLAR_JAVA
 
 using namespace rtl;
 
@@ -263,6 +265,7 @@
             }
         }
 
+#ifdef SOLAR_JAVA
         // if no javarc (e.g. in setup) exists or it failed try the UDK method
         if( ! aJREpath.getLength() )
         {
@@ -299,6 +302,7 @@
                 }
             }
         }
+#endif // SOLAR_JAVA
 
         if( aJREpath.getLength() )
         {
--- psprint-old/util/makefile.mk	2003-10-13 03:19:55.000000000 +0200
+++ psprint/util/makefile.mk	2003-10-13 03:17:08.000000000 +0200
@@ -95,6 +95,5 @@
 
 SHL1STDLIBS=$(UNOTOOLSLIB)		\
 			$(TOOLSLIB) 		\
-            $(JVMACCESSLIB)		\
 			$(SALLIB)			\
 			-lX11

.IF "$(FONTCONFIG_FONTS)" == "YES"
 SHL1STDLIBS+=`pkg-config --libs fontconfig`
 .ENDIF
+.IF "$(SOLAR_JAVA)" != ""
+SHL1STDLIBS+=$(JVMACCESSLIB)
+.ENDIF
 
 SHL1DEF=	$(MISC)$/$(SHL1TARGET).def
 
