--- oo/starmath/source/register.cxx.orig	Fri May 23 23:41:19 2003
+++ oo/starmath/source/register.cxx	Fri May 23 23:41:57 2003
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -154,7 +154,7 @@
 	RTL_CONSTASCII_USTRINGPARAM("/") ) + SmXMLImport_getImplementationName() +
 	::rtl::OUString(RTL_CONSTASCII_USTRINGPARAM( "/UNO/SERVICES") )  );
 
-	::com::sun::star::uno::Sequence< rtl::OUString > &rServices =
+	const ::com::sun::star::uno::Sequence< rtl::OUString > &rServices =
 		SmXMLImport_getSupportedServiceNames();
 	for(i = 0; i < rServices.getLength(); i++ )
 		xNewKey->createKey( rServices.getConstArray()[i]);
