diff -Nur oo.orig/crashrep/source/unx/makefile.mk oo.chng/crashrep/source/unx/makefile.mk
--- oo.orig/crashrep/source/unx/makefile.mk	Wed Jul 16 16:05:44 2003
+++ oo.chng/crashrep/source/unx/makefile.mk	Wed Jul 30 19:44:14 2003
@@ -115,7 +115,7 @@
 APP2TARGET=$(TARGET2)
 APP2OBJS=$(OBJFILES)
 
-APP2STDLIBS=$(STATIC) `pkg-config --only-mod-libs --libs gtk+-2.0` -lpng -lzlib -ljpeg -ltiff $(DYNAMIC) -lXext -lX11 -ldl -lnsl
+APP2STDLIBS=$(STATIC) `pkg-config --only-mod-libs --libs gtk+-2.0` -lpng -lz -ljpeg -ltiff $(DYNAMIC) -lXext -lX11 -ldl -lnsl
 .IF "$(OS)" == "SOLARIS"
 APP2STDLIBS+=-lsocket
 .ENDIF
diff -Nur oo.orig/package/source/zipapi/Deflater.cxx oo.chng/package/source/zipapi/Deflater.cxx
--- oo.orig/package/source/zipapi/Deflater.cxx	Thu Nov 15 21:16:11 2001
+++ oo.chng/package/source/zipapi/Deflater.cxx	Wed Jul 30 19:42:34 2003
@@ -150,7 +150,7 @@
 		pStream->avail_in  = nLength;
 		pStream->avail_out = nNewLength;
 
-		nResult = z_deflateParams(pStream, nLevel, nStrategy);
+		nResult = deflateParams(pStream, nLevel, nStrategy);
 		switch (nResult)
 		{
 			case Z_OK:
@@ -174,7 +174,7 @@
 		pStream->avail_in  = nLength;
 		pStream->avail_out = nNewLength;
 
-		nResult = z_deflate(pStream, bFinish ? Z_FINISH : Z_NO_FLUSH);
+		nResult = deflate(pStream, bFinish ? Z_FINISH : Z_NO_FLUSH);
 		switch (nResult)
 		{
 			case Z_STREAM_END:
@@ -220,7 +220,7 @@
 	{
 		// do error handling
 	}
-	sal_Int32 nResult = z_deflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray()+nOffset, nLength);
+	sal_Int32 nResult = deflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray()+nOffset, nLength);
 }
 void SAL_CALL Deflater::setDictionary( const uno::Sequence< sal_Int8 >& rBuffer ) 
 {
@@ -230,7 +230,7 @@
 		VOS_DEBUG_ONLY("No stream!");
 		
 	}
-	sal_Int32 nResult = z_deflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray(), rBuffer.getLength());
+	sal_Int32 nResult = deflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray(), rBuffer.getLength());
 }
 void SAL_CALL Deflater::setStrategy( sal_Int32 nNewStrategy ) 
 {
@@ -295,7 +295,7 @@
 }
 void SAL_CALL Deflater::reset(  ) 
 {
-	z_deflateReset(pStream);
+	deflateReset(pStream);
 	bFinish = sal_False;
 	bFinished = sal_False;
 	nOffset = nLength = 0;
@@ -304,7 +304,7 @@
 {
 	if (pStream != NULL)
 	{
-		z_deflateEnd(pStream);
+		deflateEnd(pStream);
 		delete pStream;
 	}
 	pStream = NULL;
diff -Nur oo.orig/package/source/zipapi/Inflater.cxx oo.chng/package/source/zipapi/Inflater.cxx
--- oo.orig/package/source/zipapi/Inflater.cxx	Thu Nov 15 21:17:02 2001
+++ oo.chng/package/source/zipapi/Inflater.cxx	Wed Jul 30 19:43:20 2003
@@ -133,7 +133,7 @@
 	{
 		// do error handling
 	}
-	z_inflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray() + nNewOffset, 
+	inflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray() + nNewOffset, 
 				  nNewLength);
 }
 
@@ -143,7 +143,7 @@
 	{
 		// do error handling
 	}
-	z_inflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray(), 
+	inflateSetDictionary(pStream, (const unsigned char*)rBuffer.getConstArray(), 
 				   rBuffer.getLength());
 }
 
@@ -202,7 +202,7 @@
 
 void SAL_CALL Inflater::reset(  ) 
 {
-	z_inflateReset(pStream);
+	inflateReset(pStream);
 	bFinish = bNeedDict = bFinished = sal_False;
 	nOffset = nLength = 0;
 }
@@ -211,7 +211,7 @@
 {
 	if (pStream != NULL)
 	{
-		z_inflateEnd(pStream);
+		inflateEnd(pStream);
 		delete pStream;
 	}
 	pStream = NULL;
@@ -225,7 +225,7 @@
 	pStream->next_out  = reinterpret_cast < unsigned char* > ( rBuffer.getArray() + nNewOffset );
 	pStream->avail_out = nNewLength;
 
-	nResult = ::z_inflate(pStream, bFinish ? Z_SYNC_FLUSH : Z_PARTIAL_FLUSH);
+	nResult = ::inflate(pStream, bFinish ? Z_SYNC_FLUSH : Z_PARTIAL_FLUSH);
 
 	switch (nResult)
 	{
diff -Nur oo.orig/solenv/inc/libs.mk oo.chng/solenv/inc/libs.mk
--- oo.orig/solenv/inc/libs.mk	Wed Jul 30 19:35:35 2003
+++ oo.chng/solenv/inc/libs.mk	Wed Jul 30 19:41:03 2003
@@ -173,7 +173,7 @@
 LNGLIB=-llng$(OFFICEUPD)$(DLLPOSTFIX)
 EXPAT3RDLIB=-lexpat_xmlparse -lexpat_xmltok
 EXPATASCII3RDLIB=-lascii_expat_xmlparse -lexpat_xmltok
-ZLIB3RDLIB=-lzlib
+ZLIB3RDLIB=-lz
 JPEG3RDLIB=-ljpeglib
 NEON3RDLIB=-lneon
 BERKELEYLIB=-ldb-4.1
diff -Nur oo.orig/zlib/prj/d.lst oo.chng/zlib/prj/d.lst
--- oo.orig/zlib/prj/d.lst	Mon Aug 26 12:00:28 2002
+++ oo.chng/zlib/prj/d.lst	Wed Jul 30 19:40:21 2003
@@ -1,10 +1,10 @@
 mkdir: %_DEST%\inc%_EXT%\external\zlib
 
-..\%__SRC%\inc\zlib.h %_DEST%\inc%_EXT%\external\zlib\zlib.h
-..\%__SRC%\inc\unzip.h %_DEST%\inc%_EXT%\external\zlib\unzip.h
-..\%__SRC%\inc\zconf.h %_DEST%\inc%_EXT%\external\zconf.h
+\usr\include\zlib.h %_DEST%\inc%_EXT%\external\zlib\zlib.h
+\usr\include\unzip.h %_DEST%\inc%_EXT%\external\zlib\unzip.h
+\usr\include\zconf.h %_DEST%\inc%_EXT%\external\zconf.h
 
-..\%__SRC%\lib\libzlib.a %_DEST%\lib%_EXT%\libzlib.a
+\lib\libz.so* %_DEST%\lib%_EXT%\libz.so*
 ..\%__SRC%\slb\zlib.lib %_DEST%\lib%_EXT%\zlib.lib
 ..\%__SRC%\slb\zlib.lin %_DEST%\lib%_EXT%\zlib.lin
 
