--- oo_1.0_src/config_office/configure.in~~	Thu May  2 16:48:20 2002
+++ oo_1.0_src/config_office/configure.in	Thu May  2 16:50:49 2002
@@ -975,10 +975,10 @@
 dnl ***************************************
 dnl testing for X libraries and includes...
 dnl ***************************************
-if test "$_os" = "Darwin" -a "x$x_includes" = "xno_x_includes"; then
-   dnl Mac OS X using Aqua graphics. Don't check for X11.
-   :
-elif test "$_os" != "WINNT" ; then
+dnl if test "$_os" = "Darwin" -a "x$x_includes" = "xno_x_includes"; then
+dnl    dnl Mac OS X using Aqua graphics. Don't check for X11.
+dnl    :
+dnl elif test "$_os" != "WINNT" ; then
    AC_PATH_X
    AC_PATH_XTRA
 
