Index: sfx2/source/view/prnmon.cxx
===================================================================
RCS file: /cvs/framework/sfx2/source/view/prnmon.cxx,v
retrieving revision 1.8
diff -u -p -u -r1.8 prnmon.cxx
--- sfx2/source/view/prnmon.cxx	4 Apr 2003 17:40:22 -0000	1.8
+++ sfx2/source/view/prnmon.cxx	7 Aug 2003 14:38:41 -0000
@@ -384,7 +384,7 @@ IMPL_LINK_INLINE_START( SfxPrintProgress
 {
 	if ( pImp->pMonitor )
 		pImp->pMonitor->Hide();
-	pImp->pPrinter->AbortJob();
+	pImp->pPrinter->SetErrorHdl( Link() );
 	InfoBox( pImp->GetViewShell()->GetWindow(),
 			 String( SfxResId(STR_ERROR_PRINT) ) ).Execute();
     if ( pImp->bRestoreFlag && pImp->pViewShell->GetObjectShell()->IsEnableSetModified() != pImp->bOldFlag )
Index: vcl/source/gdi/impprn.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/gdi/impprn.cxx,v
retrieving revision 1.5
diff -u -p -u -r1.5 impprn.cxx
--- vcl/source/gdi/impprn.cxx	27 Mar 2003 17:57:57 -0000	1.5
+++ vcl/source/gdi/impprn.cxx	7 Aug 2003 14:39:09 -0000
@@ -350,7 +350,8 @@ IMPL_LINK( ImplQPrinter, ImplPrintHdl, T
 	{
 		maTimer.Stop();
 		delete pActPage;
-		EndJob();
+		if (!EndJob())
+			mpParent->Error();
 		mpParent->ImplEndPrint();
 	}
 	else
