--- readlicense_oo/util/makefile.pmk.orig	2003-09-22 11:36:10.000000000 -0700
+++ readlicense_oo/util/makefile.pmk	2003-09-22 11:37:57.000000000 -0700
@@ -73,7 +73,36 @@
 
 $(MISC)$/%.html :
     @+-$(MKDIR) $(@:d)
+.IF "$(SOLAR_JAVA)"!=""
 	$(JAVA) -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $(subst,$(@::d:d:b), $(@:b).xrm) ..$/readme.xsl os1=$(OS) gui1=$(GUI) cp1=$(CPUNAME) com1=$(COM) lang1=$(iso_$(@:d:d:b)) type=html file=$@
+.ELSE
+	$(SED) 's|xmlns:xt="http://www.jclark.com/xt"||;s|extension-element-prefixes="xt"||;s|.*xt:document.*||;s|<xsl:output method = "HTML" doctype-public = "-//W3C//DTD HTML 3.2//EN" />|<xsl:output method="html" doctype-public="-//W3C//DTD HTML 3.2//EN" />|' < ..$/readme.xsl > ..$/readme2.xsl
+	$(XSLTPROC) -o $@ \
+	            --stringparam os1 $(OS) \
+	            --stringparam gui1 $(GUI) \
+	            --stringparam cp1 $(CPUNAME) \
+	            --stringparam com1 $(COM) \
+	            --stringparam lang1 $(iso_$(@:d:d:b)) \
+	            --stringparam type html \
+	            ..$/readme2.xsl \
+	            $(subst,$(@::d:d:b), $(@:b).xrm)
+	rm ..$/readme2.xsl
+.ENDIF
+
 $(MISC)$/%.txt :
     @+-$(MKDIR) $(@:d)
+.IF "$(SOLAR_JAVA)"!=""
 	$(JAVA) -classpath $(COMMON_BUILD_TOOLS)$/xt.jar$(PATH_SEPERATOR)$(COMMON_BUILD_TOOLS)$/parser.jar -Dcom.jclark.xsl.sax.parser=com.sun.xml.parser.Parser com.jclark.xsl.sax.Driver $(subst,$(@::d:d:b), $(@:b).xrm) ..$/readme.xsl os1=$(OS) gui1=$(GUI) cp1=$(CPUNAME) com1=$(COM) lang1=$(iso_$(@:d:d:b)) type=text file=$@
+.ELSE
+	$(SED) 's|xmlns:xt="http://www.jclark.com/xt"||;s|extension-element-prefixes="xt"||;s|.*xt:document.*||;s|<xsl:output method = "HTML" doctype-public = "-//W3C//DTD HTML 3.2//EN" />|<xsl:output method="text" />|' < ..$/readme.xsl > ..$/readme2.xsl
+	$(XSLTPROC) -o $@ \
+	            --stringparam os1 $(OS) \
+	            --stringparam gui1 $(GUI) \
+	            --stringparam cp1 $(CPUNAME) \
+	            --stringparam com1 $(COM) \
+	            --stringparam lang1 $(iso_$(@:d:d:b)) \
+	            --stringparam type text \
+	            ..$/readme2.xsl \
+	            $(subst,$(@::d:d:b), $(@:b).xrm)
+	rm ..$/readme2.xsl
+.ENDIF
