Index: officecfg/registry/schema/org/openoffice/Office/Common.xcs
===================================================================
RCS file: /cvs/util/officecfg/registry/schema/org/openoffice/Office/Common.xcs,v
retrieving revision 1.84
diff -u -p -u -r1.84 Common.xcs
--- officecfg/registry/schema/org/openoffice/Office/Common.xcs	24 Jul 2003 13:12:08 -0000	1.84
+++ officecfg/registry/schema/org/openoffice/Office/Common.xcs	29 Aug 2003 13:41:32 -0000
@@ -26662,9 +26662,9 @@
 					<label xml:lang="th">Use system default user interface font</label>
 					<label xml:lang="x-comment">Use system default user interface font</label>
 				</info>
-				<value>false</value>
+				<value>true</value>
 			</prop>
-			<prop oor:name="IsSelectionInReadonly" oor:type="xs:boolean">
+			<prop oor:name="IsSelectionInReadonly" oor:type="xs:boolean"> <!-- dummy -->
 				<info>
 					<author>OS</author>
 					<desc xml:lang="en-US">Indicates whether the cursor is displayed in read-only texts.</desc>
