Index: sal/osl/unx/thread.c
===================================================================
RCS file: /cvs/oo/porting/sal/osl/unx/thread.c,v
retrieving revision 1.21.8.1
diff -u -p -u -r1.21.8.1 thread.c
--- sal/osl/unx/thread.c	25 Nov 2002 12:11:18 -0000	1.21.8.1
+++ sal/osl/unx/thread.c	29 Apr 2003 12:16:53 -0000
@@ -598,7 +598,9 @@ void SAL_CALL osl_waitThread(const TimeV
 /*****************************************************************************/
 void SAL_CALL osl_yieldThread()
 {
-    sched_yield();
+#ifndef LINUX
+	sched_yield();
+#endif
 }
 
 /*****************************************************************************/
