--- ooo-build-1.3.0/build/OOO_1_1_2/sal/osl/unx/process.c.orig	2003-07-02 15:34:17.000000000 +0200
+++ ooo-build-1.3.0/build/OOO_1_1_2/sal/osl/unx/process.c	2004-07-30 09:04:58.196672392 +0200
@@ -1017,11 +1017,6 @@
 			for (i = 0; data.m_pszEnv[i] != NULL; i++)
 	     		putenv(data.m_pszEnv[i]);
 
-#if defined(LINUX)
-            /* mfe: linux likes to have just one thread when the exec family is called */
-            /*      this np function has this purpose ...                              */
-            pthread_kill_other_threads_np();
-#endif
             OSL_TRACE("ChildStatusProc : starting '%s'",data.m_pszArgs[0]);
      
 			/* Connect std IO to pipe ends */
