Index: vcl/source/window/toolbox.cxx
===================================================================
RCS file: /cvs/gsl/vcl/source/window/toolbox.cxx,v
retrieving revision 1.60
diff -u -p -u -r1.60 toolbox.cxx
--- vcl/source/window/toolbox.cxx	17 Apr 2003 15:19:46 -0000	1.60
+++ vcl/source/window/toolbox.cxx	13 May 2003 12:16:02 -0000
@@ -2795,27 +2795,29 @@ void ToolBox::ImplDrawItem( USHORT nPos,
             pTempItem = &mpData->m_aItems[nPos+1];
             if ( !pTempItem->mbShowWindow && !pTempItem->mbBreak )
             {
-                long nCenterPos;
+                long nCenterPos, nSlim;
                 SetLineColor( rStyleSettings.GetShadowColor() );
                 if ( IsHorizontal() )
                 {
+		    nSlim = (pItem->maRect.Bottom() - pItem->maRect.Top ()) / 4;
                     nCenterPos = pItem->maRect.Center().X()-1;
-                    DrawLine( Point( nCenterPos, pItem->maRect.Top() ),
-                              Point( nCenterPos, pItem->maRect.Bottom() ) );
+                    DrawLine( Point( nCenterPos, pItem->maRect.Top() + nSlim - 1),
+                              Point( nCenterPos, pItem->maRect.Bottom() - nSlim - 1) );
                     nCenterPos++;
                     SetLineColor( rStyleSettings.GetLightColor() );
-                    DrawLine( Point( nCenterPos, pItem->maRect.Top() ),
-                              Point( nCenterPos, pItem->maRect.Bottom() ) );
+                    DrawLine( Point( nCenterPos, pItem->maRect.Top() + nSlim),
+                              Point( nCenterPos, pItem->maRect.Bottom() - nSlim) );
                 }
                 else
                 {
+		    nSlim = (pItem->maRect.Right() - pItem->maRect.Left ()) / 4;
                     nCenterPos = pItem->maRect.Center().Y()-1;
-                    DrawLine( Point( pItem->maRect.Left(), nCenterPos ),
-                              Point( pItem->maRect.Right(), nCenterPos ) );
+                    DrawLine( Point( pItem->maRect.Left() + nSlim, nCenterPos),
+                              Point( pItem->maRect.Right() - nSlim, nCenterPos) );
                     nCenterPos++;
                     SetLineColor( rStyleSettings.GetLightColor() );
-                    DrawLine( Point( pItem->maRect.Left(), nCenterPos ),
-                              Point( pItem->maRect.Right(), nCenterPos ) );
+                    DrawLine( Point( pItem->maRect.Left() + nSlim, nCenterPos ),
+                              Point( pItem->maRect.Right() - nSlim, nCenterPos ) );
                 }
             }
         }
@@ -3152,14 +3154,11 @@ void ToolBox::ImplDrawItem( USHORT nPos,
     {
 
         Point aArrowPos( nOffX, nOffY );
-        // shadows
-        if( bHighlight == 2 )
-        {
-            aArrowPos.X() -= 2;
-            aArrowPos.Y() -= 2;
-        }
+#define ARROW_SIZE_HIGHLIGHT 8
+#define ARROW_SIZE_NO_HIGHLIGHT 6
 
-        aArrowPos.X() += nBtnWidth-6;
+	int arrow_size = bHighlight ? ARROW_SIZE_HIGHLIGHT : ARROW_SIZE_NO_HIGHLIGHT;
+        aArrowPos.X() += nBtnWidth - arrow_size;
 
         Color       aOldLineColor = GetLineColor();
         Color       aOldFillColor = GetFillColor();
@@ -3200,10 +3199,7 @@ void ToolBox::ImplDrawItem( USHORT nPos,
         if ( !pItem->mbEnabled || !IsEnabled() )
             SetFillColor( rStyleSettings.GetShadowColor() );
         else
-        {
-            SetFillColor( COL_LIGHTGREEN );
-            bBlack = TRUE;
-        }
+            SetFillColor( rStyleSettings.GetMenuHighlightColor() );
 
         SetClipRegion( aOldRegion );
 
