diff -uNr ooold/udm/inc/udm/tt_struct.hxx oonew/udm/inc/udm/tt_struct.hxx
--- ooold/udm/inc/udm/tt_struct.hxx	Tue May 14 10:29:34 2002
+++ oonew/udm/inc/udm/tt_struct.hxx	Thu Aug  1 17:33:50 2002
@@ -234,11 +234,11 @@
 };
 
 
-template <class SRC, class MEMBER_TYPE>
-TMemberInfo<SRC,MEMBER_TYPE>::TMemberInfo( ptr2member i_mpOriginalData )
-    :   mpOriginalData(i_mpOriginalData)
-{
-}
+//template <class SRC, class MEMBER_TYPE>
+//TMemberInfo<SRC,MEMBER_TYPE>::TMemberInfo( ptr2member i_mpOriginalData )
+//    :   mpOriginalData(i_mpOriginalData)
+//{
+//}
 
 template <class SRC, class MEMBER_TYPE>
 TMemberInfo<SRC,MEMBER_TYPE>::~TMemberInfo()
