--- oo_1.0_src/vcl/source/glyphs/gcach_ftyp.cxx.ft21	2002-06-04 10:35:42.000000000 +0200
+++ oo_1.0_src/vcl/source/glyphs/gcach_ftyp.cxx	2002-06-04 10:46:33.000000000 +0200
@@ -108,6 +108,13 @@
 #include <settings.hxx>
 #include <tools/lang.hxx>
 
+#if defined(FT_NEXT_SHORT) && !defined(NEXT_Short)
+// Freetype 2.1 isn't fully compatible with the *internal(!)* API used by OOo
+#define NEXT_Short(x)	FT_NEXT_SHORT(x)
+#define NEXT_UShort(x)	FT_NEXT_USHORT(x)
+#define NEXT_Long(x)	FT_NEXT_LONG(x)
+#endif
+
 // Glyph Flags
 #define GF_NONE     0
 #define GF_ROTMASK  3
