--- oo_1.0.1_src/setup2/jsnative/nunxos.cxx.so-ooo-coexist	2001-10-10 18:36:11.000000000 +0200
+++ oo_1.0.1_src/setup2/jsnative/nunxos.cxx	2002-08-22 21:54:44.000000000 +0200
@@ -168,7 +168,7 @@ void UnxOS::Terminate()
 void UnxOS::GetSVersion( char* buf, int len )
 {
 	strcpy( buf, getenv("HOME") );
-	strcat( buf, "/.sversionrc" );
+	strcat( buf, "/.OOoversionrc" );
 }
 
 //DONE
--- oo_1.0.1_src/setup2/mow/source/loader/loader.c.so-ooo-coexist	2002-08-21 21:18:24.000000000 +0200
+++ oo_1.0.1_src/setup2/mow/source/loader/loader.c	2002-08-22 22:01:16.000000000 +0200
@@ -152,7 +152,7 @@ typedef const char* (*FncFindArch)(const
 
 #define PATCH_FILE          "patch.inf"
 #define INSTDBINF_FILE      "instdb.ins"
-#define SVERSION_FILE       ".sversionrc"
+#define SVERSION_FILE       ".OOoversionrc"
 
 BOOL m_bNoJunk = FALSE;
 BOOL m_bUseX = TRUE;
--- oo_1.0.1_src/setup2/script/linux/1.0.0/webinstdb.inf.so-ooo-coexist	2002-08-21 21:18:24.000000000 +0200
+++ oo_1.0.1_src/setup2/script/linux/1.0.0/webinstdb.inf	2002-08-22 21:54:44.000000000 +0200
@@ -78,11 +78,11 @@ End
 
 HelpText gid_Help_WrongSetup
 	ID			 = "";
-	Text		 = "The application is already installed on your system (a sversion.ini or an sofficerc file with a matching entry has been found).";
+	Text		 = "The application is already installed on your system (a OOoversion.ini or an sofficerc file with a matching entry has been found).";
 	ID (49)		 = "";
-	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei sversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
+	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei OOoversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
 	ID (33)		 = "";
-	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier sversion.ini ou sofficerc a été détectée).";
+	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier OOoversion.ini ou sofficerc a été détectée).";
 End
 
 HelpText gid_Help_Migration
@@ -8431,7 +8431,7 @@ End
 
 Profile gid_Profile_Sversion_Ini
 	ModuleID	 = gid_Module_Root;
-	Name		 = ".sversionrc";
+	Name		 = ".OOoversionrc";
 	Dir			 = PREDEFINED_CONFIGDIR;
 End
 
@@ -12383,7 +12383,7 @@ ProfileItem gid_Profileitem_Bootstrap_Lo
 	ProfileID	 = gid_Profile_Bootstrap_Ini;
 	Section		 = "Bootstrap";
 	Key			 = "Location";
-	Value		 = "$(SYSUSERCONFIG)/.sversionrc";
+	Value		 = "$(SYSUSERCONFIG)/.OOoversionrc";
 	Order		 = 2;
 End
 
--- oo_1.0.1_src/setup2/script/linux/1.0.1/webinstdb.inf.so-ooo-coexist	2002-08-21 21:18:24.000000000 +0200
+++ oo_1.0.1_src/setup2/script/linux/1.0.1/webinstdb.inf	2002-08-22 21:54:44.000000000 +0200
@@ -74,11 +74,11 @@ End
 
 HelpText gid_Help_WrongSetup
 	ID			 = "";
-	Text		 = "The application is already installed on your system (a sversion.ini or an sofficerc file with a matching entry has been found).";
+	Text		 = "The application is already installed on your system (a OOoversion.ini or an sofficerc file with a matching entry has been found).";
 	ID (49)		 = "";
-	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei sversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
+	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei OOoversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
 	ID (33)		 = "";
-	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier sversion.ini ou sofficerc a été détectée).";
+	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier OOoversion.ini ou sofficerc a été détectée).";
 End
 
 HelpText gid_Help_Migration
@@ -1986,7 +1986,7 @@ End
 
 Profile gid_Profile_Sversion_Ini
 	ModuleID	 = gid_Module_Root;
-	Name		 = ".sversionrc";
+	Name		 = ".OOoversionrc";
 	Dir			 = PREDEFINED_CONFIGDIR;
 End
 
@@ -12055,7 +12055,7 @@ ProfileItem gid_Profileitem_Bootstrap_Lo
 	ProfileID	 = gid_Profile_Bootstrap_Ini;
 	Section		 = "Bootstrap";
 	Key			 = "Location";
-	Value		 = "$(SYSUSERCONFIG)/.sversionrc";
+	Value		 = "$(SYSUSERCONFIG)/.OOoversionrc";
 	Order		 = 2;
 End
 
--- oo_1.0.1_src/setup2/script/linux/1.1.1/webinstdb.inf.so-ooo-coexist	2002-08-21 21:18:24.000000000 +0200
+++ oo_1.0.1_src/setup2/script/linux/1.1.1/webinstdb.inf	2002-08-22 21:54:44.000000000 +0200
@@ -74,11 +74,11 @@ End
 
 HelpText gid_Help_WrongSetup
 	ID			 = "PAGE_WRONGSETUP";
-	Text		 = "The application is already installed on your system (a sversion.ini or an sofficerc file with a matching entry has been found).";
+	Text		 = "The application is already installed on your system (a Oooversion.ini or an sofficerc file with a matching entry has been found).";
 	ID (49)		 = "";
-	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei sversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
+	Text (49)	 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei OOoversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden).";
 	ID (33)		 = "";
-	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier sversion.ini ou sofficerc a été détectée).";
+	Text (33)	 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier OOoversion.ini ou sofficerc a été détectée).";
 End
 
 HelpText gid_Help_Migration
@@ -2025,7 +2025,7 @@ End
 
 Profile gid_Profile_Sversion_Ini
 	ModuleID	 = gid_Module_Root;
-	Name		 = ".sversionrc";
+	Name		 = ".OOoversionrc";
 	Dir			 = PREDEFINED_CONFIGDIR;
 End
 
@@ -13854,7 +13854,7 @@ ProfileItem gid_Profileitem_Bootstrap_Lo
 	ProfileID	 = gid_Profile_Bootstrap_Ini;
 	Section		 = "Bootstrap";
 	Key			 = "Location";
-	Value		 = "$(SYSUSERCONFIG)/.sversionrc";
+	Value		 = "$(SYSUSERCONFIG)/.OOoversionrc";
 	Order		 = 2;
 End
 
--- oo_1.0.1_src/setup2/source/custom/jvmsetup/loader/jvmapp.cxx.so-ooo-coexist	2001-12-17 13:39:27.000000000 +0100
+++ oo_1.0.1_src/setup2/source/custom/jvmsetup/loader/jvmapp.cxx	2002-08-22 21:54:44.000000000 +0200
@@ -185,9 +185,9 @@ MySiCustomEnvironment::MySiCustomEnviron
 
 			SiDirEntry aVersionfile(oStrConfigDir);
 #ifdef UNX
-			aVersionfile += UniString::CreateFromAscii(".sversionrc");
+			aVersionfile += UniString::CreateFromAscii(".OOoversionrc");
 #else
-			aVersionfile += UniString::CreateFromAscii("sversion.ini");
+			aVersionfile += UniString::CreateFromAscii("OOoversion.ini");
 #endif
 			
 			Config aVCfg( aVersionfile.GetFullUni() );
--- oo_1.0.1_src/setup2/wrapper/install.sh.so-ooo-coexist	2002-06-07 16:02:05.000000000 +0200
+++ oo_1.0.1_src/setup2/wrapper/install.sh	2002-08-22 21:54:44.000000000 +0200
@@ -82,7 +82,7 @@ fi
 if test "$installtype" = INSTALL_NORMAL; then
 
 # Remove previous .sversionrc to prevent problems with previous versions
-oo_versionrc="$HOME/.sversionrc"
+oo_versionrc="$HOME/.OOoversionrc"
 #if [ -f "$oo_versionrc" ]; then
 #  rm -f $oo_versionrc
 #fi
--- oo_1.0.1_src/scp/source/office/files.scp.so-ooo-coexist	2002-08-21 21:18:24.000000000 +0200
+++ oo_1.0.1_src/scp/source/office/files.scp	2002-08-22 21:54:44.000000000 +0200
@@ -1301,14 +1301,14 @@ End
 
 File GID_FILE_USER_RDB
 	TXT_FILE_BODY;
-    Name            = "user60.rdb";
+    Name            = "userOOo10.rdb";
     Dir             = PREDEFINED_CONFIGDIR;
     Styles          = (DELETE_ONLY);
 End
 
 File GID_FILE_PUNKT_USER_RDB
 	TXT_FILE_BODY;
-    Name            = ".user60.rdb";
+    Name            = ".userOOo10.rdb";
     Dir             = PREDEFINED_CONFIGDIR;
     Styles          = (DELETE_ONLY);
 End
--- oo_1.0.1_src/scp/source/office/profile.scp.so-ooo-coexist	2002-06-20 12:22:20.000000000 +0200
+++ oo_1.0.1_src/scp/source/office/profile.scp	2002-08-22 21:54:44.000000000 +0200
@@ -15,9 +15,9 @@ End
 Profile GID_PROFILE_SVERSION_INI
 	ModuleID = GID_MODULE_ROOT;
 #ifdef UNX
-	Name = ".sversionrc";
+	Name = ".OOoversionrc";
 #else
-	Name = "sversion.ini";
+	Name = "OOoversion.ini";
 #endif
 	Dir = PREDEFINED_CONFIGDIR;
 End
@@ -121,9 +121,9 @@ End
     Order = 2;
     Key  = "Location";
 #ifdef WNT
-    Value  = "$SYSUSERCONFIG/sversion.ini";
+    Value  = "$SYSUSERCONFIG/OOoversion.ini";
 #else
-    Value  = "$SYSUSERCONFIG/.sversionrc";
+    Value  = "$SYSUSERCONFIG/.OOoversionrc";
 #endif
 End
 
@@ -276,9 +276,9 @@ ProfileItem GID_PROFILEITEM_SOFFICE_UNO_
     Order = 4;
     Key  = "UNO_WRITERDB";
 #ifdef WNT
-	Value = "$SYSUSERCONFIG/user60.rdb";
+	Value = "$SYSUSERCONFIG/userOOo10.rdb";
 #else
-    Value = "$SYSUSERCONFIG/.user60.rdb";
+    Value = "$SYSUSERCONFIG/.userOOo10.rdb";
 #endif
 End
 #endif
--- oo_1.0.1_src/scp/source/office/setuphelp.lng.so-ooo-coexist	2002-08-21 21:18:25.000000000 +0200
+++ oo_1.0.1_src/scp/source/office/setuphelp.lng	2002-08-22 21:54:44.000000000 +0200
@@ -166,27 +166,27 @@
 96 = "ãÑÍÈÇğ Èß İí ÈÑäÇãÌ ÇáÊËÈíÊ  \n\nááÍÕæá Úáì ÇáãÒíÏ ãä ÇáãÚáæãÇÊ¡ ÇáÑÌÇÁ ÇáÑÌæÚ Åáì Ïáíá ÇáÊËÈíÊ \n\nİí ãÑÈÚÇÊ ÇáÍæÇÑ ÇáãÎÊáİÉ áÈÑäÇãÌ ÇáÅÚÏÇÏ Óæİ ÊÌÏ ÇáãÒíÏ ãä ãİÇÊíÍ ÇáãÓÇÚÏÉ. ÈäŞÑÉ æÇÍÏÉ İæŞ Ãí ãäåÇ íãßäß ÇáÍÕæá Úáì ãÓÇÚÏÉ ááÅÚÏÇÏÇÊ ÇáÎÇÕÉ ÈåĞÇ ÇáÌÒÁ ãä ÇáÊËÈíÊ."
 
 [STR_HELP_WRONGSETUP]
-01 = "The application is already installed on your system (a sversion.ini or an sofficerc file with a matching entry has been found)."
-49 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei sversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden)."
-07 = "İòî ïğèëîæåíèå  óæå ñóùåñòâóåò â Âàøåé îïåğàöèîííîé ñèñòåìå (îáíàğóæåí ôàéë sversion.ini èëè sofficerc)."
-55 = "Das %PRODUCTNAME Scheduleserver %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm hat eine bestehende Installation erkannt.\n\nHaben Sie eine alte Installation eventuell nicht ?er das %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm deinstalliert? Dann l?chen Sie bitte jetzt Ihr altes %PRODUCTNAME Scheduleserver-Verzeichnis (Ihre eventuell dort enthaltenen Dokumente brauchen Sie nicht zu l?chen). Dann l?chen Sie - je nach Betriebssystem - die Datei Sversion.ini im Windows oder OS2 Verzeichnis oder .sversionrc oder sversion.prefs. Versuchen Sie nun neu, das %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm zu starten."
-37 = "L'aplicació ja està instal·lada al sistema (s'ha trobat un fitxer sversion.ini o sofficerc amb una entrada que concideix)."
-03 = "A aplicação já se encontra instalada no seu sistema operativo (foi encontrado um ficheiro sversion.ini ou sofficerc com o respectivo registo)."
-30 = "Óôï óıóôçìÜ óáò õğÜñ÷åé Şäç ìéá İêäïóç ôçò åöáñìïãŞò (âñİèçêå ôï áñ÷åßï sversion.ini Ş sofficerc ìå ôçí áíôßóôïé÷ç åããñáöŞ)."
-31 = "Het programma is reeds in uw systeem geïnstalleerd (er werd een bestand gevonden met de naam sversion.ini resp. sofficerc met een dergelijk item)."
-33 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier sversion.ini ou sofficerc a été détectée)."
-34 = "La aplicación ya está instalada en su sistema (se ha encontrado un archivo sversion.ini o sofficerc con la respectiva entrada)."
-35 = "Sovellus on jo asennettu järjestelmään (asennusohjelma löysi sversion.ini- tai sofficerc-tiedoston, jossa on asennettavaa ohjelmaa vastaavat merkinnät)."
-39 = "L'applicazione è già installata nel vostro sistema (è stato trovato un file sversion.ini o officerc contenete l'indicazione corrispondente)."
-45 = "Der findes allerede en version af %PRODUCTNAME %PRODUCTVERSION på dit system (filen sversion.ini hhv. sofficerc er blevet fundet med et tilsvarende element).\n\nDu kan opdatere den eksisterende installation med dette program.\n\nInstallationsprogrammet (Setup) overskriver i givet fald nogle filer i dit %PRODUCTNAME installationsbibliotek med nye filer."
-46 = "Tillämpningen är redan installerad i ditt system (en sversion.ini-fil resp. sofficerc-fil med motsvarande post har hittats)."
-48 = "Aplikacja ta jest ju¿ zainstalowana w systemie komputera u¿ytkownika (poniewa¿ znaleziono informuj¹cy o tym plik sversion.ini lub sofficerc z odpowiednim wpisem)."
-81 = "ãŠä½¿ã„ã®ã‚·ã‚¹ãƒ†ãƒ ã«ã¯ãƒ—ãƒ­ã‚°ãƒ©ãƒ ãŒã™ã§ã«ã‚¤ãƒ³ã‚¹ãƒˆãƒ¼ãƒ«ã•ã‚Œã¦ã„ã¾ã™ã€‚ãã‚Œã‚‰ã®é …ç›®ã‚’å«ã‚€ sversion.ini ãƒ•ã‚¡ã‚¤ãƒ«ã€ã‚ã‚‹ã„ã¯ sofficerc ãƒ•ã‚¡ã‚¤ãƒ«ãŒè¦‹ã¤ã‹ã‚Šã¾ã—ãŸã€‚"
-82 = "ì‘ìš© í”„ë¡œê·¸ë¨ì´ ì´ë¯¸ ì‹œìŠ¤í…œì—  ì„¤ì¹˜ë˜ì–´ ìˆìŠµë‹ˆë‹¤(í•´ë‹¹ í•­ëª©ì´ ë“¤ì–´ ìˆëŠ” sversion.ini  ë˜ëŠ” sofficerc íŒŒì¼ì„ ì°¾ì•˜ìŒ)"
-86 = "åœ¨æ‚¨çš„ç³»ç»Ÿä¸­å·²ç»å­˜åœ¨ä¸€ä¸ªç¨‹åºç‰ˆæœ¬(å› ä¸ºæ‰¾åˆ°ä¸€ä¸ªsversion.ini æˆ–è€… sofficerc æ–‡ä»¶æ¡ç›®)ã€‚"
-88 = "åœ¨ç³»çµ±å…§å·²ç¶“å­˜åœ¨ä¸€å€‹ç¨‹å¼ç‰ˆæœ¬(åœ¨æª”æ¡ˆ sversion.ini æˆ– sofficerc å…§è¨˜éŒ„ç›¸æ‡‰çš„ç¾å­˜çš„ç‰ˆæœ¬çš„å®‰è£è·¯å¾‘)ã€‚\n\næ‚¨èƒ½å¤ æ›´æ–°ç¾æœ‰çš„ç‰ˆæœ¬ã€‚"
-90 = "Sisteminizde bir %PRODUCTNAME %PRODUCTVERSION sürümü mevcut durumda (sversion.ini ya da sofficerc dosyası bulundu).\n\nMevcut kurulumu şimdi kuracağınız sürümle güncelleyebilirsiniz.\n\nBu kurulum programı, %PRODUCTNAME kurulum dizinindeki bazı dosyalarını yeni dosyalarla değiştirecektir."
-96 = "åĞÇ ÇáÊØÈíŞ ãËÈÊ  ÈÇáİÚá Úáì äÙÇã ÇáÊÔÛíá ÇáÎÇÕ Èß (ÍíË Êã ÇáÚËæÑ Úáì ãáİ sversion.ini Ãæ sofficerc Èå ÅÏÎÇá íËÈÊ Ğáß)."
+01 = "The application is already installed on your system (a OOoversion.ini or an sofficerc file with a matching entry has been found)."
+49 = "Die Anwendung ist in Ihrem System bereits installiert (eine Datei OOoversion.ini bzw. sofficerc mit entsprechendem Eintrag wurde gefunden)."
+07 = "İòî ïğèëîæåíèå  óæå ñóùåñòâóåò â Âàøåé îïåğàöèîííîé ñèñòåìå (îáíàğóæåí ôàéë OOoversion.ini èëè sofficerc)."
+55 = "Das %PRODUCTNAME Scheduleserver %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm hat eine bestehende Installation erkannt.\n\nHaben Sie eine alte Installation eventuell nicht ?er das %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm deinstalliert? Dann l?chen Sie bitte jetzt Ihr altes %PRODUCTNAME Scheduleserver-Verzeichnis (Ihre eventuell dort enthaltenen Dokumente brauchen Sie nicht zu l?chen). Dann l?chen Sie - je nach Betriebssystem - die Datei Sversion.ini im Windows oder OS2 Verzeichnis oder .OOoversionrc oder sversion.prefs. Versuchen Sie nun neu, das %PRODUCTNAME %PRODUCTNAME %PRODUCTNAME Setup Programm zu starten."
+37 = "L'aplicació ja està instal·lada al sistema (s'ha trobat un fitxer OOoversion.ini o sofficerc amb una entrada que concideix)."
+03 = "A aplicação já se encontra instalada no seu sistema operativo (foi encontrado um ficheiro OOoversion.ini ou sofficerc com o respectivo registo)."
+30 = "Óôï óıóôçìÜ óáò õğÜñ÷åé Şäç ìéá İêäïóç ôçò åöáñìïãŞò (âñİèçêå ôï áñ÷åßï OOoversion.ini Ş sofficerc ìå ôçí áíôßóôïé÷ç åããñáöŞ)."
+31 = "Het programma is reeds in uw systeem geïnstalleerd (er werd een bestand gevonden met de naam OOoversion.ini resp. sofficerc met een dergelijk item)."
+33 = "L'application est déjà installée dans votre système (une entrée présente dans le fichier OOoversion.ini ou sofficerc a été détectée)."
+34 = "La aplicación ya está instalada en su sistema (se ha encontrado un archivo OOoversion.ini o sofficerc con la respectiva entrada)."
+35 = "Sovellus on jo asennettu järjestelmään (asennusohjelma löysi OOoversion.ini- tai sofficerc-tiedoston, jossa on asennettavaa ohjelmaa vastaavat merkinnät)."
+39 = "L'applicazione è già installata nel vostro sistema (è stato trovato un file OOoversion.ini o officerc contenete l'indicazione corrispondente)."
+45 = "Der findes allerede en version af %PRODUCTNAME %PRODUCTVERSION på dit system (filen OOoversion.ini hhv. sofficerc er blevet fundet med et tilsvarende element).\n\nDu kan opdatere den eksisterende installation med dette program.\n\nInstallationsprogrammet (Setup) overskriver i givet fald nogle filer i dit %PRODUCTNAME installationsbibliotek med nye filer."
+46 = "Tillämpningen är redan installerad i ditt system (en OOoversion.ini-fil resp. sofficerc-fil med motsvarande post har hittats)."
+48 = "Aplikacja ta jest ju¿ zainstalowana w systemie komputera u¿ytkownika (poniewa¿ znaleziono informuj¹cy o tym plik OOoversion.ini lub sofficerc z odpowiednim wpisem)."
+81 = "ãŠä½¿ã„ã®ã‚·ã‚¹ãƒ†ãƒ ã«ã¯ãƒ—ãƒ­ã‚°ãƒ©ãƒ ãŒã™ã§ã«ã‚¤ãƒ³ã‚¹ãƒˆãƒ¼ãƒ«ã•ã‚Œã¦ã„ã¾ã™ã€‚ãã‚Œã‚‰ã®é …ç›®ã‚’å«ã‚€ OOoversion.ini ãƒ•ã‚¡ã‚¤ãƒ«ã€ã‚ã‚‹ã„ã¯ sofficerc ãƒ•ã‚¡ã‚¤ãƒ«ãŒè¦‹ã¤ã‹ã‚Šã¾ã—ãŸã€‚"
+82 = "ì‘ìš© í”„ë¡œê·¸ë¨ì´ ì´ë¯¸ ì‹œìŠ¤í…œì—  ì„¤ì¹˜ë˜ì–´ ìˆìŠµë‹ˆë‹¤(í•´ë‹¹ í•­ëª©ì´ ë“¤ì–´ ìˆëŠ” OOoversion.ini  ë˜ëŠ” sofficerc íŒŒì¼ì„ ì°¾ì•˜ìŒ)"
+86 = "åœ¨æ‚¨çš„ç³»ç»Ÿä¸­å·²ç»å­˜åœ¨ä¸€ä¸ªç¨‹åºç‰ˆæœ¬(å› ä¸ºæ‰¾åˆ°ä¸€ä¸ªOOoversion.ini æˆ–è€… sofficerc æ–‡ä»¶æ¡ç›®)ã€‚"
+88 = "åœ¨ç³»çµ±å…§å·²ç¶“å­˜åœ¨ä¸€å€‹ç¨‹å¼ç‰ˆæœ¬(åœ¨æª”æ¡ˆ OOoversion.ini æˆ– sofficerc å…§è¨˜éŒ„ç›¸æ‡‰çš„ç¾å­˜çš„ç‰ˆæœ¬çš„å®‰è£è·¯å¾‘)ã€‚\n\næ‚¨èƒ½å¤ æ›´æ–°ç¾æœ‰çš„ç‰ˆæœ¬ã€‚"
+90 = "Sisteminizde bir %PRODUCTNAME %PRODUCTVERSION sürümü mevcut durumda (OOoversion.ini ya da sofficerc dosyası bulundu).\n\nMevcut kurulumu şimdi kuracağınız sürümle güncelleyebilirsiniz.\n\nBu kurulum programı, %PRODUCTNAME kurulum dizinindeki bazı dosyalarını yeni dosyalarla değiştirecektir."
+96 = "åĞÇ ÇáÊØÈíŞ ãËÈÊ  ÈÇáİÚá Úáì äÙÇã ÇáÊÔÛíá ÇáÎÇÕ Èß (ÍíË Êã ÇáÚËæÑ Úáì ãáİ OOoversion.ini Ãæ sofficerc Èå ÅÏÎÇá íËÈÊ Ğáß)."
 
 [STR_HELP_MIGRATION]
 01 = "%PRODUCTNAME can adopt personal data and configuration settings from a previous %PRODUCTNAME installation. To take advantage of this, activate the 'Import personal data' field. The path to the directory in the previous installation appears. If you do not want to accept data and settings from a previous version, unmark the 'Import personal data' check box."
--- oo_1.0.1_src/scp/source/player/player.scp.so-ooo-coexist	2001-11-12 19:47:25.000000000 +0100
+++ oo_1.0.1_src/scp/source/player/player.scp	2002-08-22 21:54:44.000000000 +0200
@@ -751,9 +751,9 @@ ProfileItem GID_PROFILEITEM_BOOTSTRAP_LO
     Order = 2;
     Key  = "Location";
 #ifdef WNT
-    Value  = "$SYSUSERCONFIG/sversion.ini";
+    Value  = "$SYSUSERCONFIG/OOoversion.ini";
 #else
-    Value  = "$SYSUSERCONFIG/.sversionrc";
+    Value  = "$SYSUSERCONFIG/.OOoversionrc";
 #endif
 End
 
@@ -1537,9 +1537,9 @@ End
 Profile GID_PROFILE_SVERSION
 	ModuleID		= GID_MODULE_ROOT;
 #ifdef UNX
-	Name = ".sversionrc";
+	Name = ".OOoversionrc";
 #else
-	Name = "sversion.ini";
+	Name = "OOoversion.ini";
 #endif
 //    Name            = PROFILENAME(sversion);
 	Dir				= PREDEFINED_CONFIGDIR;
--- oo_1.0.1_src/sal/osl/unx/profile.c.so-ooo-coexist	2001-11-21 15:29:50.000000000 +0100
+++ oo_1.0.1_src/sal/osl/unx/profile.c	2002-08-22 21:54:45.000000000 +0200
@@ -96,7 +96,7 @@
 
 #define SVERSION_LOCATION   STR_INI_METACFG
 #define SVERSION_FALLBACK   STR_INI_METASYS
-#define SVERSION_NAME   	"sversion"
+#define SVERSION_NAME   	"OOoversion"
 #define SVERSION_SECTION    "Versions"
 #define SVERSION_SOFFICE    "StarOffice"
 #define SVERSION_PROFILE    "sofficerc"
--- oo_1.0.1_src/tools/source/generic/config.cxx.so-ooo-coexist	2001-06-13 17:32:25.000000000 +0200
+++ oo_1.0.1_src/tools/source/generic/config.cxx	2002-08-22 21:54:45.000000000 +0200
@@ -239,9 +239,9 @@ static String ImplMakeConfigName( const 
 	else
 	{
 #ifdef UNX
-		aFileName = ::rtl::OUString::createFromAscii( ".sversionrc" );
+		aFileName = ::rtl::OUString::createFromAscii( ".OOoversionrc" );
 #else
-		aFileName = ::rtl::OUString::createFromAscii( "sversion.ini" );
+		aFileName = ::rtl::OUString::createFromAscii( "OOoversion.ini" );
 #endif
 	}
 
--- oo_1.0.1_src/unotools/inc/unotools/regpathhelper.hxx.so-ooo-coexist	2000-09-18 19:03:53.000000000 +0200
+++ oo_1.0.1_src/unotools/inc/unotools/regpathhelper.hxx	2002-08-22 21:54:45.000000000 +0200
@@ -73,7 +73,7 @@ namespace utl
  * This function returns a path to the user registry file.
  * Search for the user registry using the following rules:
  * <dl>
- * <dt> 1. (for further use) search in sversion.ini (.sversionrc) for an entry
+ * <dt> 1. (for further use) search in OOoversion.ini (.OOoversionrc) for an entry
  *		REGISTRY_VERSION_STRING (example: UserRegistry 5.0/505=test.rdb) in the section 
  *		[Registry]. If found, then take this value instead of the name "user.rdb".
  * <dt> 2. Search in the config directory of the user for a file "user.rdb". If 
--- oo_1.0.1_src/comphelper/inc/comphelper/regpathhelper.hxx.so-ooo-coexist	2000-09-29 13:28:15.000000000 +0200
+++ oo_1.0.1_src/comphelper/inc/comphelper/regpathhelper.hxx	2002-08-22 21:54:45.000000000 +0200
@@ -73,7 +73,7 @@ namespace comphelper
  * This function returns a path to the user registry file.
  * Search for the user registry using the following rules:
  * <dl>
- * <dt> 1. (for further use) search in sversion.ini (.sversionrc) for an entry
+ * <dt> 1. (for further use) search in OOoversion.ini (.OOoversionrc) for an entry
  *		REGISTRY_VERSION_STRING (example: UserRegistry 5.0/505=test.rdb) in the section
  *		[Registry]. If found, then take this value instead of the name "user.rdb".
  * <dt> 2. Search in the config directory of the user for a file "user.rdb". If
--- oo_1.0.1_src/comphelper/source/misc/regpathhelper.cxx.so-ooo-coexist	2001-07-23 12:14:42.000000000 +0200
+++ oo_1.0.1_src/comphelper/source/misc/regpathhelper.cxx	2002-08-22 21:54:45.000000000 +0200
@@ -93,7 +93,7 @@ using namespace rtl;
 #define SYSTEM_REGISTRY_NAME_ENV 	"STAR_REGISTRY"
 #define REGISTRY_SYSTEM_NAME	"applicat.rdb"
 
-#define REGISTRY_LOCAL_NAME		"user60.rdb"
+#define REGISTRY_LOCAL_NAME		"userOOo10.rdb"
 
 #ifdef SAL_UNX
 #define CONFIG_PATH_PREFIX		"."
--- oo_1.0.1_src/udkapi/com/sun/star/registry/DefaultRegistry.idl.so-ooo-coexist	2001-11-16 16:05:59.000000000 +0100
+++ oo_1.0.1_src/udkapi/com/sun/star/registry/DefaultRegistry.idl	2002-08-22 21:54:45.000000000 +0200
@@ -158,7 +158,7 @@
 	</dt>
 	<dd>
 	<dl>
-	<dt> 1. (for further use) search in sversion.ini (.sversionrc) for an entry
+	<dt> 1. (for further use) search in OOoversion.ini (.OOoversionrc) for an entry
 	REGISTRY_VERSION_STRING (example: UserRegistry 5.0/505=test.rdb) in the section 
 	[Registry]. If found, then take this value instead of "user.rdb" as "REGISTRY NAME".
 	</dt>
--- oo_1.0.1_src/padmin/source/adddlg.cxx.so-ooo-coexist	2001-12-19 16:36:55.000000000 +0100
+++ oo_1.0.1_src/padmin/source/adddlg.cxx	2002-08-22 21:54:45.000000000 +0200
@@ -1077,7 +1077,7 @@ String AddPrinterDialog::getOldPrinterLo
         if( access( aFileName.GetBuffer(), F_OK ) )
         {
             aFileName = pHome;
-            aFileName.Append( "/.sversionrc" );
+            aFileName.Append( "/.OOoversionrc" );
             Config aSVer( String( aFileName, aEncoding ) );
             aSVer.SetGroup( "Versions" );
             aFileName = aSVer.ReadKey( "StarOffice 5.2" );
