--- javaunohelper/source/makefile.mk.orig	2003-09-22 12:14:58.000000000 -0700
+++ javaunohelper/source/makefile.mk	2003-09-22 12:18:51.000000000 -0700
@@ -76,6 +76,8 @@
 
 # ------------------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)"!=""
+
 UNOUCRDEP=$(SOLARBINDIR)$/udkapi.rdb
 UNOUCRRDB=$(SOLARBINDIR)$/udkapi.rdb
 
@@ -153,6 +155,8 @@
 
 DEF2NAME=$(SHL2TARGET)
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 # --- Targets ------------------------------------------------------
 
 .INCLUDE :	target.mk
