diff -urN ooo-build-1.9.78.org/bin/build-ooo ooo-build-1.9.78/bin/build-ooo
--- ooo-build-1.9.78.org/bin/build-ooo	2005-02-08 20:15:30.000000000 +0100
+++ ooo-build-1.9.78/bin/build-ooo	2005-02-12 22:46:42.877076883 +0100
@@ -6,7 +6,7 @@
 . ./setup
 
 if test "z$BUILD_WIN32" = "z"; then
-    if ! test -f /proc/kcore; then
+    if ! test -f /proc/cpuinfo; then
         echo "Looks like proc isn't mounted - this means almost certain"
         echo "Java related weird build failure: please check /proc"
         exit 1;
diff -urN ooo-build-1.9.78.org/patches/src680-m78/apply ooo-build-1.9.78/patches/src680-m78/apply
--- ooo-build-1.9.78.org/patches/src680-m78/apply	2005-02-10 16:27:47.000000000 +0100
+++ ooo-build-1.9.78/patches/src680-m78/apply	2005-02-12 22:26:54.861576226 +0100
@@ -24,6 +24,10 @@
 Debian : DebianBase
 # Ark Linux
 Ark: LinuxCommon, 64bit, ArkOnly
+# www.pld-linux.org
+PLDBase: LinuxCommon, NonArkCommon
+PLD: PLDBase
+PLD64: PLDBase, 64bit
 # Win32
 Win32 : Win32Common, NovellOnly, NovellOnlyWin32
 
