--- oo_1.0.1_src/sj2/stardiv/applet/DocumentProxy.java.old	Mon Sep 18 18:54:03 2000
+++ oo_1.0.1_src/sj2/stardiv/applet/DocumentProxy.java	Fri Nov  1 21:34:29 2002
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -73,6 +73,8 @@
 import java.util.Hashtable;
 import java.util.Observer;
 import java.util.Observable;
+import java.util.Iterator;
+import java.io.InputStream;
 
 import java.net.URL;
 
@@ -96,6 +98,22 @@
         return documentProxy;
     }
 
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public void setStream( String key,InputStream stream)
+		throws java.io.IOException {
+	}
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public InputStream getStream( String key) {
+		return null;
+	}
+	// AppletContext. This method is new since 1.4. We insert it so as to
+	// have the project buildable
+	public Iterator getStreamKeys() {
+		return null;
+	}
+		
 
     /*
     ** interface cachable methods
--- oo_1.0.1_src/sj2/stardiv/applet/AppletViewer.java.old	Mon Sep 18 18:54:03 2000
+++ oo_1.0.1_src/sj2/stardiv/applet/AppletViewer.java	Fri Nov  1 21:46:41 2002
@@ -2,9 +2,9 @@
  *
  *  $RCSfile$
  *
- *  $Revision$
+ *  $Revision$
  *
- *  last change: $Author$ $Date$
+ *  last change: $Author$ $Date$
  *
  *  The Contents of this file are made available subject to the terms of
  *  either of the following licenses
@@ -196,7 +196,8 @@
     public AppletViewer(int x, int y, URL doc, Hashtable atts, PrintStream statusMsgStream) {
 //  		resourceViewer = new stardiv.util.ResourceViewer();
 //  		resourceViewer.show();
-		System.err.println("#*#*#*:" + sun.awt.ScreenUpdater.updater);
+		//sun.awt.ScreenUpdater does not exist in Java 1.4
+//		System.err.println("#*#*#*:" + sun.awt.ScreenUpdater.updater);
 		this.statusMsgStream = statusMsgStream;
 		this.atts = atts;
 
