--- setup2/source/custom/jvmsetup/loader/makefile.mk.orig	2003-09-22 11:08:19.000000000 -0700
+++ setup2/source/custom/jvmsetup/loader/makefile.mk	2003-09-22 13:28:10.000000000 -0700
@@ -72,6 +72,9 @@
 .INCLUDE : sv.mk
 
 # --- Files --------------------------------------------------------
+
+.IF "$(SOLAR_JAVA)"!=""
+
 CXXFILES=	jvmapp.cxx
 
 OBJFILES=	$(OBJ)$/jvmapp.obj
@@ -128,6 +131,8 @@
 UNIXTEXT= \
 	$(MISC)$/jvmsetup.sh
 
+.ENDIF "$(SOLAR_JAVA)"!=""
+
 .INCLUDE :  target.mk
 
 # ------------------------------------------------------------------
--- setup2/source/custom/jvmsetup/makefile.mk.orig	2003-09-22 11:08:19.000000000 -0700
+++ setup2/source/custom/jvmsetup/makefile.mk	2003-09-22 13:31:00.000000000 -0700
@@ -73,6 +73,8 @@
 
 # --- Files --------------------------------------------------------
 
+.IF "$(SOLAR_JAVA)"!=""
+
 .DIRCACHE=NO
 
 SRCFILES =	jvmdlg.src
@@ -113,6 +115,8 @@
 
 RC_SUBDIRSDEPS=$(SLB)$/jvm.lib
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 # --- Targets --------------------------------------------------------------
 
 .INCLUDE : target.mk
--- setup2/source/ui/main.cxx.orig	2003-09-22 11:08:19.000000000 -0700
+++ setup2/source/ui/main.cxx	2003-09-22 13:27:05.000000000 -0700
@@ -78,7 +78,9 @@
 #include <sal/main.h>
 #endif
 
+#ifdef SOLAR_JAVA
 #include "jvmaccess/javainfo.hxx"
+#endif
 
 #ifndef _SV_HELP_HXX //autogen
 #include <vcl/help.hxx>
@@ -2523,6 +2525,7 @@
 static sal_Bool hasJavarc();
 void prepareJava(const char * szStartPath)
 {
+#ifdef SOLAR_JAVA
     if(szStartPath)
     {
         //check if we are started from an installation. This is done by looking for the classes
@@ -2656,6 +2659,7 @@
 #endif            
          }
     }
+#endif // SOLAR_JAVA
 }    
 
 static sal_Bool hasJavarc()
--- setup2/util/makefile.mk.orig	2003-09-22 11:08:19.000000000 -0700
+++ setup2/util/makefile.mk	2003-09-22 13:27:05.000000000 -0700
@@ -186,7 +186,6 @@
 		$(SVTOOLLIB) \
 		$(VCLLIB) \
 		$(TOOLSLIB) \
-		$(JVMACCESSLIB) \
 		$(COMPHELPERLIB) \
 		$(CPPUHELPERLIB) \
 		$(CPPULIB) \
@@ -195,7 +194,10 @@
 		$(SALHELPERLIB) \
 		$(UNOTOOLSLIB)  \
 		$(SALLIB)
-
+.IF "$(SOLAR_JAVA)"!=""
+APP2STDLIBS+=\
+		$(JVMACCESSLIB)
+.ENDIF
 
 .IF "$(GUI)"=="WNT"
 APP2STDLIBS += uwinapi.lib advapi32.lib gdi32.lib shell32.lib ole32.lib uuid.lib version.lib \
