--- ooo-build-ooc680-m7/configure.in~	2006-06-28 17:19:08.000000000 +0200
+++ ooo-build-ooc680-m7/configure.in	2006-07-09 15:34:05.290959182 +0200
@@ -408,6 +408,7 @@
         Gentoo*) VENDORNAME="Gentoo" ;;
         Mandriva*) VENDORNAME="Mandriva" ;;
         SUSE*|Novell*|NLD*) VENDORNAME="Novell" ;;
+	PLD*) VENDORNAME="PLD" ;;
         *) AC_MSG_WARN([Cannot guess vendor from distro - please add it to configure.in]) ;;
 esac
 AC_MSG_RESULT([$VENDORNAME])
@@ -886,6 +887,7 @@
 distro-configs/translate-org-za-win32.conf
 distro-configs/rpl2.conf
 distro-configs/rpl2-64.conf
+distro-configs/PLD.conf
 etc/Makefile
 patches/Makefile
 patches/cairo/Makefile
