--- jvmaccess/source/makefile.mk.orig	2003-09-22 12:40:14.000000000 -0700
+++ jvmaccess/source/makefile.mk	2003-09-22 12:41:58.000000000 -0700
@@ -67,6 +67,8 @@
 
 .INCLUDE: settings.mk
 
+.IF "$(SOLAR_JAVA)"!=""
+
 UNOTYPES= com.sun.star.uno.TypeClass com.sun.star.uno.XInterface
 
 UNOUCRDEP = $(SOLARBINDIR)$/udkapi.rdb
@@ -80,4 +82,6 @@
     $(SLO)$/virtualmachine.obj \
     $(SLO)$/windows.obj
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 .INCLUDE: target.mk
--- jvmaccess/util/makefile.mk.orig	2003-09-22 12:40:19.000000000 -0700
+++ jvmaccess/util/makefile.mk	2003-09-22 12:41:35.000000000 -0700
@@ -71,6 +71,8 @@
 
 .INCLUDE: settings.mk
 
+.IF "$(SOLAR_JAVA)"!=""
+
 .IF "$(UNIXVERSIONNAMES)" == ""
 SHL1TARGET = $(TARGET)$(UDK_MAJOR)$(COMID)
 .ELSE # UNIXVERSIONNAMES
@@ -96,4 +98,6 @@
 
 DEF1NAME = $(SHL1TARGET)
 
+.ENDIF # "$(SOLAR_JAVA)"!=""
+
 .INCLUDE: target.mk
