diff -urN ooo-build-1.1.54.org/bin/setup.in ooo-build-1.1.54/bin/setup.in
--- ooo-build-1.1.54.org/bin/setup.in	2004-04-30 02:50:20.270649456 +0200
+++ ooo-build-1.1.54/bin/setup.in	2004-04-30 15:34:39.655963192 +0200
@@ -31,6 +31,7 @@
 	LANGS="$LANGS,HUNG";
 fi
 
+if test "z$CONFIGURE_OPTIONS" = "z"; then
 if test "z$DISTRO" = "zRHFedora"; then
     CONFIGURE_OPTIONS="--with-lang=$LANGS	\
 				--enable-libart		\
@@ -58,6 +59,7 @@
 if test "z$DISTRO" = "zKDE"; then
     CONFIGURE_OPTIONS="$CONFIGURE_OPTIONS --with-widgetset=kde"
 fi
+fi
 
 ENABLE_BONOBO=@ENABLE_BONOBO@
 ENABLE_JAVA=@ENABLE_JAVA@
