--- libreoffice-6.0.5.2/pyuno/source/module/pyuno_impl.hxx~	2018-06-20 00:27:30.000000000 +0200
+++ libreoffice-6.0.5.2/pyuno/source/module/pyuno_impl.hxx	2018-07-02 20:29:01.747956284 +0200
@@ -82,7 +82,7 @@ inline PyObject* PyStr_FromString(const
 
 inline char * PyStr_AsString(PyObject *object)
 {
-    return PyUnicode_AsUTF8(object);
+    return (char *)PyUnicode_AsUTF8(object);
 }
 
 inline bool PyStr_Check(PyObject const *object)
--- libreoffice-6.0.5.2/pyuno/source/module/pyuno_util.cxx~	2018-06-20 00:27:30.000000000 +0200
+++ libreoffice-6.0.5.2/pyuno/source/module/pyuno_util.cxx	2018-07-02 21:11:29.419418239 +0200
@@ -69,7 +69,7 @@ OUString pyString2ustring( PyObject *pys
 #else
 #if PY_MAJOR_VERSION >= 3
     Py_ssize_t size(0);
-    char *pUtf8(PyUnicode_AsUTF8AndSize(pystr, &size));
+    char *pUtf8((char *)PyUnicode_AsUTF8AndSize(pystr, &size));
     ret = OUString(pUtf8, size, RTL_TEXTENCODING_UTF8);
 #else
     PyObject* pUtf8 = PyUnicode_AsUTF8String(pystr);
